<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>7 Population Structure via conStruct | MarGen2022</title>
  <meta name="description" content="" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="7 Population Structure via conStruct | MarGen2022" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="7 Population Structure via conStruct | MarGen2022" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="week-5--r-continued.html"/>
<link rel="next" href="week-6--principle-component-analyses.html"/>
<script src="libs/header-attrs-2.11.16/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Marine Genomics 2022</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction to Marine Genomics<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#cloud-computing-versus-individual-laptops"><i class="fa fa-check"></i><b>1.1</b> Cloud computing versus individual laptops<span></span></a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#meet-our-instructors"><i class="fa fa-check"></i><b>1.2</b> Meet our instructors!<span></span></a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#schedule"><i class="fa fa-check"></i><b>1.3</b> Schedule<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="week-1--welcome.html"><a href="week-1--welcome.html"><i class="fa fa-check"></i><b>2</b> Week 1- Welcome!<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="week-1--welcome.html"><a href="week-1--welcome.html#introduction-to-shell-computing-via-the-data-carpentry-tutorial"><i class="fa fa-check"></i><b>2.1</b> Introduction to shell computing via the data carpentry tutorial<span></span></a></li>
<li class="chapter" data-level="2.2" data-path="week-1--welcome.html"><a href="week-1--welcome.html#how-to-access-the-shell"><i class="fa fa-check"></i><b>2.2</b> How to access the shell<span></span></a></li>
<li class="chapter" data-level="2.3" data-path="week-1--welcome.html"><a href="week-1--welcome.html#best-practices-for-jetstream"><i class="fa fa-check"></i><b>2.3</b> Best practices for Jetstream<span></span></a></li>
<li class="chapter" data-level="2.4" data-path="week-1--welcome.html"><a href="week-1--welcome.html#week-1-objectives"><i class="fa fa-check"></i><b>2.4</b> Week 1 Objectives<span></span></a></li>
<li class="chapter" data-level="2.5" data-path="week-1--welcome.html"><a href="week-1--welcome.html#navigating-your-file-system"><i class="fa fa-check"></i><b>2.5</b> Navigating your file system<span></span></a></li>
<li class="chapter" data-level="2.6" data-path="week-1--welcome.html"><a href="week-1--welcome.html#shortcut-tab-completion"><i class="fa fa-check"></i><b>2.6</b> Shortcut: Tab Completion<span></span></a></li>
<li class="chapter" data-level="2.7" data-path="week-1--welcome.html"><a href="week-1--welcome.html#summary-key-points"><i class="fa fa-check"></i><b>2.7</b> Summary &amp; Key Points<span></span></a></li>
<li class="chapter" data-level="2.8" data-path="week-1--welcome.html"><a href="week-1--welcome.html#navigating-files-and-directories"><i class="fa fa-check"></i><b>2.8</b> Navigating Files and Directories<span></span></a></li>
<li class="chapter" data-level="2.9" data-path="week-1--welcome.html"><a href="week-1--welcome.html#moving-around-the-file-system"><i class="fa fa-check"></i><b>2.9</b> Moving around the file system<span></span></a></li>
<li class="chapter" data-level="2.10" data-path="week-1--welcome.html"><a href="week-1--welcome.html#navigating-practice-and-finding-hidden-directories"><i class="fa fa-check"></i><b>2.10</b> Navigating practice and finding hidden directories<span></span></a></li>
<li class="chapter" data-level="2.11" data-path="week-1--welcome.html"><a href="week-1--welcome.html#examining-the-contents-of-other-directories"><i class="fa fa-check"></i><b>2.11</b> Examining the contents of other directories<span></span></a></li>
<li class="chapter" data-level="2.12" data-path="week-1--welcome.html"><a href="week-1--welcome.html#full-vs-relative-paths"><i class="fa fa-check"></i><b>2.12</b> Full vs Relative Paths<span></span></a></li>
<li class="chapter" data-level="2.13" data-path="week-1--welcome.html"><a href="week-1--welcome.html#navigational-shortcuts"><i class="fa fa-check"></i><b>2.13</b> Navigational shortcuts<span></span></a></li>
<li class="chapter" data-level="2.14" data-path="week-1--welcome.html"><a href="week-1--welcome.html#key-points"><i class="fa fa-check"></i><b>2.14</b> Key Points<span></span></a></li>
<li class="chapter" data-level="2.15" data-path="week-1--welcome.html"><a href="week-1--welcome.html#creature-of-the-week"><i class="fa fa-check"></i><b>2.15</b> Creature of the Week!<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html"><i class="fa fa-check"></i><b>3</b> Week 2-Working With Files<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#our-data-set-fastq-files"><i class="fa fa-check"></i><b>3.1</b> Our data set: FASTQ files<span></span></a></li>
<li class="chapter" data-level="3.2" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#wildcards"><i class="fa fa-check"></i><b>3.2</b> Wildcards<span></span></a></li>
<li class="chapter" data-level="3.3" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#command-history"><i class="fa fa-check"></i><b>3.3</b> Command History<span></span></a></li>
<li class="chapter" data-level="3.4" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#examining-files"><i class="fa fa-check"></i><b>3.4</b> Examining Files<span></span></a></li>
<li class="chapter" data-level="3.5" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#details-on-the-fastq-format"><i class="fa fa-check"></i><b>3.5</b> Details on the FASTQ format<span></span></a></li>
<li class="chapter" data-level="3.6" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#creating-moving-copying-and-removing"><i class="fa fa-check"></i><b>3.6</b> Creating, moving, copying, and removing<span></span></a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#copying-files"><i class="fa fa-check"></i><b>3.6.1</b> Copying Files<span></span></a></li>
<li class="chapter" data-level="3.6.2" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#creating-directories"><i class="fa fa-check"></i><b>3.6.2</b> Creating Directories<span></span></a></li>
<li class="chapter" data-level="3.6.3" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#moving-renaming"><i class="fa fa-check"></i><b>3.6.3</b> Moving / Renaming<span></span></a></li>
<li class="chapter" data-level="3.6.4" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#file-permissions"><i class="fa fa-check"></i><b>3.6.4</b> File Permissions<span></span></a></li>
<li class="chapter" data-level="3.6.5" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#removing"><i class="fa fa-check"></i><b>3.6.5</b> Removing<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#redirections"><i class="fa fa-check"></i><b>3.7</b> Redirections<span></span></a></li>
<li class="chapter" data-level="3.8" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#searching-files"><i class="fa fa-check"></i><b>3.8</b> Searching files<span></span></a></li>
<li class="chapter" data-level="3.9" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#redirecting-output"><i class="fa fa-check"></i><b>3.9</b> Redirecting output<span></span></a></li>
<li class="chapter" data-level="3.10" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#writing-for-loops"><i class="fa fa-check"></i><b>3.10</b> Writing for loops<span></span></a></li>
<li class="chapter" data-level="3.11" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#using-basename-in-for-loops"><i class="fa fa-check"></i><b>3.11</b> Using Basename in for loops<span></span></a></li>
<li class="chapter" data-level="3.12" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#writing-scripts-and-working-with-data"><i class="fa fa-check"></i><b>3.12</b> Writing Scripts and Working with Data<span></span></a></li>
<li class="chapter" data-level="3.13" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#writing-files"><i class="fa fa-check"></i><b>3.13</b> Writing files<span></span></a></li>
<li class="chapter" data-level="3.14" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#writing-scripts"><i class="fa fa-check"></i><b>3.14</b> Writing scripts<span></span></a></li>
<li class="chapter" data-level="3.15" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#moving-and-downloading-data"><i class="fa fa-check"></i><b>3.15</b> Moving and Downloading Data<span></span></a>
<ul>
<li class="chapter" data-level="3.15.1" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#getting-data-from-the-cloud"><i class="fa fa-check"></i><b>3.15.1</b> Getting data from the cloud<span></span></a></li>
<li class="chapter" data-level="3.15.2" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#uploading-and-downloading-data-to-your-virtual-machine-with-scp---unix"><i class="fa fa-check"></i><b>3.15.2</b> Uploading and Downloading Data to your Virtual Machine with scp - UNIX<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.16" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#awk"><i class="fa fa-check"></i><b>3.16</b> Awk<span></span></a></li>
<li class="chapter" data-level="3.17" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#creature-of-the-week-1"><i class="fa fa-check"></i><b>3.17</b> Creature of the Week!<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html"><i class="fa fa-check"></i><b>4</b> Week 3- What is a Genetic Variant?<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#to-get-started-lets-dowload-the-data-and-install-a-few-programs"><i class="fa fa-check"></i><b>4.1</b> To get started lets dowload the data and install a few programs<span></span></a></li>
<li class="chapter" data-level="4.2" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#raw-read-quality-control"><i class="fa fa-check"></i><b>4.2</b> Raw read quality control<span></span></a></li>
<li class="chapter" data-level="4.3" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#trimming-to-remove-adapters"><i class="fa fa-check"></i><b>4.3</b> Trimming to remove adapters<span></span></a></li>
<li class="chapter" data-level="4.4" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#building-an-index-of-our-genome"><i class="fa fa-check"></i><b>4.4</b> Building an index of our genome<span></span></a></li>
<li class="chapter" data-level="4.5" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#map-reads-to-the-genome"><i class="fa fa-check"></i><b>4.5</b> Map reads to the genome<span></span></a></li>
<li class="chapter" data-level="4.6" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#sam-to-bam-file-conversion"><i class="fa fa-check"></i><b>4.6</b> sam to bam file conversion<span></span></a></li>
<li class="chapter" data-level="4.7" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#genotype-likelihoods"><i class="fa fa-check"></i><b>4.7</b> Genotype likelihoods<span></span></a></li>
<li class="chapter" data-level="4.8" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#creature-of-the-week-2"><i class="fa fa-check"></i><b>4.8</b> Creature of the Week!<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html"><i class="fa fa-check"></i><b>5</b> Week 4- The R environment<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#lesson-1-orientation-to-r"><i class="fa fa-check"></i><b>5.1</b> Lesson 1: Orientation to R<span></span></a></li>
<li class="chapter" data-level="5.2" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#manipulating-a-vector-object"><i class="fa fa-check"></i><b>5.2</b> Manipulating a vector object<span></span></a></li>
<li class="chapter" data-level="5.3" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#operations-act-on-each-element-of-a-vector"><i class="fa fa-check"></i><b>5.3</b> Operations act on each element of a vector:<span></span></a></li>
<li class="chapter" data-level="5.4" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#operations-can-also-work-with-two-vectors"><i class="fa fa-check"></i><b>5.4</b> Operations can also work with two vectors:<span></span></a></li>
<li class="chapter" data-level="5.5" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#a-few-tips-below-for-working-with-objects"><i class="fa fa-check"></i><b>5.5</b> A few tips below for working with objects:<span></span></a></li>
<li class="chapter" data-level="5.6" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#practice-r-operations"><i class="fa fa-check"></i><b>5.6</b> Practice R Operations<span></span></a></li>
<li class="chapter" data-level="5.7" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#characterizing-a-dataframe"><i class="fa fa-check"></i><b>5.7</b> 1.2 Characterizing a dataframe<span></span></a></li>
<li class="chapter" data-level="5.8" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#how-to-access-parts-of-the-data"><i class="fa fa-check"></i><b>5.8</b> How to access parts of the data:<span></span></a></li>
<li class="chapter" data-level="5.9" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#data-manipulation"><i class="fa fa-check"></i><b>5.9</b> Data Manipulation<span></span></a></li>
<li class="chapter" data-level="5.10" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#practice-exploring-a-dataframe"><i class="fa fa-check"></i><b>5.10</b> Practice exploring a dataframe<span></span></a></li>
<li class="chapter" data-level="5.11" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#subsetting-datasets-logicals"><i class="fa fa-check"></i><b>5.11</b> 1.3 Subsetting datasets &amp; logicals<span></span></a></li>
<li class="chapter" data-level="5.12" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#practice-subsetting-datasetslogicals"><i class="fa fa-check"></i><b>5.12</b> Practice Subsetting datasets/logicals<span></span></a></li>
<li class="chapter" data-level="5.13" data-path="week-4--the-r-environment.html"><a href="week-4--the-r-environment.html#creature-of-the-week-3"><i class="fa fa-check"></i><b>5.13</b> Creature of the Week!<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html"><i class="fa fa-check"></i><b>6</b> Week 5- R Continued<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#plotting"><i class="fa fa-check"></i><b>6.1</b> 2.1 Plotting<span></span></a></li>
<li class="chapter" data-level="6.2" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#scatterplots"><i class="fa fa-check"></i><b>6.2</b> Scatterplots<span></span></a></li>
<li class="chapter" data-level="6.3" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#customizing-your-plot"><i class="fa fa-check"></i><b>6.3</b> Customizing your plot<span></span></a></li>
<li class="chapter" data-level="6.4" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#practice-problems-2.1"><i class="fa fa-check"></i><b>6.4</b> Practice Problems 2.1<span></span></a></li>
<li class="chapter" data-level="6.5" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#plotting-with-ggplot2"><i class="fa fa-check"></i><b>6.5</b> 2.2 plotting with ggplot2<span></span></a></li>
<li class="chapter" data-level="6.6" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#practice-problems-2.2"><i class="fa fa-check"></i><b>6.6</b> Practice Problems 2.2<span></span></a></li>
<li class="chapter" data-level="6.7" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#for-loops-and-the-apply-family-of-functions"><i class="fa fa-check"></i><b>6.7</b> For loops and the apply family of functions<span></span></a></li>
<li class="chapter" data-level="6.8" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#apply-family"><i class="fa fa-check"></i><b>6.8</b> apply family<span></span></a>
<ul>
<li class="chapter" data-level="6.8.1" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#sapply"><i class="fa fa-check"></i><b>6.8.1</b> sapply<span></span></a></li>
<li class="chapter" data-level="6.8.2" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#apply"><i class="fa fa-check"></i><b>6.8.2</b> apply<span></span></a></li>
<li class="chapter" data-level="6.8.3" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#lapply-list-apply"><i class="fa fa-check"></i><b>6.8.3</b> lapply – “list” apply<span></span></a></li>
<li class="chapter" data-level="6.8.4" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#tapply---per-type-apply"><i class="fa fa-check"></i><b>6.8.4</b> tapply - “per Type” apply<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6.9" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#exercise-2.3-apply-and-tapply"><i class="fa fa-check"></i><b>6.9</b> Exercise 2.3 apply and tapply<span></span></a></li>
<li class="chapter" data-level="6.10" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#creature-of-the-week-4"><i class="fa fa-check"></i><b>6.10</b> Creature of the Week!<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="population-structure-via-construct.html"><a href="population-structure-via-construct.html"><i class="fa fa-check"></i><b>7</b> Population Structure via conStruct<span></span></a>
<ul>
<li class="chapter" data-level="7.1" data-path="population-structure-via-construct.html"><a href="population-structure-via-construct.html#download-the-data"><i class="fa fa-check"></i><b>7.1</b> Download the data<span></span></a></li>
<li class="chapter" data-level="7.2" data-path="population-structure-via-construct.html"><a href="population-structure-via-construct.html#update-our-compiler-in-bashunix-and-install-everything-for-r-in-the-terminal-in-bashunix-not-in-rstudio"><i class="fa fa-check"></i><b>7.2</b> update our compiler in bash/UNIX and install everything for R in the terminal (in bash/UNIX, NOT in Rstudio)<span></span></a></li>
<li class="chapter" data-level="7.3" data-path="population-structure-via-construct.html"><a href="population-structure-via-construct.html#install-construct-in-r"><i class="fa fa-check"></i><b>7.3</b> install conStruct in R<span></span></a></li>
<li class="chapter" data-level="7.4" data-path="population-structure-via-construct.html"><a href="population-structure-via-construct.html#read-in-the-data"><i class="fa fa-check"></i><b>7.4</b> read in the data<span></span></a></li>
<li class="chapter" data-level="7.5" data-path="population-structure-via-construct.html"><a href="population-structure-via-construct.html#get-our-meta-data-read-into-r-and-ready-to-use"><i class="fa fa-check"></i><b>7.5</b> Get our meta data read into R and ready to use<span></span></a></li>
<li class="chapter" data-level="7.6" data-path="population-structure-via-construct.html"><a href="population-structure-via-construct.html#run-a-structure-analysis"><i class="fa fa-check"></i><b>7.6</b> Run a structure analysis<span></span></a></li>
<li class="chapter" data-level="7.7" data-path="population-structure-via-construct.html"><a href="population-structure-via-construct.html#incoorporate-geographic-distances-in-a-spatial-model"><i class="fa fa-check"></i><b>7.7</b> Incoorporate geographic distances in a spatial model<span></span></a></li>
<li class="chapter" data-level="7.8" data-path="population-structure-via-construct.html"><a href="population-structure-via-construct.html#comparisons-across-different-values-of-k"><i class="fa fa-check"></i><b>7.8</b> Comparisons across different values of K<span></span></a></li>
<li class="chapter" data-level="7.9" data-path="population-structure-via-construct.html"><a href="population-structure-via-construct.html#exercises"><i class="fa fa-check"></i><b>7.9</b> Exercises<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="week-6--principle-component-analyses.html"><a href="week-6--principle-component-analyses.html"><i class="fa fa-check"></i><b>8</b> Week 6- Principle Component Analyses<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="week-6--principle-component-analyses.html"><a href="week-6--principle-component-analyses.html#download-the-data-1"><i class="fa fa-check"></i><b>8.1</b> Download the data<span></span></a></li>
<li class="chapter" data-level="8.2" data-path="week-6--principle-component-analyses.html"><a href="week-6--principle-component-analyses.html#installing-programs"><i class="fa fa-check"></i><b>8.2</b> Installing programs<span></span></a></li>
<li class="chapter" data-level="8.3" data-path="week-6--principle-component-analyses.html"><a href="week-6--principle-component-analyses.html#run-pcangsd-on-our-data"><i class="fa fa-check"></i><b>8.3</b> Run pcangsd on our data<span></span></a></li>
<li class="chapter" data-level="8.4" data-path="week-6--principle-component-analyses.html"><a href="week-6--principle-component-analyses.html#setting-up-a-new-project-in-r"><i class="fa fa-check"></i><b>8.4</b> Setting up a new project in R<span></span></a></li>
<li class="chapter" data-level="8.5" data-path="week-6--principle-component-analyses.html"><a href="week-6--principle-component-analyses.html#reading-data-into-r"><i class="fa fa-check"></i><b>8.5</b> Reading data into R<span></span></a></li>
<li class="chapter" data-level="8.6" data-path="week-6--principle-component-analyses.html"><a href="week-6--principle-component-analyses.html#exercises-1"><i class="fa fa-check"></i><b>8.6</b> Exercises<span></span></a></li>
<li class="chapter" data-level="8.7" data-path="week-6--principle-component-analyses.html"><a href="week-6--principle-component-analyses.html#creature-of-the-week-5"><i class="fa fa-check"></i><b>8.7</b> Creature of the Week!<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="week-8--fst-and-outlier-analysis.html"><a href="week-8--fst-and-outlier-analysis.html"><i class="fa fa-check"></i><b>9</b> Week 8- F<sub>st</sub> and outlier analysis<span></span></a>
<ul>
<li class="chapter" data-level="9.1" data-path="week-8--fst-and-outlier-analysis.html"><a href="week-8--fst-and-outlier-analysis.html#download-the-data-2"><i class="fa fa-check"></i><b>9.1</b> Download the data<span></span></a></li>
<li class="chapter" data-level="9.2" data-path="week-8--fst-and-outlier-analysis.html"><a href="week-8--fst-and-outlier-analysis.html#getting-r-set-up"><i class="fa fa-check"></i><b>9.2</b> Getting R set up<span></span></a></li>
<li class="chapter" data-level="9.3" data-path="week-8--fst-and-outlier-analysis.html"><a href="week-8--fst-and-outlier-analysis.html#finding-outliers-using-pcadapt"><i class="fa fa-check"></i><b>9.3</b> Finding outliers using pcadapt<span></span></a></li>
<li class="chapter" data-level="9.4" data-path="week-8--fst-and-outlier-analysis.html"><a href="week-8--fst-and-outlier-analysis.html#pcadapt-exercises"><i class="fa fa-check"></i><b>9.4</b> pcadapt Exercises<span></span></a></li>
<li class="chapter" data-level="9.5" data-path="week-8--fst-and-outlier-analysis.html"><a href="week-8--fst-and-outlier-analysis.html#using-fst-to-find-outliers"><i class="fa fa-check"></i><b>9.5</b> Using F<sub>st</sub> to find outliers<span></span></a></li>
<li class="chapter" data-level="9.6" data-path="week-8--fst-and-outlier-analysis.html"><a href="week-8--fst-and-outlier-analysis.html#outflank-practice"><i class="fa fa-check"></i><b>9.6</b> OutFLANK Practice<span></span></a></li>
<li class="chapter" data-level="9.7" data-path="week-8--fst-and-outlier-analysis.html"><a href="week-8--fst-and-outlier-analysis.html#creature-of-the-week-6"><i class="fa fa-check"></i><b>9.7</b> Creature of the Week!<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html"><i class="fa fa-check"></i><b>10</b> Week 8: Genome wide association study (GWAS)<span></span></a>
<ul>
<li class="chapter" data-level="10.1" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#load-in-the-data"><i class="fa fa-check"></i><b>10.1</b> load in the data<span></span></a></li>
<li class="chapter" data-level="10.2" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#install-packages-in-r"><i class="fa fa-check"></i><b>10.2</b> install packages in R<span></span></a></li>
<li class="chapter" data-level="10.3" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#install-angsd-and-pcangsd-again"><i class="fa fa-check"></i><b>10.3</b> install angsd and pcangsd (again?)<span></span></a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#install-angsd"><i class="fa fa-check"></i><b>10.3.1</b> install angsd<span></span></a></li>
<li class="chapter" data-level="10.3.2" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#for-pcangsd"><i class="fa fa-check"></i><b>10.3.2</b> for pcangsd<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#the-data"><i class="fa fa-check"></i><b>10.4</b> The data<span></span></a></li>
<li class="chapter" data-level="10.5" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#take-beagle-file-and-generate-lrt-file"><i class="fa fa-check"></i><b>10.5</b> take Beagle file and generate lrt file<span></span></a></li>
<li class="chapter" data-level="10.6" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#take-lrt-file-and-make-a-manhattan-plot"><i class="fa fa-check"></i><b>10.6</b> take lrt file and make a manhattan plot<span></span></a></li>
<li class="chapter" data-level="10.7" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#exercises-2"><i class="fa fa-check"></i><b>10.7</b> Exercises<span></span></a></li>
<li class="chapter" data-level="10.8" data-path="week-8-genome-wide-association-study-gwas.html"><a href="week-8-genome-wide-association-study-gwas.html#creature-of-the-week-7"><i class="fa fa-check"></i><b>10.8</b> Creature of the Week!<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="genome-environment-association.html"><a href="genome-environment-association.html"><i class="fa fa-check"></i><b>11</b> Genome Environment Association<span></span></a>
<ul>
<li class="chapter" data-level="11.1" data-path="genome-environment-association.html"><a href="genome-environment-association.html#download-the-data-3"><i class="fa fa-check"></i><b>11.1</b> Download the data<span></span></a></li>
<li class="chapter" data-level="11.2" data-path="genome-environment-association.html"><a href="genome-environment-association.html#install-a-compiler-for-unixbash"><i class="fa fa-check"></i><b>11.2</b> install a compiler for Unix/bash<span></span></a></li>
<li class="chapter" data-level="11.3" data-path="genome-environment-association.html"><a href="genome-environment-association.html#install-r-packages"><i class="fa fa-check"></i><b>11.3</b> install R packages<span></span></a></li>
<li class="chapter" data-level="11.4" data-path="genome-environment-association.html"><a href="genome-environment-association.html#get-the-environmental-data"><i class="fa fa-check"></i><b>11.4</b> Get the Environmental Data<span></span></a></li>
<li class="chapter" data-level="11.5" data-path="genome-environment-association.html"><a href="genome-environment-association.html#extract-site-specific-information-from-our-envirornmental-layers"><i class="fa fa-check"></i><b>11.5</b> Extract Site Specific Information from our Envirornmental Layers<span></span></a></li>
<li class="chapter" data-level="11.6" data-path="genome-environment-association.html"><a href="genome-environment-association.html#make-a-map-of-our-environmental-data"><i class="fa fa-check"></i><b>11.6</b> Make a Map of our environmental data<span></span></a></li>
<li class="chapter" data-level="11.7" data-path="genome-environment-association.html"><a href="genome-environment-association.html#load-in-genetic-data"><i class="fa fa-check"></i><b>11.7</b> Load in genetic data<span></span></a></li>
<li class="chapter" data-level="11.8" data-path="genome-environment-association.html"><a href="genome-environment-association.html#exercises-3"><i class="fa fa-check"></i><b>11.8</b> Exercises<span></span></a></li>
<li class="chapter" data-level="11.9" data-path="genome-environment-association.html"><a href="genome-environment-association.html#download-the-data-4"><i class="fa fa-check"></i><b>11.9</b> Download the data<span></span></a></li>
<li class="chapter" data-level="11.10" data-path="genome-environment-association.html"><a href="genome-environment-association.html#install-transdecoder"><i class="fa fa-check"></i><b>11.10</b> Install Transdecoder<span></span></a></li>
<li class="chapter" data-level="11.11" data-path="genome-environment-association.html"><a href="genome-environment-association.html#run-transdecoder"><i class="fa fa-check"></i><b>11.11</b> Run Transdecoder<span></span></a></li>
<li class="chapter" data-level="11.12" data-path="genome-environment-association.html"><a href="genome-environment-association.html#run-ncbi-blast"><i class="fa fa-check"></i><b>11.12</b> Run NCBI-BLAST<span></span></a></li>
<li class="chapter" data-level="11.13" data-path="genome-environment-association.html"><a href="genome-environment-association.html#what-did-we-find"><i class="fa fa-check"></i><b>11.13</b> What did we find?<span></span></a></li>
<li class="chapter" data-level="11.14" data-path="genome-environment-association.html"><a href="genome-environment-association.html#the-importance-of-genomic-architecture"><i class="fa fa-check"></i><b>11.14</b> The importance of Genomic Architecture<span></span></a></li>
<li class="chapter" data-level="11.15" data-path="genome-environment-association.html"><a href="genome-environment-association.html#exercises-4"><i class="fa fa-check"></i><b>11.15</b> Exercises<span></span></a></li>
<li class="chapter" data-level="11.16" data-path="genome-environment-association.html"><a href="genome-environment-association.html#creature-of-the-week-8"><i class="fa fa-check"></i><b>11.16</b> Creature of the Week!<span></span></a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="population-structure-via-construct" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">7</span> Population Structure via conStruct<a href="population-structure-via-construct.html#population-structure-via-construct" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>For this week we will be exploring another way to document population genetic structure (the first way we covered was PCA in Week 8) via a “Structure” plot implemented in the R package conStruct.</p>
<p>conStruct itself has a nice series of tutorials with example data that you can find <a href="https://cran.r-project.org/web/packages/conStruct/vignettes/format-data.html">here</a></p>
<p>We’ll be using data from the <a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/mec.14589">Xuereb et al. paper on P. californicus</a>. This data consists of a vcf file that is in structure format and contains SNP data (3699 SNPs) from 717 individuals (this is the same data we used for our PCA plot in week 9 for pcadapt, and for week 11 for GEA).</p>
<p>The lecture for this week can be found <a href="https://github.com/BayLab/MarineGenomics/blob/4f22a87ad800f98a6e138a166b8e94d016310914/ppt/Week9.pdf">here</a> and describes the basics of a structure plot.</p>
<div id="download-the-data" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> Download the data<a href="population-structure-via-construct.html#download-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We’ll be using data from the <a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/mec.14589">Xuereb et al. paper on P. californicus</a>. This data consists of a vcf file that is in structure format and contains SNP data (3699 SNPs) from 100 individuals (this is the same data we used for our PCA plot in week 9 for pcadapt, and for week 11 for GEA, just with fewer individuals for this week).</p>
<p>We’ll download a metafile as well that has sample coordinates and names</p>
<div class="sourceCode" id="cb471"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb471-1"><a href="population-structure-via-construct.html#cb471-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb471-2"><a href="population-structure-via-construct.html#cb471-2" aria-hidden="true" tabindex="-1"></a>wget https://raw.githubusercontent.com/BayLab/MarineGenomicsData/main/week12_semester.tar.gz</span>
<span id="cb471-3"><a href="population-structure-via-construct.html#cb471-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb471-4"><a href="population-structure-via-construct.html#cb471-4" aria-hidden="true" tabindex="-1"></a>tar -xzvf week12_semester.tar.gz</span></code></pre></div>
</div>
<div id="update-our-compiler-in-bashunix-and-install-everything-for-r-in-the-terminal-in-bashunix-not-in-rstudio" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> update our compiler in bash/UNIX and install everything for R in the terminal (in bash/UNIX, NOT in Rstudio)<a href="population-structure-via-construct.html#update-our-compiler-in-bashunix-and-install-everything-for-r-in-the-terminal-in-bashunix-not-in-rstudio" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We need to follow this guide <a href="https://fahim-sikder.github.io/post/how-to-install-r-ubuntu-20/">here</a></p>
<div class="sourceCode" id="cb472"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb472-1"><a href="population-structure-via-construct.html#cb472-1" aria-hidden="true" tabindex="-1"></a># first run this to get the compiliers</span>
<span id="cb472-2"><a href="population-structure-via-construct.html#cb472-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb472-3"><a href="population-structure-via-construct.html#cb472-3" aria-hidden="true" tabindex="-1"></a>sudo apt-get install -y libxml2-dev libcurl4-openssl-dev libssl-dev libv8-dev</span>
<span id="cb472-4"><a href="population-structure-via-construct.html#cb472-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb472-5"><a href="population-structure-via-construct.html#cb472-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb472-6"><a href="population-structure-via-construct.html#cb472-6" aria-hidden="true" tabindex="-1"></a>#Then install tidyverse into R from the terminal</span>
<span id="cb472-7"><a href="population-structure-via-construct.html#cb472-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb472-8"><a href="population-structure-via-construct.html#cb472-8" aria-hidden="true" tabindex="-1"></a>sudo R</span>
<span id="cb472-9"><a href="population-structure-via-construct.html#cb472-9" aria-hidden="true" tabindex="-1"></a>install.packages(&#39;tidyverse&#39;)</span>
<span id="cb472-10"><a href="population-structure-via-construct.html#cb472-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb472-11"><a href="population-structure-via-construct.html#cb472-11" aria-hidden="true" tabindex="-1"></a># then exit R with q() and save your workspace image</span>
<span id="cb472-12"><a href="population-structure-via-construct.html#cb472-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb472-13"><a href="population-structure-via-construct.html#cb472-13" aria-hidden="true" tabindex="-1"></a># now go back into R through the terminal and do the following</span>
<span id="cb472-14"><a href="population-structure-via-construct.html#cb472-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb472-15"><a href="population-structure-via-construct.html#cb472-15" aria-hidden="true" tabindex="-1"></a>sudo R</span>
<span id="cb472-16"><a href="population-structure-via-construct.html#cb472-16" aria-hidden="true" tabindex="-1"></a>remove.packages(&quot;rstan&quot;)</span>
<span id="cb472-17"><a href="population-structure-via-construct.html#cb472-17" aria-hidden="true" tabindex="-1"></a>if (file.exists(&quot;.RData&quot;)) file.remove(&quot;.RData&quot;)</span>
<span id="cb472-18"><a href="population-structure-via-construct.html#cb472-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb472-19"><a href="population-structure-via-construct.html#cb472-19" aria-hidden="true" tabindex="-1"></a># now check that we have the compilers we need</span>
<span id="cb472-20"><a href="population-structure-via-construct.html#cb472-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb472-21"><a href="population-structure-via-construct.html#cb472-21" aria-hidden="true" tabindex="-1"></a>pkgbuild::has_build_tools(debug = TRUE)</span>
<span id="cb472-22"><a href="population-structure-via-construct.html#cb472-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb472-23"><a href="population-structure-via-construct.html#cb472-23" aria-hidden="true" tabindex="-1"></a># if you see a &#39;TRUE&#39; then you can go on to the next step, installing rstan</span>
<span id="cb472-24"><a href="population-structure-via-construct.html#cb472-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb472-25"><a href="population-structure-via-construct.html#cb472-25" aria-hidden="true" tabindex="-1"></a>install.packages(&quot;rstan&quot;, repos = &quot;https://cloud.r-project.org/&quot;, dependencies = TRUE)</span></code></pre></div>
</div>
<div id="install-construct-in-r" class="section level2 hasAnchor" number="7.3">
<h2><span class="header-section-number">7.3</span> install conStruct in R<a href="population-structure-via-construct.html#install-construct-in-r" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now that we have rstan installed we should be able to go into Rstudio and install conStruct (you could also stay in the terminal and run install.packages ("conStruct) there if you want.)</p>
</div>
<div id="read-in-the-data" class="section level2 hasAnchor" number="7.4">
<h2><span class="header-section-number">7.4</span> read in the data<a href="population-structure-via-construct.html#read-in-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This first step reads in our file which is in structure format and converts it to conStruct format (there are lots of file formats in population genetics, isn’t it fun?)</p>
<p>We need to make sure that we have the file coding right and this can be confirmed by looking at the structure file in bash to make sure that our samples start at the 3rd row and 3rd column and that our missing data is encoded as a -9.</p>
<p>You can do the above by using wc -l to find the number of lines in the structure file.</p>
<div class="sourceCode" id="cb473"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb473-1"><a href="population-structure-via-construct.html#cb473-1" aria-hidden="true" tabindex="-1"></a>conStruct.data <span class="ot">&lt;-</span> <span class="fu">structure2conStruct</span>(<span class="at">infile=</span><span class="st">&quot;filtered_3699snps_100indcalifornicus.recode.strct_in&quot;</span>,</span>
<span id="cb473-2"><a href="population-structure-via-construct.html#cb473-2" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">onerowperind =</span> <span class="cn">TRUE</span>,</span>
<span id="cb473-3"><a href="population-structure-via-construct.html#cb473-3" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">start.loci =</span> <span class="dv">3</span>,</span>
<span id="cb473-4"><a href="population-structure-via-construct.html#cb473-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">start.samples =</span> <span class="dv">3</span>,</span>
<span id="cb473-5"><a href="population-structure-via-construct.html#cb473-5" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">missing.datum =</span> <span class="sc">-</span><span class="dv">9</span>,</span>
<span id="cb473-6"><a href="population-structure-via-construct.html#cb473-6" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">outfile =</span> <span class="st">&quot;Construct_100inds&quot;</span>)</span></code></pre></div>
</div>
<div id="get-our-meta-data-read-into-r-and-ready-to-use" class="section level2 hasAnchor" number="7.5">
<h2><span class="header-section-number">7.5</span> Get our meta data read into R and ready to use<a href="population-structure-via-construct.html#get-our-meta-data-read-into-r-and-ready-to-use" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This file contains the lat and lon of our samples as well as the site name and individual id.</p>
<p>For conStruct we can make an object that contains the lat and lon and has the site name as the row names.</p>
<div class="sourceCode" id="cb474"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb474-1"><a href="population-structure-via-construct.html#cb474-1" aria-hidden="true" tabindex="-1"></a><span class="co">#read in the data</span></span>
<span id="cb474-2"><a href="population-structure-via-construct.html#cb474-2" aria-hidden="true" tabindex="-1"></a>meta<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;californicus_metadata.csv&quot;</span>, <span class="at">header=</span>T, <span class="at">sep=</span><span class="st">&quot;,&quot;</span>)</span>
<span id="cb474-3"><a href="population-structure-via-construct.html#cb474-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb474-4"><a href="population-structure-via-construct.html#cb474-4" aria-hidden="true" tabindex="-1"></a><span class="co">#pull out the lat and lon and put it in a new matrix object</span></span>
<span id="cb474-5"><a href="population-structure-via-construct.html#cb474-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb474-6"><a href="population-structure-via-construct.html#cb474-6" aria-hidden="true" tabindex="-1"></a>sites<span class="ot">&lt;-</span><span class="fu">as.matrix</span>(meta[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">51</span><span class="sc">:</span><span class="dv">667</span>), <span class="fu">c</span>(<span class="st">&quot;LAT&quot;</span>, <span class="st">&quot;LONG&quot;</span>)])</span>
<span id="cb474-7"><a href="population-structure-via-construct.html#cb474-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb474-8"><a href="population-structure-via-construct.html#cb474-8" aria-hidden="true" tabindex="-1"></a><span class="co">#assign row names that are the site id for each sample</span></span>
<span id="cb474-9"><a href="population-structure-via-construct.html#cb474-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb474-10"><a href="population-structure-via-construct.html#cb474-10" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sites)<span class="ot">&lt;-</span>meta<span class="sc">$</span>SITE[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">51</span><span class="sc">:</span><span class="dv">667</span>)]</span></code></pre></div>
</div>
<div id="run-a-structure-analysis" class="section level2 hasAnchor" number="7.6">
<h2><span class="header-section-number">7.6</span> Run a structure analysis<a href="population-structure-via-construct.html#run-a-structure-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We now have our data read into conStruct and converted to conStruct format and our site information from the metadata in a matrix format.</p>
<p>We can now run a structure analyses. For these analyses we chose a value of K (the number of clusters) and typically run the analyses for a range of K (2:7 for example).</p>
<p>We’ll run K = 2 through K = 4 and then we’ll do some model comparison to discuss whether one of these is “better” than the other.</p>
<p>And we’ll only run it with 1 chains and 10 iterations. This is simply so that the program runs quickly within our class time. For a “real” analysis you’d want to run more than one chain (2-5) and iterations up to 1000. It’s important to compare the different chains to ensure the analyses converges, as described <a href="https://cran.r-project.org/web/packages/conStruct/vignettes/run-conStruct.html#model-diagnosis">here</a></p>
<div class="sourceCode" id="cb475"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb475-1"><a href="population-structure-via-construct.html#cb475-1" aria-hidden="true" tabindex="-1"></a>my.run <span class="ot">&lt;-</span> <span class="fu">conStruct</span>(<span class="at">spatial =</span> <span class="cn">FALSE</span>,</span>
<span id="cb475-2"><a href="population-structure-via-construct.html#cb475-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">K =</span> <span class="dv">2</span>,</span>
<span id="cb475-3"><a href="population-structure-via-construct.html#cb475-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">freqs =</span> conStruct.data,</span>
<span id="cb475-4"><a href="population-structure-via-construct.html#cb475-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">geoDist =</span> <span class="cn">NULL</span>,</span>
<span id="cb475-5"><a href="population-structure-via-construct.html#cb475-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">coords =</span> sites,</span>
<span id="cb475-6"><a href="population-structure-via-construct.html#cb475-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">prefix =</span> <span class="st">&quot;Sea_cucu_K2&quot;</span>,</span>
<span id="cb475-7"><a href="population-structure-via-construct.html#cb475-7" aria-hidden="true" tabindex="-1"></a>                    <span class="at">n.chains =</span> <span class="dv">1</span>,</span>
<span id="cb475-8"><a href="population-structure-via-construct.html#cb475-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">n.iter =</span> <span class="dv">10</span>, </span>
<span id="cb475-9"><a href="population-structure-via-construct.html#cb475-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">make.figs=</span>F)</span>
<span id="cb475-10"><a href="population-structure-via-construct.html#cb475-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb475-11"><a href="population-structure-via-construct.html#cb475-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb475-12"><a href="population-structure-via-construct.html#cb475-12" aria-hidden="true" tabindex="-1"></a>my.run <span class="ot">&lt;-</span> <span class="fu">conStruct</span>(<span class="at">spatial =</span> <span class="cn">FALSE</span>,</span>
<span id="cb475-13"><a href="population-structure-via-construct.html#cb475-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">K =</span> <span class="dv">3</span>,</span>
<span id="cb475-14"><a href="population-structure-via-construct.html#cb475-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">freqs =</span> conStruct.data,</span>
<span id="cb475-15"><a href="population-structure-via-construct.html#cb475-15" aria-hidden="true" tabindex="-1"></a>                    <span class="at">geoDist =</span> <span class="cn">NULL</span>,</span>
<span id="cb475-16"><a href="population-structure-via-construct.html#cb475-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">coords =</span> sites,</span>
<span id="cb475-17"><a href="population-structure-via-construct.html#cb475-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">prefix =</span> <span class="st">&quot;Sea_cucu_K3&quot;</span>,</span>
<span id="cb475-18"><a href="population-structure-via-construct.html#cb475-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">n.chains =</span> <span class="dv">1</span>,</span>
<span id="cb475-19"><a href="population-structure-via-construct.html#cb475-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">n.iter =</span> <span class="dv">10</span>, </span>
<span id="cb475-20"><a href="population-structure-via-construct.html#cb475-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">make.figs=</span>F)</span>
<span id="cb475-21"><a href="population-structure-via-construct.html#cb475-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb475-22"><a href="population-structure-via-construct.html#cb475-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb475-23"><a href="population-structure-via-construct.html#cb475-23" aria-hidden="true" tabindex="-1"></a>my.run <span class="ot">&lt;-</span> <span class="fu">conStruct</span>(<span class="at">spatial =</span> <span class="cn">FALSE</span>,</span>
<span id="cb475-24"><a href="population-structure-via-construct.html#cb475-24" aria-hidden="true" tabindex="-1"></a>                    <span class="at">K =</span> <span class="dv">4</span>,</span>
<span id="cb475-25"><a href="population-structure-via-construct.html#cb475-25" aria-hidden="true" tabindex="-1"></a>                    <span class="at">freqs =</span> conStruct.data,</span>
<span id="cb475-26"><a href="population-structure-via-construct.html#cb475-26" aria-hidden="true" tabindex="-1"></a>                    <span class="at">geoDist =</span> <span class="cn">NULL</span>,</span>
<span id="cb475-27"><a href="population-structure-via-construct.html#cb475-27" aria-hidden="true" tabindex="-1"></a>                    <span class="at">coords =</span> sites,</span>
<span id="cb475-28"><a href="population-structure-via-construct.html#cb475-28" aria-hidden="true" tabindex="-1"></a>                    <span class="at">prefix =</span> <span class="st">&quot;Sea_cucu_K4&quot;</span>,</span>
<span id="cb475-29"><a href="population-structure-via-construct.html#cb475-29" aria-hidden="true" tabindex="-1"></a>                    <span class="at">n.chains =</span> <span class="dv">1</span>,</span>
<span id="cb475-30"><a href="population-structure-via-construct.html#cb475-30" aria-hidden="true" tabindex="-1"></a>                    <span class="at">n.iter =</span> <span class="dv">10</span>, </span>
<span id="cb475-31"><a href="population-structure-via-construct.html#cb475-31" aria-hidden="true" tabindex="-1"></a>                    <span class="at">make.figs=</span>F)</span>
<span id="cb475-32"><a href="population-structure-via-construct.html#cb475-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb475-33"><a href="population-structure-via-construct.html#cb475-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb475-34"><a href="population-structure-via-construct.html#cb475-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb475-35"><a href="population-structure-via-construct.html#cb475-35" aria-hidden="true" tabindex="-1"></a>my.run <span class="ot">&lt;-</span> <span class="fu">conStruct</span>(<span class="at">spatial =</span> <span class="cn">FALSE</span>,</span>
<span id="cb475-36"><a href="population-structure-via-construct.html#cb475-36" aria-hidden="true" tabindex="-1"></a>                    <span class="at">K =</span> <span class="dv">5</span>,</span>
<span id="cb475-37"><a href="population-structure-via-construct.html#cb475-37" aria-hidden="true" tabindex="-1"></a>                    <span class="at">freqs =</span> conStruct.data,</span>
<span id="cb475-38"><a href="population-structure-via-construct.html#cb475-38" aria-hidden="true" tabindex="-1"></a>                    <span class="at">geoDist =</span> <span class="cn">NULL</span>,</span>
<span id="cb475-39"><a href="population-structure-via-construct.html#cb475-39" aria-hidden="true" tabindex="-1"></a>                    <span class="at">coords =</span> sites,</span>
<span id="cb475-40"><a href="population-structure-via-construct.html#cb475-40" aria-hidden="true" tabindex="-1"></a>                    <span class="at">prefix =</span> <span class="st">&quot;Sea_cucu_K5&quot;</span>,</span>
<span id="cb475-41"><a href="population-structure-via-construct.html#cb475-41" aria-hidden="true" tabindex="-1"></a>                    <span class="at">n.chains =</span> <span class="dv">1</span>,</span>
<span id="cb475-42"><a href="population-structure-via-construct.html#cb475-42" aria-hidden="true" tabindex="-1"></a>                    <span class="at">n.iter =</span> <span class="dv">10</span>, </span>
<span id="cb475-43"><a href="population-structure-via-construct.html#cb475-43" aria-hidden="true" tabindex="-1"></a>                    <span class="at">make.figs=</span>F)</span></code></pre></div>
<p>This step produces several output files (a conStruct.results file and a data.block file) called after whatever you put in the prefix = section (sea_cucu_K3 in our case). We need to load that output file back into R to access all the things it produced.</p>
<p>Let’s just look at the first one for k = 3.</p>
<div class="sourceCode" id="cb476"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb476-1"><a href="population-structure-via-construct.html#cb476-1" aria-hidden="true" tabindex="-1"></a><span class="co">#load the construct results file</span></span>
<span id="cb476-2"><a href="population-structure-via-construct.html#cb476-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;Sea_cucu_K3_conStruct.results.Robj&quot;</span>)</span>
<span id="cb476-3"><a href="population-structure-via-construct.html#cb476-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb476-4"><a href="population-structure-via-construct.html#cb476-4" aria-hidden="true" tabindex="-1"></a><span class="co">#pull out the admix proportions from this results file</span></span>
<span id="cb476-5"><a href="population-structure-via-construct.html#cb476-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb476-6"><a href="population-structure-via-construct.html#cb476-6" aria-hidden="true" tabindex="-1"></a>admix.props<span class="ot">&lt;-</span> conStruct.results<span class="sc">$</span>chain_1<span class="sc">$</span>MAP<span class="sc">$</span>admix.proportions</span>
<span id="cb476-7"><a href="population-structure-via-construct.html#cb476-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb476-8"><a href="population-structure-via-construct.html#cb476-8" aria-hidden="true" tabindex="-1"></a><span class="co">#load the data block file</span></span>
<span id="cb476-9"><a href="population-structure-via-construct.html#cb476-9" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;Sea_cucu_K3_data.block.Robj&quot;</span>)</span>
<span id="cb476-10"><a href="population-structure-via-construct.html#cb476-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb476-11"><a href="population-structure-via-construct.html#cb476-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb476-12"><a href="population-structure-via-construct.html#cb476-12" aria-hidden="true" tabindex="-1"></a><span class="co"># make a structure plot</span></span>
<span id="cb476-13"><a href="population-structure-via-construct.html#cb476-13" aria-hidden="true" tabindex="-1"></a><span class="fu">make.structure.plot</span>(<span class="at">admix.proportions =</span> admix.props,</span>
<span id="cb476-14"><a href="population-structure-via-construct.html#cb476-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">sample.names=</span><span class="fu">row.names</span>(sites),</span>
<span id="cb476-15"><a href="population-structure-via-construct.html#cb476-15" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">4</span>,<span class="dv">2</span>,<span class="dv">2</span>))</span></code></pre></div>
<p><img src="06_Week6.structure_files/figure-html/unnamed-chunk-5-1.png" width="768" /></p>
<div class="sourceCode" id="cb477"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb477-1"><a href="population-structure-via-construct.html#cb477-1" aria-hidden="true" tabindex="-1"></a><span class="co">#let&#39;s also make a pie chart</span></span>
<span id="cb477-2"><a href="population-structure-via-construct.html#cb477-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb477-3"><a href="population-structure-via-construct.html#cb477-3" aria-hidden="true" tabindex="-1"></a><span class="fu">make.admix.pie.plot</span>(<span class="at">admix.proportions =</span> admix.props,</span>
<span id="cb477-4"><a href="population-structure-via-construct.html#cb477-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">coords =</span> data.block<span class="sc">$</span>coords,</span>
<span id="cb477-5"><a href="population-structure-via-construct.html#cb477-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">4</span>,<span class="dv">2</span>,<span class="dv">2</span>))</span></code></pre></div>
<p><img src="06_Week6.structure_files/figure-html/unnamed-chunk-5-2.png" width="768" /></p>
</div>
<div id="incoorporate-geographic-distances-in-a-spatial-model" class="section level2 hasAnchor" number="7.7">
<h2><span class="header-section-number">7.7</span> Incoorporate geographic distances in a spatial model<a href="population-structure-via-construct.html#incoorporate-geographic-distances-in-a-spatial-model" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>One of the main advantages of using conStruct is the ability to incorporate within the model measures of geographic distance.</p>
<p>Let’s do that for our K = 3 from above.</p>
<p>First we need to calculate the geographic distance between our samples. There are many ways to do this, but we’ll take a simple approach of calculating the euclidean (straight-line) distance between each site.</p>
<div class="sourceCode" id="cb478"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb478-1"><a href="population-structure-via-construct.html#cb478-1" aria-hidden="true" tabindex="-1"></a><span class="co">#first install and then load the package geosphere</span></span>
<span id="cb478-2"><a href="population-structure-via-construct.html#cb478-2" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages(&quot;geodist&quot;)</span></span>
<span id="cb478-3"><a href="population-structure-via-construct.html#cb478-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb478-4"><a href="population-structure-via-construct.html#cb478-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;geodist&quot;</span>)</span>
<span id="cb478-5"><a href="population-structure-via-construct.html#cb478-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb478-6"><a href="population-structure-via-construct.html#cb478-6" aria-hidden="true" tabindex="-1"></a><span class="co">#we&#39;ll use the function distm and the method distHaversine, which is the shortest distance between two points</span></span>
<span id="cb478-7"><a href="population-structure-via-construct.html#cb478-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb478-8"><a href="population-structure-via-construct.html#cb478-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb478-9"><a href="population-structure-via-construct.html#cb478-9" aria-hidden="true" tabindex="-1"></a>geo_dist<span class="ot">&lt;-</span><span class="fu">geodist</span>(sites, <span class="at">measure=</span><span class="st">&quot;haversine&quot;</span>)</span>
<span id="cb478-10"><a href="population-structure-via-construct.html#cb478-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb478-11"><a href="population-structure-via-construct.html#cb478-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb478-12"><a href="population-structure-via-construct.html#cb478-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb478-13"><a href="population-structure-via-construct.html#cb478-13" aria-hidden="true" tabindex="-1"></a><span class="co">#now we can put the distance into our conStruct analyses</span></span>
<span id="cb478-14"><a href="population-structure-via-construct.html#cb478-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb478-15"><a href="population-structure-via-construct.html#cb478-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb478-16"><a href="population-structure-via-construct.html#cb478-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb478-17"><a href="population-structure-via-construct.html#cb478-17" aria-hidden="true" tabindex="-1"></a>my.run <span class="ot">&lt;-</span> <span class="fu">conStruct</span>(<span class="at">spatial =</span> <span class="cn">TRUE</span>,</span>
<span id="cb478-18"><a href="population-structure-via-construct.html#cb478-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">K =</span> <span class="dv">3</span>,</span>
<span id="cb478-19"><a href="population-structure-via-construct.html#cb478-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">freqs =</span> conStruct.data,</span>
<span id="cb478-20"><a href="population-structure-via-construct.html#cb478-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">geoDist =</span> geo_dist,</span>
<span id="cb478-21"><a href="population-structure-via-construct.html#cb478-21" aria-hidden="true" tabindex="-1"></a>                    <span class="at">coords =</span> sites,</span>
<span id="cb478-22"><a href="population-structure-via-construct.html#cb478-22" aria-hidden="true" tabindex="-1"></a>                    <span class="at">prefix =</span> <span class="st">&quot;Sea_cucu_K3_dist&quot;</span>,</span>
<span id="cb478-23"><a href="population-structure-via-construct.html#cb478-23" aria-hidden="true" tabindex="-1"></a>                    <span class="at">n.chains =</span> <span class="dv">1</span>,</span>
<span id="cb478-24"><a href="population-structure-via-construct.html#cb478-24" aria-hidden="true" tabindex="-1"></a>                    <span class="at">n.iter =</span> <span class="dv">10</span>, </span>
<span id="cb478-25"><a href="population-structure-via-construct.html#cb478-25" aria-hidden="true" tabindex="-1"></a>                    <span class="at">make.figs=</span>T)</span></code></pre></div>
<pre><code>## 
## SAMPLING FOR MODEL &#39;space_multiK&#39; NOW (CHAIN 1).
## Chain 1: 
## Chain 1: Gradient evaluation took 0.008307 seconds
## Chain 1: 1000 transitions using 10 leapfrog steps per transition would take 83.07 seconds.
## Chain 1: Adjust your expectations accordingly!
## Chain 1: 
## Chain 1: 
## Chain 1: WARNING: No variance estimation is
## Chain 1:          performed for num_warmup &lt; 20
## Chain 1: 
## Chain 1: Iteration: 1 / 10 [ 10%]  (Warmup)
## Chain 1: Iteration: 2 / 10 [ 20%]  (Warmup)
## Chain 1: Iteration: 3 / 10 [ 30%]  (Warmup)
## Chain 1: Iteration: 4 / 10 [ 40%]  (Warmup)
## Chain 1: Iteration: 5 / 10 [ 50%]  (Warmup)
## Chain 1: Iteration: 6 / 10 [ 60%]  (Sampling)
## Chain 1: Iteration: 7 / 10 [ 70%]  (Sampling)
## Chain 1: Iteration: 8 / 10 [ 80%]  (Sampling)
## Chain 1: Iteration: 9 / 10 [ 90%]  (Sampling)
## Chain 1: Iteration: 10 / 10 [100%]  (Sampling)
## Chain 1: 
## Chain 1:  Elapsed Time: 0.089511 seconds (Warm-up)
## Chain 1:                0.085649 seconds (Sampling)
## Chain 1:                0.17516 seconds (Total)
## Chain 1:</code></pre>
<div class="sourceCode" id="cb480"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb480-1"><a href="population-structure-via-construct.html#cb480-1" aria-hidden="true" tabindex="-1"></a><span class="co">#remake our structure plots</span></span>
<span id="cb480-2"><a href="population-structure-via-construct.html#cb480-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb480-3"><a href="population-structure-via-construct.html#cb480-3" aria-hidden="true" tabindex="-1"></a><span class="co">#load the construct results file</span></span>
<span id="cb480-4"><a href="population-structure-via-construct.html#cb480-4" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;Sea_cucu_K3_dist_conStruct.results.Robj&quot;</span>)</span>
<span id="cb480-5"><a href="population-structure-via-construct.html#cb480-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb480-6"><a href="population-structure-via-construct.html#cb480-6" aria-hidden="true" tabindex="-1"></a>admix.props<span class="ot">&lt;-</span> conStruct.results<span class="sc">$</span>chain_1<span class="sc">$</span>MAP<span class="sc">$</span>admix.proportions</span>
<span id="cb480-7"><a href="population-structure-via-construct.html#cb480-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb480-8"><a href="population-structure-via-construct.html#cb480-8" aria-hidden="true" tabindex="-1"></a><span class="co">#load the data block file</span></span>
<span id="cb480-9"><a href="population-structure-via-construct.html#cb480-9" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;Sea_cucu_K3_dist_data.block.Robj&quot;</span>)</span>
<span id="cb480-10"><a href="population-structure-via-construct.html#cb480-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb480-11"><a href="population-structure-via-construct.html#cb480-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb480-12"><a href="population-structure-via-construct.html#cb480-12" aria-hidden="true" tabindex="-1"></a><span class="co"># make a structure plot</span></span>
<span id="cb480-13"><a href="population-structure-via-construct.html#cb480-13" aria-hidden="true" tabindex="-1"></a><span class="fu">make.structure.plot</span>(<span class="at">admix.proportions =</span> admix.props,</span>
<span id="cb480-14"><a href="population-structure-via-construct.html#cb480-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">sample.names=</span><span class="fu">row.names</span>(sites),</span>
<span id="cb480-15"><a href="population-structure-via-construct.html#cb480-15" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">4</span>,<span class="dv">2</span>,<span class="dv">2</span>))</span></code></pre></div>
<p><img src="06_Week6.structure_files/figure-html/unnamed-chunk-6-1.png" width="768" /></p>
<div class="sourceCode" id="cb481"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb481-1"><a href="population-structure-via-construct.html#cb481-1" aria-hidden="true" tabindex="-1"></a><span class="co">#and a pie chart</span></span>
<span id="cb481-2"><a href="population-structure-via-construct.html#cb481-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb481-3"><a href="population-structure-via-construct.html#cb481-3" aria-hidden="true" tabindex="-1"></a><span class="fu">make.admix.pie.plot</span>(<span class="at">admix.proportions =</span> admix.props,</span>
<span id="cb481-4"><a href="population-structure-via-construct.html#cb481-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">coords =</span> data.block<span class="sc">$</span>coords,</span>
<span id="cb481-5"><a href="population-structure-via-construct.html#cb481-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">4</span>,<span class="dv">2</span>,<span class="dv">2</span>))</span></code></pre></div>
<p><img src="06_Week6.structure_files/figure-html/unnamed-chunk-6-2.png" width="768" /></p>
</div>
<div id="comparisons-across-different-values-of-k" class="section level2 hasAnchor" number="7.8">
<h2><span class="header-section-number">7.8</span> Comparisons across different values of K<a href="population-structure-via-construct.html#comparisons-across-different-values-of-k" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>You may want to show a single structure plot for one value of K and may want that one plot to be the “best” fitting model for your data.</p>
<p>There are a few ways to do this, however most often there’s not a single “best” model and care should be taken as models with more parameters often appear to be a better fit, but are not necessarily more biologically meaningful.</p>
<p>The vignette here has several ways in which you can choose an “optimal” K or several K <a href="https://cran.r-project.org/web/packages/conStruct/vignettes/model-comparison.html">here</a></p>
<p>Note that the vignette on layer contributions only works with a spatial conStruct analyses.</p>
</div>
<div id="exercises" class="section level2 hasAnchor" number="7.9">
<h2><span class="header-section-number">7.9</span> Exercises<a href="population-structure-via-construct.html#exercises" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ol style="list-style-type: decimal">
<li>change the colors in the plot. THe default red and blue for K = 2 is not very friendly for those with color blindness. Using the command <code>?make.structure.plot</code> determine how you would change the colors and then using this <a href="http://www.cookbook-r.com/Graphs/Colors_(ggplot2)/#a-colorblind-friendly-palette">site</a> choose color-blind-friendly colors. Make a strucutre plot for K = 2 and K = 4 using your new colors.</li>
</ol>
<details>
<summary>
<span style="color: Purple;">Solution</span>
</summary>
<p>
<div class="sourceCode" id="cb482"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb482-1"><a href="population-structure-via-construct.html#cb482-1" aria-hidden="true" tabindex="-1"></a><span class="co">#first check how to do this</span></span>
<span id="cb482-2"><a href="population-structure-via-construct.html#cb482-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb482-3"><a href="population-structure-via-construct.html#cb482-3" aria-hidden="true" tabindex="-1"></a>?<span class="fu">make.structure.plot</span>()</span>
<span id="cb482-4"><a href="population-structure-via-construct.html#cb482-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb482-5"><a href="population-structure-via-construct.html#cb482-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb482-6"><a href="population-structure-via-construct.html#cb482-6" aria-hidden="true" tabindex="-1"></a><span class="do">###### For K = 2   ############</span></span>
<span id="cb482-7"><a href="population-structure-via-construct.html#cb482-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb482-8"><a href="population-structure-via-construct.html#cb482-8" aria-hidden="true" tabindex="-1"></a><span class="co">#load the construct results file </span></span>
<span id="cb482-9"><a href="population-structure-via-construct.html#cb482-9" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;Sea_cucu_K2_conStruct.results.Robj&quot;</span>)</span>
<span id="cb482-10"><a href="population-structure-via-construct.html#cb482-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb482-11"><a href="population-structure-via-construct.html#cb482-11" aria-hidden="true" tabindex="-1"></a>admix.props<span class="ot">&lt;-</span> conStruct.results<span class="sc">$</span>chain_1<span class="sc">$</span>MAP<span class="sc">$</span>admix.proportions</span>
<span id="cb482-12"><a href="population-structure-via-construct.html#cb482-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb482-13"><a href="population-structure-via-construct.html#cb482-13" aria-hidden="true" tabindex="-1"></a><span class="co">#load the data block file</span></span>
<span id="cb482-14"><a href="population-structure-via-construct.html#cb482-14" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;Sea_cucu_K2_data.block.Robj&quot;</span>)</span>
<span id="cb482-15"><a href="population-structure-via-construct.html#cb482-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb482-16"><a href="population-structure-via-construct.html#cb482-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb482-17"><a href="population-structure-via-construct.html#cb482-17" aria-hidden="true" tabindex="-1"></a><span class="co"># make a structure plot</span></span>
<span id="cb482-18"><a href="population-structure-via-construct.html#cb482-18" aria-hidden="true" tabindex="-1"></a><span class="fu">make.structure.plot</span>(<span class="at">admix.proportions =</span> admix.props,</span>
<span id="cb482-19"><a href="population-structure-via-construct.html#cb482-19" aria-hidden="true" tabindex="-1"></a>                    <span class="at">layer.colors =</span> <span class="fu">c</span>(<span class="st">&quot;#E69F00&quot;</span>, <span class="st">&quot;#0072B2&quot;</span>),</span>
<span id="cb482-20"><a href="population-structure-via-construct.html#cb482-20" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">4</span>,<span class="dv">2</span>,<span class="dv">2</span>))</span></code></pre></div>
<p><img src="06_Week6.structure_files/figure-html/e1_w12-1.png" width="768" /></p>
<div class="sourceCode" id="cb483"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb483-1"><a href="population-structure-via-construct.html#cb483-1" aria-hidden="true" tabindex="-1"></a><span class="do">###### For K = 4   ############</span></span>
<span id="cb483-2"><a href="population-structure-via-construct.html#cb483-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb483-3"><a href="population-structure-via-construct.html#cb483-3" aria-hidden="true" tabindex="-1"></a><span class="co">#load the construct results file </span></span>
<span id="cb483-4"><a href="population-structure-via-construct.html#cb483-4" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;Sea_cucu_K4_conStruct.results.Robj&quot;</span>)</span>
<span id="cb483-5"><a href="population-structure-via-construct.html#cb483-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb483-6"><a href="population-structure-via-construct.html#cb483-6" aria-hidden="true" tabindex="-1"></a>admix.props<span class="ot">&lt;-</span> conStruct.results<span class="sc">$</span>chain_1<span class="sc">$</span>MAP<span class="sc">$</span>admix.proportions</span>
<span id="cb483-7"><a href="population-structure-via-construct.html#cb483-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb483-8"><a href="population-structure-via-construct.html#cb483-8" aria-hidden="true" tabindex="-1"></a><span class="co">#load the data block file</span></span>
<span id="cb483-9"><a href="population-structure-via-construct.html#cb483-9" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;Sea_cucu_K4_data.block.Robj&quot;</span>)</span>
<span id="cb483-10"><a href="population-structure-via-construct.html#cb483-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb483-11"><a href="population-structure-via-construct.html#cb483-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb483-12"><a href="population-structure-via-construct.html#cb483-12" aria-hidden="true" tabindex="-1"></a><span class="co"># make a structure plot</span></span>
<span id="cb483-13"><a href="population-structure-via-construct.html#cb483-13" aria-hidden="true" tabindex="-1"></a><span class="fu">make.structure.plot</span>(<span class="at">admix.proportions =</span> admix.props,</span>
<span id="cb483-14"><a href="population-structure-via-construct.html#cb483-14" aria-hidden="true" tabindex="-1"></a>                    <span class="at">layer.colors =</span> <span class="fu">c</span>(<span class="st">&quot;#E69F00&quot;</span>, <span class="st">&quot;#0072B2&quot;</span>, <span class="st">&quot;#999999&quot;</span>, <span class="st">&quot;#CC79A7&quot;</span>),</span>
<span id="cb483-15"><a href="population-structure-via-construct.html#cb483-15" aria-hidden="true" tabindex="-1"></a>                    <span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">4</span>,<span class="dv">2</span>,<span class="dv">2</span>))</span></code></pre></div>
<p><img src="06_Week6.structure_files/figure-html/e1_w12-2.png" width="768" /></p>
</p>
</details>
<ol start="2" style="list-style-type: decimal">
<li>Make a pie-chart that overlays on a map. You will find a guide for doing this <a href="https://cran.r-project.org/web/packages/conStruct/vignettes/visualize-results.html">here</a></li>
</ol>
<details>
<summary>
<span style="color: Purple;">Solution</span>
</summary>
<p>
<div class="sourceCode" id="cb484"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb484-1"><a href="population-structure-via-construct.html#cb484-1" aria-hidden="true" tabindex="-1"></a><span class="co">#first load your data</span></span>
<span id="cb484-2"><a href="population-structure-via-construct.html#cb484-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;Sea_cucu_K2_conStruct.results.Robj&quot;</span>)</span>
<span id="cb484-3"><a href="population-structure-via-construct.html#cb484-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb484-4"><a href="population-structure-via-construct.html#cb484-4" aria-hidden="true" tabindex="-1"></a>admix.props<span class="ot">&lt;-</span> conStruct.results<span class="sc">$</span>chain_1<span class="sc">$</span>MAP<span class="sc">$</span>admix.proportions</span>
<span id="cb484-5"><a href="population-structure-via-construct.html#cb484-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb484-6"><a href="population-structure-via-construct.html#cb484-6" aria-hidden="true" tabindex="-1"></a><span class="co">#load the data block file</span></span>
<span id="cb484-7"><a href="population-structure-via-construct.html#cb484-7" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;Sea_cucu_K2_data.block.Robj&quot;</span>)</span>
<span id="cb484-8"><a href="population-structure-via-construct.html#cb484-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb484-9"><a href="population-structure-via-construct.html#cb484-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb484-10"><a href="population-structure-via-construct.html#cb484-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb484-11"><a href="population-structure-via-construct.html#cb484-11" aria-hidden="true" tabindex="-1"></a><span class="co"># make the desired map</span></span>
<span id="cb484-12"><a href="population-structure-via-construct.html#cb484-12" aria-hidden="true" tabindex="-1"></a>maps<span class="sc">::</span><span class="fu">map</span>(<span class="at">xlim =</span> <span class="fu">range</span>(data.block<span class="sc">$</span>coords[,<span class="dv">1</span>]) <span class="sc">+</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>,<span class="dv">5</span>), <span class="at">ylim =</span> <span class="fu">range</span>(data.block<span class="sc">$</span>coords[,<span class="dv">2</span>])<span class="sc">+</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>,<span class="dv">2</span>), <span class="at">col=</span><span class="st">&quot;gray&quot;</span>)</span>
<span id="cb484-13"><a href="population-structure-via-construct.html#cb484-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb484-14"><a href="population-structure-via-construct.html#cb484-14" aria-hidden="true" tabindex="-1"></a><span class="co"># add the admixture pie plot</span></span>
<span id="cb484-15"><a href="population-structure-via-construct.html#cb484-15" aria-hidden="true" tabindex="-1"></a><span class="fu">make.admix.pie.plot</span>(<span class="at">admix.proportions =</span> admix.props,</span>
<span id="cb484-16"><a href="population-structure-via-construct.html#cb484-16" aria-hidden="true" tabindex="-1"></a>                        <span class="at">coords =</span> data.block<span class="sc">$</span>coords,</span>
<span id="cb484-17"><a href="population-structure-via-construct.html#cb484-17" aria-hidden="true" tabindex="-1"></a>                        <span class="at">add =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="06_Week6.structure_files/figure-html/e2_w12-1.png" width="768" /></p>
</p>
</details>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="week-5--r-continued.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="week-6--principle-component-analyses.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["MarGen2022.pdf", "MarGen2022.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
