<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3 Week 2-Working With Files | MarGen2022</title>
  <meta name="description" content="" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="3 Week 2-Working With Files | MarGen2022" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3 Week 2-Working With Files | MarGen2022" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="week-1--welcome.html"/>

<script src="libs/header-attrs-2.11.16/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Marine Genomics 2022</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction to Marine Genomics<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#cloud-computing-versus-individual-laptops"><i class="fa fa-check"></i><b>1.1</b> Cloud computing versus individual laptops<span></span></a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#meet-our-instructors"><i class="fa fa-check"></i><b>1.2</b> Meet our instructors!<span></span></a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#schedule"><i class="fa fa-check"></i><b>1.3</b> Schedule<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="week-1--welcome.html"><a href="week-1--welcome.html"><i class="fa fa-check"></i><b>2</b> Week 1- Welcome!<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="week-1--welcome.html"><a href="week-1--welcome.html#introduction-to-shell-computing-via-the-data-carpentry-tutorial"><i class="fa fa-check"></i><b>2.1</b> Introduction to shell computing via the data carpentry tutorial<span></span></a></li>
<li class="chapter" data-level="2.2" data-path="week-1--welcome.html"><a href="week-1--welcome.html#how-to-access-the-shell"><i class="fa fa-check"></i><b>2.2</b> How to access the shell<span></span></a></li>
<li class="chapter" data-level="2.3" data-path="week-1--welcome.html"><a href="week-1--welcome.html#best-practices-for-jetstream"><i class="fa fa-check"></i><b>2.3</b> Best practices for Jetstream<span></span></a></li>
<li class="chapter" data-level="2.4" data-path="week-1--welcome.html"><a href="week-1--welcome.html#week-1-objectives"><i class="fa fa-check"></i><b>2.4</b> Week 1 Objectives<span></span></a></li>
<li class="chapter" data-level="2.5" data-path="week-1--welcome.html"><a href="week-1--welcome.html#navigating-your-file-system"><i class="fa fa-check"></i><b>2.5</b> Navigating your file system<span></span></a></li>
<li class="chapter" data-level="2.6" data-path="week-1--welcome.html"><a href="week-1--welcome.html#shortcut-tab-completion"><i class="fa fa-check"></i><b>2.6</b> Shortcut: Tab Completion<span></span></a></li>
<li class="chapter" data-level="2.7" data-path="week-1--welcome.html"><a href="week-1--welcome.html#summary-key-points"><i class="fa fa-check"></i><b>2.7</b> Summary &amp; Key Points<span></span></a></li>
<li class="chapter" data-level="2.8" data-path="week-1--welcome.html"><a href="week-1--welcome.html#navigating-files-and-directories"><i class="fa fa-check"></i><b>2.8</b> Navigating Files and Directories<span></span></a></li>
<li class="chapter" data-level="2.9" data-path="week-1--welcome.html"><a href="week-1--welcome.html#moving-around-the-file-system"><i class="fa fa-check"></i><b>2.9</b> Moving around the file system<span></span></a></li>
<li class="chapter" data-level="2.10" data-path="week-1--welcome.html"><a href="week-1--welcome.html#navigating-practice-and-finding-hidden-directories"><i class="fa fa-check"></i><b>2.10</b> Navigating practice and finding hidden directories<span></span></a></li>
<li class="chapter" data-level="2.11" data-path="week-1--welcome.html"><a href="week-1--welcome.html#examining-the-contents-of-other-directories"><i class="fa fa-check"></i><b>2.11</b> Examining the contents of other directories<span></span></a></li>
<li class="chapter" data-level="2.12" data-path="week-1--welcome.html"><a href="week-1--welcome.html#full-vs-relative-paths"><i class="fa fa-check"></i><b>2.12</b> Full vs Relative Paths<span></span></a></li>
<li class="chapter" data-level="2.13" data-path="week-1--welcome.html"><a href="week-1--welcome.html#navigational-shortcuts"><i class="fa fa-check"></i><b>2.13</b> Navigational shortcuts<span></span></a></li>
<li class="chapter" data-level="2.14" data-path="week-1--welcome.html"><a href="week-1--welcome.html#key-points"><i class="fa fa-check"></i><b>2.14</b> Key Points<span></span></a></li>
<li class="chapter" data-level="2.15" data-path="week-1--welcome.html"><a href="week-1--welcome.html#creature-of-the-week"><i class="fa fa-check"></i><b>2.15</b> Creature of the Week!<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html"><i class="fa fa-check"></i><b>3</b> Week 2-Working With Files<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#our-data-set-fastq-files"><i class="fa fa-check"></i><b>3.1</b> Our data set: FASTQ files<span></span></a></li>
<li class="chapter" data-level="3.2" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#wildcards"><i class="fa fa-check"></i><b>3.2</b> Wildcards<span></span></a></li>
<li class="chapter" data-level="3.3" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#command-history"><i class="fa fa-check"></i><b>3.3</b> Command History<span></span></a></li>
<li class="chapter" data-level="3.4" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#examining-files"><i class="fa fa-check"></i><b>3.4</b> Examining Files<span></span></a></li>
<li class="chapter" data-level="3.5" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#details-on-the-fastq-format"><i class="fa fa-check"></i><b>3.5</b> Details on the FASTQ format<span></span></a></li>
<li class="chapter" data-level="3.6" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#creating-moving-copying-and-removing"><i class="fa fa-check"></i><b>3.6</b> Creating, moving, copying, and removing<span></span></a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#copying-files"><i class="fa fa-check"></i><b>3.6.1</b> Copying Files<span></span></a></li>
<li class="chapter" data-level="3.6.2" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#creating-directories"><i class="fa fa-check"></i><b>3.6.2</b> Creating Directories<span></span></a></li>
<li class="chapter" data-level="3.6.3" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#moving-renaming"><i class="fa fa-check"></i><b>3.6.3</b> Moving / Renaming<span></span></a></li>
<li class="chapter" data-level="3.6.4" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#file-permissions"><i class="fa fa-check"></i><b>3.6.4</b> File Permissions<span></span></a></li>
<li class="chapter" data-level="3.6.5" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#removing"><i class="fa fa-check"></i><b>3.6.5</b> Removing<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#searching-within-editing-and-redirecting-files"><i class="fa fa-check"></i><b>3.7</b> Searching within, editing, and redirecting files<span></span></a></li>
<li class="chapter" data-level="3.8" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#searching-files"><i class="fa fa-check"></i><b>3.8</b> Searching files<span></span></a></li>
<li class="chapter" data-level="3.9" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#redirecting-output"><i class="fa fa-check"></i><b>3.9</b> Redirecting output<span></span></a></li>
<li class="chapter" data-level="3.10" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#writing-scripts-and-working-with-data"><i class="fa fa-check"></i><b>3.10</b> Writing Scripts and Working with Data<span></span></a></li>
<li class="chapter" data-level="3.11" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#writing-files"><i class="fa fa-check"></i><b>3.11</b> Writing files<span></span></a></li>
<li class="chapter" data-level="3.12" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#writing-scripts"><i class="fa fa-check"></i><b>3.12</b> Writing scripts<span></span></a>
<ul>
<li class="chapter" data-level="3.12.1" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#uploading-and-downloading-data-to-your-virtual-machine-with-scp---unix"><i class="fa fa-check"></i><b>3.12.1</b> Uploading and Downloading Data to your Virtual Machine with scp - UNIX<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.13" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#creature-of-the-week-1"><i class="fa fa-check"></i><b>3.13</b> Creature of the Week!<span></span></a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="week-2-working-with-files" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">3</span> Week 2-Working With Files<a href="week-2-working-with-files.html#week-2-working-with-files" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Welcome to week 2! In this session we will be going over more tools to help us navigate our file system, view and manipulate files, and write basic scripts with bash/unix.</p>
<p> </p>
<ul>
<li>Questions:
<ul>
<li>How can I view and search file contents?</li>
<li>How can I create, copy and delete files and directories?</li>
<li>How can I control who has permission to modify a file?</li>
<li>How can I repeat recently used commands?</li>
</ul></li>
<li>Objectives:
<ul>
<li>View, search within, copy, move, and rename files. Create new directories.</li>
<li>Use wildcards (<code>*</code>) to perform operations on multiple files.</li>
<li>Make a file read only.</li>
<li>Use the <code>history</code> command to view and repeat recently used commands.</li>
</ul></li>
</ul>
<p> </p>
<p>Before we start we need to download our data. The data for each week will be in a .tar.gz file that we will uncompress into our MarineGenomics Directory.</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb56-1"><a href="week-2-working-with-files.html#cb56-1" aria-hidden="true" tabindex="-1"></a>$ cd /home/exouser/</span>
<span id="cb56-2"><a href="week-2-working-with-files.html#cb56-2" aria-hidden="true" tabindex="-1"></a>$ wget https://raw.githubusercontent.com/BayLab/MarineGenomicsData/main/week2_quarter.tar.gz</span>
<span id="cb56-3"><a href="week-2-working-with-files.html#cb56-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-4"><a href="week-2-working-with-files.html#cb56-4" aria-hidden="true" tabindex="-1"></a># use tar to uncompress the file</span>
<span id="cb56-5"><a href="week-2-working-with-files.html#cb56-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-6"><a href="week-2-working-with-files.html#cb56-6" aria-hidden="true" tabindex="-1"></a>$ tar -xzvf week2_quarter.tar.gz</span></code></pre></div>
<div id="our-data-set-fastq-files" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Our data set: FASTQ files<a href="week-2-working-with-files.html#our-data-set-fastq-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now that we know how to navigate around our directory structure, let’s
start working with our sequencing files. We did a sequencing experiment and
have two results files, which are stored in our <code>untrimmed_fastq</code> directory.</p>
</div>
<div id="wildcards" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Wildcards<a href="week-2-working-with-files.html#wildcards" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Navigate to your <code>untrimmed_fastq</code> directory:</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb57-1"><a href="week-2-working-with-files.html#cb57-1" aria-hidden="true" tabindex="-1"></a>$ cd ~/MarineGenomicsData/Week2/untrimmed_fastq</span></code></pre></div>
<p>We are interested in looking at the FASTQ files in this directory. We can list
all files with the .fastq extension using the command:</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb58-1"><a href="week-2-working-with-files.html#cb58-1" aria-hidden="true" tabindex="-1"></a>$ ls *.fastq</span></code></pre></div>
<div class="sourceCode" id="cb59"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb59-1"><a href="week-2-working-with-files.html#cb59-1" aria-hidden="true" tabindex="-1"></a>SRR097977.fastq  SRR098026.fastq</span></code></pre></div>
<p>The <code>*</code> character is a special type of character called a wildcard, which can be used to represent any type of character.
Thus, <code>*.fastq</code> matches every file that ends with <code>.fastq</code>.</p>
<p>This command:</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb60-1"><a href="week-2-working-with-files.html#cb60-1" aria-hidden="true" tabindex="-1"></a>$ ls *977.fastq</span></code></pre></div>
<div class="sourceCode" id="cb61"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb61-1"><a href="week-2-working-with-files.html#cb61-1" aria-hidden="true" tabindex="-1"></a>SRR097977.fastq</span></code></pre></div>
<p>lists only the file that ends with <code>977.fastq</code>.</p>
<p>This command:</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb62-1"><a href="week-2-working-with-files.html#cb62-1" aria-hidden="true" tabindex="-1"></a>$ ls /usr/bin/*.sh</span></code></pre></div>
<div class="sourceCode" id="cb63"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb63-1"><a href="week-2-working-with-files.html#cb63-1" aria-hidden="true" tabindex="-1"></a>/usr/bin/gettext.sh  /usr/bin/nvidia-bug-report.sh  /usr/bin/rescan-scsi-bus.sh</span></code></pre></div>
<p>Lists every file in <code>/usr/bin</code> that ends in the characters <code>.sh</code>.
Note that the output displays <strong>full</strong> paths to files, since
each result starts with <code>/</code>.</p>
<blockquote>
<h2 id="exercise-4" class="hasAnchor">Exercise<a href="week-2-working-with-files.html#exercise-4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Do each of the following tasks from your current directory using a single
<code>ls</code> command for each:</p>
<ol style="list-style-type: decimal">
<li>List all of the files in <code>/usr/bin</code> that start with the letter ‘c’</li>
<li>List all of the files in <code>/usr/bin</code> that contain the letter ‘a’</li>
<li>List all of the files in <code>/usr/bin</code> that end with the letter ‘o’</li>
</ol>
<p>Bonus: List all of the files in <code>/usr/bin</code> that contain the letter ‘a’ or the
letter ‘c’</p>
<p>Hint: The bonus question requires a Unix wildcard that we haven’t talked about
yet. Try searching the internet for information about Unix wildcards to find
what you need to solve the bonus problem.</p>
</blockquote>
<details>
<summary>
<span style="color: orange;">Solution</span>
</summary>
<p>
<blockquote>
<blockquote>
<h2 id="solution" class="hasAnchor">Solution<a href="week-2-working-with-files.html#solution" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ol style="list-style-type: decimal">
<li><code>ls /usr/bin/c*</code></li>
<li><code>ls /usr/bin/*a*</code></li>
<li><code>ls /usr/bin/*o</code><br />
Bonus: <code>ls /usr/bin/*[ac]*</code></li>
</ol>
</blockquote>
</blockquote>
</p>
</details>
<p> </p>
<blockquote>
<h2 id="exercise-5" class="hasAnchor">Exercise<a href="week-2-working-with-files.html#exercise-5" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><code>echo</code> is a built-in shell command that writes its arguments, like a line of text to standard output.
The <code>echo</code> command can also be used with pattern matching characters, such as wildcard characters.
Here we will use the <code>echo</code> command to see how the wildcard character is interpreted by the shell.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb64-1"><a href="week-2-working-with-files.html#cb64-1" aria-hidden="true" tabindex="-1"></a>$ echo *.fastq</span></code></pre></div>
<div class="sourceCode" id="cb65"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb65-1"><a href="week-2-working-with-files.html#cb65-1" aria-hidden="true" tabindex="-1"></a>SRR097977.fastq SRR098026.fastq</span></code></pre></div>
<p>The <code>*</code> is expanded to include any file that ends with <code>.fastq</code>. We can see that the output of
<code>echo *.fastq</code> is the same as that of <code>ls *.fastq</code>.</p>
<p>What would the output look like if the wildcard could <em>not</em> be matched? Compare the outputs of
<code>echo *.missing</code> and <code>ls *.missing</code>.</p>
</blockquote>
<details>
<summary>
<span style="color: orange;">Solution</span>
</summary>
<p>
<blockquote>
<blockquote>
<h2 id="solution-1" class="hasAnchor">Solution<a href="week-2-working-with-files.html#solution-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb66"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb66-1"><a href="week-2-working-with-files.html#cb66-1" aria-hidden="true" tabindex="-1"></a>$ echo *.missing</span></code></pre></div>
<div class="sourceCode" id="cb67"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb67-1"><a href="week-2-working-with-files.html#cb67-1" aria-hidden="true" tabindex="-1"></a>*.missing</span></code></pre></div>
<div class="sourceCode" id="cb68"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb68-1"><a href="week-2-working-with-files.html#cb68-1" aria-hidden="true" tabindex="-1"></a>$ ls *.missing</span></code></pre></div>
<div class="sourceCode" id="cb69"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb69-1"><a href="week-2-working-with-files.html#cb69-1" aria-hidden="true" tabindex="-1"></a>ls: cannot access &#39;*.missing&#39;: No such file or directory</span></code></pre></div>
</blockquote>
</blockquote>
</p>
</details>
<p> </p>
</div>
<div id="command-history" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Command History<a href="week-2-working-with-files.html#command-history" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>If you want to repeat a command that you’ve run recently, you can access previous
commands using the up arrow on your keyboard to go back to the most recent
command. Likewise, the down arrow takes you forward in the command history.</p>
<p>A few more useful shortcuts:</p>
<ul>
<li><kbd>Ctrl</kbd>+<kbd>C</kbd> will cancel the command you are writing, and give you a
fresh prompt.</li>
<li><kbd>Ctrl</kbd>+<kbd>R</kbd> will do a reverse-search through your command history. This
is very useful.</li>
<li><kbd>Ctrl</kbd>+<kbd>L</kbd> or the <code>clear</code> command will clear your screen.</li>
</ul>
<p>You can also review your recent commands with the <code>history</code> command, by entering:</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb70-1"><a href="week-2-working-with-files.html#cb70-1" aria-hidden="true" tabindex="-1"></a>$ history</span></code></pre></div>
<p>to see a numbered list of recent commands. You can reuse one of these commands
directly by referring to the number of that command.</p>
<p>For example, if your history looked like this:</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb71-1"><a href="week-2-working-with-files.html#cb71-1" aria-hidden="true" tabindex="-1"></a>259  ls *</span>
<span id="cb71-2"><a href="week-2-working-with-files.html#cb71-2" aria-hidden="true" tabindex="-1"></a>260  ls /usr/bin/*.sh</span>
<span id="cb71-3"><a href="week-2-working-with-files.html#cb71-3" aria-hidden="true" tabindex="-1"></a>261  ls *R1*fastq</span></code></pre></div>
<p>then you could repeat command #260 by entering:</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb72-1"><a href="week-2-working-with-files.html#cb72-1" aria-hidden="true" tabindex="-1"></a>$ !260</span></code></pre></div>
<p>Type <code>!</code> (exclamation point) and then the number of the command from your history.
You will be glad you learned this when you need to re-run very complicated commands.
For more information on advanced usage of <code>history</code>, read section 9.3 of
<a href="https://www.gnu.org/software/bash/manual/html_node/index.html">Bash manual</a>.</p>
<blockquote>
<h2 id="exercise-6" class="hasAnchor">Exercise<a href="week-2-working-with-files.html#exercise-6" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Find the line number in your history for the command that listed all the .sh
files in <code>/usr/bin</code>. Rerun that command.</p>
</blockquote>
<details>
<summary>
<span style="color: orange;">Solution</span>
</summary>
<p>
<blockquote>
<blockquote>
<h2 id="solution-2" class="hasAnchor">Solution<a href="week-2-working-with-files.html#solution-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>First type <code>history</code>. Then use <code>!</code> followed by the line number to rerun that command.</p>
</blockquote>
</blockquote>
</p>
</details>
<p> </p>
</div>
<div id="examining-files" class="section level2 hasAnchor" number="3.4">
<h2><span class="header-section-number">3.4</span> Examining Files<a href="week-2-working-with-files.html#examining-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We now know how to switch directories, run commands, and look at the
contents of directories, but how do we look at the contents of files?</p>
<p>One way to examine a file is to print out all of the
contents using the command <code>cat</code>.</p>
<p>Enter the following command from within the <code>untrimmed_fastq</code> directory:</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb73-1"><a href="week-2-working-with-files.html#cb73-1" aria-hidden="true" tabindex="-1"></a>$ cat SRR098026.fastq</span></code></pre></div>
<p>This will print out all of the contents of the <code>SRR098026.fastq</code> to the screen.</p>
<blockquote>
<h2 id="exercise-7" class="hasAnchor">Exercise<a href="week-2-working-with-files.html#exercise-7" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ol style="list-style-type: decimal">
<li>Print out the contents of the <code>./MarineGenomicsData/Week2/untrimmed_fastq/SRR097977.fastq</code> file. What is the last line of the file?</li>
<li>From your home directory, and without changing directories,
use one short command to print the contents of all of the files in
the <code>./MarineGenomicsData/Week2/untrimmed_fastq</code> directory.</li>
</ol>
</blockquote>
<details>
<summary>
<span style="color: orange;">Solution</span>
</summary>
<p>
<blockquote>
<blockquote>
<h2 id="solution-3" class="hasAnchor">Solution<a href="week-2-working-with-files.html#solution-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ol style="list-style-type: decimal">
<li>The last line of the file is <code>C:CCC::CCCCCCCC&lt;8?6A:C28C&lt;608'&amp;&amp;&amp;,'$</code>.</li>
<li><code>cat ./MarineGenomicsData/Week2/untrimmed_fastq/*</code></li>
</ol>
</blockquote>
</blockquote>
</p>
</details>
<p> </p>
<p><code>cat</code> is a terrific command, but when the file is really big, it can
be annoying to use. The program, <code>less</code>, is useful for this
case. <code>less</code> opens the file as read only, and lets you navigate through it.</p>
<p>Enter the following command:</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb74-1"><a href="week-2-working-with-files.html#cb74-1" aria-hidden="true" tabindex="-1"></a>$ less SRR097977.fastq</span></code></pre></div>
<p>Some navigation commands in <code>less</code>:</p>
<table>
<thead>
<tr class="header">
<th>key</th>
<th>action</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><kbd>Space</kbd></td>
<td>to go forward</td>
</tr>
<tr class="even">
<td><kbd>b</kbd></td>
<td>to go backward</td>
</tr>
<tr class="odd">
<td><kbd>g</kbd></td>
<td>to go to the beginning</td>
</tr>
<tr class="even">
<td><kbd>G</kbd></td>
<td>to go to the end</td>
</tr>
<tr class="odd">
<td><kbd>q</kbd></td>
<td>to quit</td>
</tr>
</tbody>
</table>
<p><code>less</code> also gives you a way of searching through files. Use the
“/” key to begin a search. Enter the word you would like
to search for and press <code>enter</code>. The screen will jump to the next location where
that word is found.</p>
<p><strong>Shortcut:</strong> If you hit “/” then “enter,” <code>less</code> will repeat
the previous search. <code>less</code> searches from the current location and
works its way forward. Note, if you are at the end of the file and search
for the sequence “CAA,” <code>less</code> will not find it. You either need to go to the
beginning of the file (by typing <code>g</code>) and search again using <code>/</code> or you
can use <code>?</code> to search backwards in the same way you used <code>/</code> previously.</p>
<p>For instance, let’s search forward for the sequence <code>TTTTT</code> in our file.
You can see that we go right to that sequence, what it looks like,
and where it is in the file. If you continue to type <code>/</code> and hit return, you will move
forward to the next instance of this sequence motif. If you instead type <code>?</code> and hit
return, you will search backwards and move up the file to previous examples of this motif.</p>
<blockquote>
<h2 id="exercise-8" class="hasAnchor">Exercise<a href="week-2-working-with-files.html#exercise-8" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>What are the next three nucleotides (characters) after the first instance of ‘TTTTT?’</p>
</blockquote>
<details>
<summary>
<span style="color: orange;">Solution</span>
</summary>
<p>
<blockquote>
<blockquote>
<h2 id="solution-4" class="hasAnchor">Solution<a href="week-2-working-with-files.html#solution-4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><code>CAC</code></p>
</blockquote>
</blockquote>
</p>
</details>
<p> </p>
<p>Remember, the <code>man</code> command actually uses <code>less</code> internally and
therefore uses the same commands, so you can search documentation
using “/” as well!</p>
<p>There’s another way that we can look at files, and in this case, just
look at part of them. This can be particularly useful if we just want
to see the beginning or end of the file, or see how it’s formatted.</p>
<p>The commands are <code>head</code> and <code>tail</code> and they let you look at
the beginning and end of a file, respectively.</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb75-1"><a href="week-2-working-with-files.html#cb75-1" aria-hidden="true" tabindex="-1"></a>$ head SRR098026.fastq</span></code></pre></div>
<div class="sourceCode" id="cb76"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb76-1"><a href="week-2-working-with-files.html#cb76-1" aria-hidden="true" tabindex="-1"></a>@SRR098026.1 HWUSI-EAS1599_1:2:1:0:968 length=35</span>
<span id="cb76-2"><a href="week-2-working-with-files.html#cb76-2" aria-hidden="true" tabindex="-1"></a>NNNNNNNNNNNNNNNNCNNNNNNNNNNNNNNNNNN</span>
<span id="cb76-3"><a href="week-2-working-with-files.html#cb76-3" aria-hidden="true" tabindex="-1"></a>+SRR098026.1 HWUSI-EAS1599_1:2:1:0:968 length=35</span>
<span id="cb76-4"><a href="week-2-working-with-files.html#cb76-4" aria-hidden="true" tabindex="-1"></a>!!!!!!!!!!!!!!!!#!!!!!!!!!!!!!!!!!!</span>
<span id="cb76-5"><a href="week-2-working-with-files.html#cb76-5" aria-hidden="true" tabindex="-1"></a>@SRR098026.2 HWUSI-EAS1599_1:2:1:0:312 length=35</span>
<span id="cb76-6"><a href="week-2-working-with-files.html#cb76-6" aria-hidden="true" tabindex="-1"></a>NNNNNNNNNNNNNNNNANNNNNNNNNNNNNNNNNN</span>
<span id="cb76-7"><a href="week-2-working-with-files.html#cb76-7" aria-hidden="true" tabindex="-1"></a>+SRR098026.2 HWUSI-EAS1599_1:2:1:0:312 length=35</span>
<span id="cb76-8"><a href="week-2-working-with-files.html#cb76-8" aria-hidden="true" tabindex="-1"></a>!!!!!!!!!!!!!!!!#!!!!!!!!!!!!!!!!!!</span>
<span id="cb76-9"><a href="week-2-working-with-files.html#cb76-9" aria-hidden="true" tabindex="-1"></a>@SRR098026.3 HWUSI-EAS1599_1:2:1:0:570 length=35</span>
<span id="cb76-10"><a href="week-2-working-with-files.html#cb76-10" aria-hidden="true" tabindex="-1"></a>NNNNNNNNNNNNNNNNANNNNNNNNNNNNNNNNNN</span></code></pre></div>
<div class="sourceCode" id="cb77"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb77-1"><a href="week-2-working-with-files.html#cb77-1" aria-hidden="true" tabindex="-1"></a>$ tail SRR098026.fastq</span></code></pre></div>
<div class="sourceCode" id="cb78"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb78-1"><a href="week-2-working-with-files.html#cb78-1" aria-hidden="true" tabindex="-1"></a>+SRR098026.247 HWUSI-EAS1599_1:2:1:2:1311 length=35</span>
<span id="cb78-2"><a href="week-2-working-with-files.html#cb78-2" aria-hidden="true" tabindex="-1"></a>#!##!#################!!!!!!!######</span>
<span id="cb78-3"><a href="week-2-working-with-files.html#cb78-3" aria-hidden="true" tabindex="-1"></a>@SRR098026.248 HWUSI-EAS1599_1:2:1:2:118 length=35</span>
<span id="cb78-4"><a href="week-2-working-with-files.html#cb78-4" aria-hidden="true" tabindex="-1"></a>GNTGNGGTCATCATACGCGCCCNNNNNNNGGCATG</span>
<span id="cb78-5"><a href="week-2-working-with-files.html#cb78-5" aria-hidden="true" tabindex="-1"></a>+SRR098026.248 HWUSI-EAS1599_1:2:1:2:118 length=35</span>
<span id="cb78-6"><a href="week-2-working-with-files.html#cb78-6" aria-hidden="true" tabindex="-1"></a>B!;?!A=5922:##########!!!!!!!######</span>
<span id="cb78-7"><a href="week-2-working-with-files.html#cb78-7" aria-hidden="true" tabindex="-1"></a>@SRR098026.249 HWUSI-EAS1599_1:2:1:2:1057 length=35</span>
<span id="cb78-8"><a href="week-2-working-with-files.html#cb78-8" aria-hidden="true" tabindex="-1"></a>CNCTNTATGCGTACGGCAGTGANNNNNNNGGAGAT</span>
<span id="cb78-9"><a href="week-2-working-with-files.html#cb78-9" aria-hidden="true" tabindex="-1"></a>+SRR098026.249 HWUSI-EAS1599_1:2:1:2:1057 length=35</span>
<span id="cb78-10"><a href="week-2-working-with-files.html#cb78-10" aria-hidden="true" tabindex="-1"></a>A!@B!BBB@ABAB#########!!!!!!!######</span></code></pre></div>
<p>The <code>-n</code> option to either of these commands can be used to print the
first or last <code>n</code> lines of a file.</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb79-1"><a href="week-2-working-with-files.html#cb79-1" aria-hidden="true" tabindex="-1"></a>$ head -n 1 SRR098026.fastq</span></code></pre></div>
<div class="sourceCode" id="cb80"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb80-1"><a href="week-2-working-with-files.html#cb80-1" aria-hidden="true" tabindex="-1"></a>@SRR098026.1 HWUSI-EAS1599_1:2:1:0:968 length=35</span></code></pre></div>
<div class="sourceCode" id="cb81"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb81-1"><a href="week-2-working-with-files.html#cb81-1" aria-hidden="true" tabindex="-1"></a>$ tail -n 1 SRR098026.fastq</span></code></pre></div>
<div class="sourceCode" id="cb82"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb82-1"><a href="week-2-working-with-files.html#cb82-1" aria-hidden="true" tabindex="-1"></a>A!@B!BBB@ABAB#########!!!!!!!######</span></code></pre></div>
</div>
<div id="details-on-the-fastq-format" class="section level2 hasAnchor" number="3.5">
<h2><span class="header-section-number">3.5</span> Details on the FASTQ format<a href="week-2-working-with-files.html#details-on-the-fastq-format" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Although it looks complicated (and it is), it’s easy to understand the
<a href="https://en.wikipedia.org/wiki/FASTQ_format">fastq</a> format with a little decoding. Some rules about the format
include…</p>
<table>
<colgroup>
<col width="26%" />
<col width="73%" />
</colgroup>
<thead>
<tr class="header">
<th>Line</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>Always begins with ‘@’ and then information about the read</td>
</tr>
<tr class="even">
<td>2</td>
<td>The actual DNA sequence</td>
</tr>
<tr class="odd">
<td>3</td>
<td>Always begins with a ‘+’ and sometimes the same info in line 1</td>
</tr>
<tr class="even">
<td>4</td>
<td>Has a string of characters which represent the quality scores; must have same number of characters as line 2</td>
</tr>
</tbody>
</table>
<p>We can view the first complete read in one of the files in our dataset by using <code>head</code> to look at
the first four lines.</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb83-1"><a href="week-2-working-with-files.html#cb83-1" aria-hidden="true" tabindex="-1"></a>$ head -n 4 SRR098026.fastq</span></code></pre></div>
<div class="sourceCode" id="cb84"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb84-1"><a href="week-2-working-with-files.html#cb84-1" aria-hidden="true" tabindex="-1"></a>@SRR098026.1 HWUSI-EAS1599_1:2:1:0:968 length=35</span>
<span id="cb84-2"><a href="week-2-working-with-files.html#cb84-2" aria-hidden="true" tabindex="-1"></a>NNNNNNNNNNNNNNNNCNNNNNNNNNNNNNNNNNN</span>
<span id="cb84-3"><a href="week-2-working-with-files.html#cb84-3" aria-hidden="true" tabindex="-1"></a>+SRR098026.1 HWUSI-EAS1599_1:2:1:0:968 length=35</span>
<span id="cb84-4"><a href="week-2-working-with-files.html#cb84-4" aria-hidden="true" tabindex="-1"></a>!!!!!!!!!!!!!!!!#!!!!!!!!!!!!!!!!!!</span></code></pre></div>
<p>All but one of the nucleotides in this read are unknown (<code>N</code>). This is a pretty bad read!</p>
<p>Line 4 shows the quality for each nucleotide in the read. Quality is interpreted as the
probability of an incorrect base call. To make it possible to line up each individual nucleotide with its quality score, the numerical score is converted into a code where each character
represents the numerical quality score for an individual nucleotide. For example, in the line
above, the quality score line is:</p>
<div class="sourceCode" id="cb85"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb85-1"><a href="week-2-working-with-files.html#cb85-1" aria-hidden="true" tabindex="-1"></a>!!!!!!!!!!!!!!!!#!!!!!!!!!!!!!!!!!!</span></code></pre></div>
<p>The <code>#</code> character and each of the <code>!</code> characters represent the encoded quality for an
individual nucleotide. The numerical value assigned to each of these characters depends on the
sequencing platform that generated the reads. The sequencing machine used to generate our data
uses the standard Sanger quality PHRED score encoding, Illumina version 1.8 onwards.</p>
<p>Here is a link showing what those different symbols mean for quality scores: <a href="https://support.illumina.com/help/BaseSpace_OLH_009008/Content/Source/Informatics/BS/QualityScoreEncoding_swBS.htm" class="uri">https://support.illumina.com/help/BaseSpace_OLH_009008/Content/Source/Informatics/BS/QualityScoreEncoding_swBS.htm</a></p>
<p>Each character is assigned a quality score between 0 and 42 as shown in the chart below.</p>
<div class="sourceCode" id="cb86"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb86-1"><a href="week-2-working-with-files.html#cb86-1" aria-hidden="true" tabindex="-1"></a>Quality encoding: !&quot;#$%<span class="er">&amp;</span>&#39;()*+,-./0123456789:;<span class="er">&lt;</span>=&gt;?@ABCDEFGHIJK</span>
<span id="cb86-2"><a href="week-2-working-with-files.html#cb86-2" aria-hidden="true" tabindex="-1"></a>                  |         |         |         |         |</span>
<span id="cb86-3"><a href="week-2-working-with-files.html#cb86-3" aria-hidden="true" tabindex="-1"></a>Quality score:    0........10........20........30........40..                          </span></code></pre></div>
<p>This quality score is logarithmically based, so a quality score of 10 reflects a
base call accuracy of 90%, but a quality score of 20 reflects a base call accuracy of 99%.</p>
<p>Looking back at our read:</p>
<div class="sourceCode" id="cb87"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb87-1"><a href="week-2-working-with-files.html#cb87-1" aria-hidden="true" tabindex="-1"></a>@SRR098026.1 HWUSI-EAS1599_1:2:1:0:968 length=35</span>
<span id="cb87-2"><a href="week-2-working-with-files.html#cb87-2" aria-hidden="true" tabindex="-1"></a>NNNNNNNNNNNNNNNNCNNNNNNNNNNNNNNNNNN</span>
<span id="cb87-3"><a href="week-2-working-with-files.html#cb87-3" aria-hidden="true" tabindex="-1"></a>+SRR098026.1 HWUSI-EAS1599_1:2:1:0:968 length=35</span>
<span id="cb87-4"><a href="week-2-working-with-files.html#cb87-4" aria-hidden="true" tabindex="-1"></a>!!!!!!!!!!!!!!!!#!!!!!!!!!!!!!!!!!!</span></code></pre></div>
<p>We can now see that the quality of each of the <code>N</code>s is 0 and the quality of the only
nucleotide call (<code>C</code>) is also very poor (<code>#</code> = a quality score of 2). This is indeed a very
bad read.</p>
</div>
<div id="creating-moving-copying-and-removing" class="section level2 hasAnchor" number="3.6">
<h2><span class="header-section-number">3.6</span> Creating, moving, copying, and removing<a href="week-2-working-with-files.html#creating-moving-copying-and-removing" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now we can move around in the file structure, look at files, and search files. But what if we want to copy files or move
them around or get rid of them? Most of the time, you can do these sorts of file manipulations without the command line,
but there will be some cases (like when you’re working with a remote computer like we are for this lesson) where it will be
impossible. You’ll also find that you may be working with hundreds of files and want to do similar manipulations to all
of those files. In cases like this, it’s much faster to do these operations at the command line.</p>
<div id="copying-files" class="section level3 hasAnchor" number="3.6.1">
<h3><span class="header-section-number">3.6.1</span> Copying Files<a href="week-2-working-with-files.html#copying-files" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>When working with computational data, it’s important to keep a safe copy of that data that can’t be accidentally overwritten or deleted.
For this lesson, that would be our raw data is our FASTQ files. We don’t want to accidentally change the original files, so we’ll make a copy of them
and change the file permissions so that we can read from, but not write to, the files.</p>
<p>First, let’s make a copy of one of our FASTQ files using the <code>cp</code> command.</p>
<p>Navigate to the <code>~/MarineGenomicsData/Week2/untrimmed_fastq</code> directory and enter:</p>
<div class="sourceCode" id="cb88"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb88-1"><a href="week-2-working-with-files.html#cb88-1" aria-hidden="true" tabindex="-1"></a>$ cp SRR098026.fastq SRR098026-copy.fastq</span>
<span id="cb88-2"><a href="week-2-working-with-files.html#cb88-2" aria-hidden="true" tabindex="-1"></a>$ ls -F</span></code></pre></div>
<div class="sourceCode" id="cb89"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb89-1"><a href="week-2-working-with-files.html#cb89-1" aria-hidden="true" tabindex="-1"></a>SRR097977.fastq*  SRR098026-copy.fastq*  SRR098026.fastq*</span></code></pre></div>
<p>We now have two copies of the <code>SRR098026.fastq</code> file, one of them named <code>SRR098026-copy.fastq</code>. We’ll move this file to a new directory
called <code>backup</code> where we’ll store our backup data files.</p>
</div>
<div id="creating-directories" class="section level3 hasAnchor" number="3.6.2">
<h3><span class="header-section-number">3.6.2</span> Creating Directories<a href="week-2-working-with-files.html#creating-directories" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <code>mkdir</code> command is used to make a directory. Enter <code>mkdir</code>
followed by a space, then the directory name you want to create:</p>
<div class="sourceCode" id="cb90"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb90-1"><a href="week-2-working-with-files.html#cb90-1" aria-hidden="true" tabindex="-1"></a>$ mkdir backup</span></code></pre></div>
</div>
<div id="moving-renaming" class="section level3 hasAnchor" number="3.6.3">
<h3><span class="header-section-number">3.6.3</span> Moving / Renaming<a href="week-2-working-with-files.html#moving-renaming" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We can now move our backup file to this directory. We can
move files around using the command <code>mv</code>:</p>
<div class="sourceCode" id="cb91"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb91-1"><a href="week-2-working-with-files.html#cb91-1" aria-hidden="true" tabindex="-1"></a>$ mv SRR098026-copy.fastq backup</span>
<span id="cb91-2"><a href="week-2-working-with-files.html#cb91-2" aria-hidden="true" tabindex="-1"></a>$ ls backup</span></code></pre></div>
<div class="sourceCode" id="cb92"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb92-1"><a href="week-2-working-with-files.html#cb92-1" aria-hidden="true" tabindex="-1"></a>SRR098026-copy.fastq</span></code></pre></div>
<p>The <code>mv</code> command is also how you rename files. Let’s rename this file to make it clear that this is a backup:</p>
<div class="sourceCode" id="cb93"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb93-1"><a href="week-2-working-with-files.html#cb93-1" aria-hidden="true" tabindex="-1"></a>$ cd backup</span>
<span id="cb93-2"><a href="week-2-working-with-files.html#cb93-2" aria-hidden="true" tabindex="-1"></a>$ mv SRR098026-copy.fastq SRR098026-backup.fastq</span>
<span id="cb93-3"><a href="week-2-working-with-files.html#cb93-3" aria-hidden="true" tabindex="-1"></a>$ ls</span></code></pre></div>
<div class="sourceCode" id="cb94"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb94-1"><a href="week-2-working-with-files.html#cb94-1" aria-hidden="true" tabindex="-1"></a>SRR098026-backup.fastq</span></code></pre></div>
</div>
<div id="file-permissions" class="section level3 hasAnchor" number="3.6.4">
<h3><span class="header-section-number">3.6.4</span> File Permissions<a href="week-2-working-with-files.html#file-permissions" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We’ve now made a backup copy of our file, but just because we have two copies, it doesn’t make us safe. We can still accidentally delete or
overwrite both copies. To make sure we can’t accidentally mess up this backup file, we’re going to change the permissions on the file so
that we’re only allowed to read (i.e. view) the file, not write to it (i.e. make new changes).</p>
<p>View the current permissions on a file using the <code>-l</code> (long) flag for the <code>ls</code> command:</p>
<div class="sourceCode" id="cb95"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb95-1"><a href="week-2-working-with-files.html#cb95-1" aria-hidden="true" tabindex="-1"></a>$ ls -l</span></code></pre></div>
<div class="sourceCode" id="cb96"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb96-1"><a href="week-2-working-with-files.html#cb96-1" aria-hidden="true" tabindex="-1"></a>-rwxrwxr-x 1 exouser exouser 43332 Mar 25 01:23 SRR098026-backup.fastq</span></code></pre></div>
<p>The first part of the output for the <code>-l</code> flag gives you information about the file’s current permissions. There are ten slots in the
permissions list. The first character in this list is related to file type, not permissions, so we’ll ignore it for now. The next three
characters relate to the permissions that the file owner has, the next three relate to the permissions for group members, and the final
three characters specify what other users outside of your group can do with the file. We’re going to concentrate on the three positions
that deal with your permissions (as the file user).</p>
<p><img src="figs/fig_week2/rwx_fig.png" alt="Permissions breakdown" />
Figure by <a href="https://www.guru99.com/file-permissions.html">Mary Brent</a></p>
<p>Here the three positions that relate to the file user/owner are <code>rwx</code>. The <code>r</code> means that you have permission to read the file, the <code>w</code>
indicates that you have permission to write to (i.e. make changes to) the file, and the third position is <code>x</code>, indicating that you have permission to carry out the ability encoded by that space (this is the space where <code>x</code> or executable ability is stored, we’ll
talk more about this in <a href="http://www.datacarpentry.org/shell-genomics/05-writing-scripts/">a later lesson</a>).</p>
<p>Our goal for now is to change permissions on this file so that you no longer have <code>w</code> or write permissions. We can do this using the <code>chmod</code> (change mode) command and subtracting (<code>-</code>) the write permission <code>-w</code>.</p>
<div class="sourceCode" id="cb97"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb97-1"><a href="week-2-working-with-files.html#cb97-1" aria-hidden="true" tabindex="-1"></a>$ chmod -w SRR098026-backup.fastq</span>
<span id="cb97-2"><a href="week-2-working-with-files.html#cb97-2" aria-hidden="true" tabindex="-1"></a>$ ls -l </span></code></pre></div>
<div class="sourceCode" id="cb98"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb98-1"><a href="week-2-working-with-files.html#cb98-1" aria-hidden="true" tabindex="-1"></a>-r-xr-xr-x 1 exouser exouser 43332 Mar 25 01:23 SRR098026-backup.fastq</span></code></pre></div>
<p>Chmod can also change the permission to only the user (u), group (g), and/or other (o). Let’s add reading (r), writing (x), and execute (x) permissions to group owners (because coding is better with friends!).</p>
<div class="sourceCode" id="cb99"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb99-1"><a href="week-2-working-with-files.html#cb99-1" aria-hidden="true" tabindex="-1"></a>$ chmod g=rwx SRR098026-backup.fastq</span>
<span id="cb99-2"><a href="week-2-working-with-files.html#cb99-2" aria-hidden="true" tabindex="-1"></a>$ ls -l </span></code></pre></div>
<div class="sourceCode" id="cb100"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb100-1"><a href="week-2-working-with-files.html#cb100-1" aria-hidden="true" tabindex="-1"></a>-r-xrwxr-x 1 exouser exouser 43332 Mar 25 01:23 SRR098026-backup.fastq</span></code></pre></div>
</div>
<div id="removing" class="section level3 hasAnchor" number="3.6.5">
<h3><span class="header-section-number">3.6.5</span> Removing<a href="week-2-working-with-files.html#removing" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To prove to ourselves that you no longer have the ability to modify this file, try deleting it with the <code>rm</code> command:</p>
<div class="sourceCode" id="cb101"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb101-1"><a href="week-2-working-with-files.html#cb101-1" aria-hidden="true" tabindex="-1"></a>$ rm SRR098026-backup.fastq</span></code></pre></div>
<p>You’ll be asked if you want to override your file permissions:</p>
<div class="sourceCode" id="cb102"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb102-1"><a href="week-2-working-with-files.html#cb102-1" aria-hidden="true" tabindex="-1"></a>rm: remove write-protected regular file ‘SRR098026-backup.fastq’? </span></code></pre></div>
<p>You should enter <code>n</code> for no. If you enter <code>n</code> (for no), the file will not be deleted. If you enter <code>y</code>, you will delete the file. This gives us an extra
measure of security, as there is one more step between us and deleting our data files.</p>
<p><strong>Important</strong>: The <code>rm</code> command permanently removes the file. Be careful with this command. It doesn’t
just nicely put the files in the Trash. They’re really gone.</p>
<p>By default, <code>rm</code> will not delete directories. You can tell <code>rm</code> to
delete a directory using the <code>-r</code> (recursive) option. Let’s delete the backup directory
we just made.</p>
<p>Enter the following command:</p>
<div class="sourceCode" id="cb103"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb103-1"><a href="week-2-working-with-files.html#cb103-1" aria-hidden="true" tabindex="-1"></a>$ cd ..</span>
<span id="cb103-2"><a href="week-2-working-with-files.html#cb103-2" aria-hidden="true" tabindex="-1"></a>$ rm -r backup</span></code></pre></div>
<p>This will delete not only the directory, but all files within the directory. If you have write-protected files in the directory,
you will be asked whether you want to override your permission settings (say yes here).</p>
<blockquote>
<h2 id="exercise-9" class="hasAnchor">Exercise<a href="week-2-working-with-files.html#exercise-9" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Starting in the <code>~/MarineGenomicsData/Week2/untrimmed_fastq/</code> directory, do the following:
1. Make sure that you have deleted your backup directory and all files it contains.<br />
2. Create a backup of each of your FASTQ files using <code>cp</code>. (Note: You’ll need to do this individually for each of the two FASTQ files. We haven’t
learned yet how to do this
with a wildcard.)<br />
3. Use a wildcard to move all of your backup files to a new backup directory.<br />
4. Change the permissions on all of your backup files to be write-protected.</p>
</blockquote>
<details>
<summary>
<span style="color: orange;">Solution</span>
</summary>
<p>
<blockquote>
<blockquote>
<h2 id="solution-5" class="hasAnchor">Solution<a href="week-2-working-with-files.html#solution-5" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ol style="list-style-type: decimal">
<li><code>rm -r backup</code><br />
</li>
<li><code>cp SRR098026.fastq SRR098026-backup.fastq</code> and <code>cp SRR097977.fastq SRR097977-backup.fastq</code><br />
</li>
<li><code>mkdir backup</code> and <code>mv *-backup.fastq backup</code></li>
<li><code>chmod -w backup/*-backup.fastq</code><br />
It’s always a good idea to check your work with <code>ls -l backup</code>. You should see something like:</li>
</ol>
<div class="sourceCode" id="cb104"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb104-1"><a href="week-2-working-with-files.html#cb104-1" aria-hidden="true" tabindex="-1"></a>-r--r--r-- 1 dcuser dcuser 47552 Nov 15 23:06 SRR097977-backup.fastq</span>
<span id="cb104-2"><a href="week-2-working-with-files.html#cb104-2" aria-hidden="true" tabindex="-1"></a>-r--r--r-- 1 dcuser dcuser 43332 Nov 15 23:06 SRR098026-backup.fastq</span></code></pre></div>
</blockquote>
</blockquote>
</p>
</details>
<p> </p>
<p>Now you have the ability to edit and explore your directories, and manipulate the files within them!</p>
<ul>
<li>Part 1 Keypoints:
<ul>
<li>You can view file contents using <code>less</code>, <code>cat</code>, <code>head</code> or <code>tail</code>.</li>
</ul></li>
<li>The commands <code>cp</code>, <code>mv</code>, <code>rm`` and</code>mkdir` are useful for manipulating existing files and creating new directories.</li>
<li>You can view file permissions using <code>ls -l</code> and change permissions using <code>chmod</code>.</li>
<li>The <code>history</code> command and the up arrow on your keyboard can be used to repeat recently used commands.</li>
</ul>
</div>
</div>
<div id="searching-within-editing-and-redirecting-files" class="section level2 hasAnchor" number="3.7">
<h2><span class="header-section-number">3.7</span> Searching within, editing, and redirecting files<a href="week-2-working-with-files.html#searching-within-editing-and-redirecting-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Questions:
<ul>
<li>How can I search within files?</li>
<li>How can I combine existing commands to do new things?</li>
</ul></li>
<li>Objectives:
<ul>
<li>Employ the <code>grep</code> command to search for information within files.</li>
<li>Print the results of a command to a file.</li>
<li>Construct command pipelines with two or more stages.</li>
</ul></li>
</ul>
</div>
<div id="searching-files" class="section level2 hasAnchor" number="3.8">
<h2><span class="header-section-number">3.8</span> Searching files<a href="week-2-working-with-files.html#searching-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In addition to <code>less</code>, we can also
search within files without even opening them, using <code>grep</code>. <code>grep</code> is a command-line
utility for searching plain-text files for lines matching a specific set of
characters (sometimes called a string).</p>
<p>Let’s give it a try!</p>
<blockquote>
<h3 id="nucleotide-abbreviations" class="hasAnchor">Nucleotide abbreviations<a href="week-2-working-with-files.html#nucleotide-abbreviations" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The four nucleotides that appear in DNA are abbreviated <code>A</code>, <code>C</code>, <code>T</code> and <code>G</code>.
Unknown nucleotides are represented with the letter <code>N</code>. An <code>N</code> appearing
in a sequencing file represents a position where the sequencing machine was not able to
confidently determine the nucleotide in that position. You can think of an <code>N</code> as being aNy
nucleotide at that position in the DNA sequence.</p>
</blockquote>
<p>We’ll search for strings inside of our fastq files. Let’s first make sure we are in the correct
directory:</p>
<div class="sourceCode" id="cb105"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb105-1"><a href="week-2-working-with-files.html#cb105-1" aria-hidden="true" tabindex="-1"></a>$ cd ~/MarineGenomicsData/Week2/data_week2/untrimmed_fastq</span></code></pre></div>
<p>Suppose we want to see how many reads in our file have really bad segments containing 10 consecutive unknown nucleotides (Ns).</p>
<blockquote>
<h3 id="determining-quality" class="hasAnchor">Determining quality<a href="week-2-working-with-files.html#determining-quality" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In this lesson, we’re going to be manually searching for strings of <code>N</code>s within our sequence
results to illustrate some principles of file searching. It can be really useful to do this
type of searching to get a feel for the quality of your sequencing results, however, in your
research you will most likely use a bioinformatics tool that has a built-in program for
filtering out low-quality reads. You’ll learn how to use one such tool in
<a href="https://datacarpentry.org/wrangling-genomics/02-quality-control/index.html">a later lesson</a>.</p>
</blockquote>
<p>Let’s search for the string NNNNNNNNNN in the SRR098026 file:</p>
<div class="sourceCode" id="cb106"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb106-1"><a href="week-2-working-with-files.html#cb106-1" aria-hidden="true" tabindex="-1"></a>$ grep NNNNNNNNNN SRR098026.fastq</span></code></pre></div>
<p>This command returns a lot of output to the terminal. Every single line in the SRR098026
file that contains at least 10 consecutive Ns is printed to the terminal, regardless of how long or short the file is.
We may be interested not only in the actual sequence which contains this string, but
in the name (or identifier) of that sequence. We discussed previously
that the identifier line immediately precedes the nucleotide sequence for each read
in a FASTQ file. We may also want to inspect the quality scores associated with
each of these reads. To get all of this information, we will return the line
immediately before each match and the two lines immediately after each match.</p>
<p>We can use the <code>-B</code> argument for grep to return a specific number of lines before
each match. The <code>-A</code> argument returns a specific number of lines after each matching line. Here we want the line <em>before</em> and the two lines <em>after</em> each
matching line, so we add <code>-B1 -A2</code> to our grep command:</p>
<div class="sourceCode" id="cb107"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb107-1"><a href="week-2-working-with-files.html#cb107-1" aria-hidden="true" tabindex="-1"></a>$ grep -B1 -A2 NNNNNNNNNN SRR098026.fastq</span></code></pre></div>
<p>One of the sets of lines returned by this command is:</p>
<div class="sourceCode" id="cb108"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb108-1"><a href="week-2-working-with-files.html#cb108-1" aria-hidden="true" tabindex="-1"></a>@SRR098026.177 HWUSI-EAS1599_1:2:1:1:2025 length=35</span>
<span id="cb108-2"><a href="week-2-working-with-files.html#cb108-2" aria-hidden="true" tabindex="-1"></a>CNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</span>
<span id="cb108-3"><a href="week-2-working-with-files.html#cb108-3" aria-hidden="true" tabindex="-1"></a>+SRR098026.177 HWUSI-EAS1599_1:2:1:1:2025 length=35</span>
<span id="cb108-4"><a href="week-2-working-with-files.html#cb108-4" aria-hidden="true" tabindex="-1"></a>#!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!</span></code></pre></div>
<blockquote>
<h3 id="exercise-10" class="hasAnchor">Exercise<a href="week-2-working-with-files.html#exercise-10" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p>Search for the sequence <code>GNATNACCACTTCC</code> in the <code>SRR098026.fastq</code> file.
Have your search return all matching lines and the name (or identifier) for each sequence
that contains a match.</p></li>
<li><p>Search for the sequence <code>AAGTT</code> in both FASTQ files.
Have your search return all matching lines and the name (or identifier) for each sequence
that contains a match.</p></li>
</ol>
</blockquote>
<details>
<summary>
<span style="color: orange;">Solution</span>
</summary>
<p>
<blockquote>
<blockquote>
<h3 id="solution-6" class="hasAnchor">Solution<a href="week-2-working-with-files.html#solution-6" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li><p><code>grep -B1 GNATNACCACTTCC SRR098026.fastq</code></p>
<pre><code>@SRR098026.245 HWUSI-EAS1599_1:2:1:2:801 length=35
GNATNACCACTTCCAGTGCTGANNNNNNNGGGATG</code></pre></li>
<li><p><code>grep -B1 AAGTT *.fastq</code></p>
<pre><code>SRR097977.fastq-@SRR097977.11 209DTAAXX_Lenski2_1_7:8:3:247:351 length=36
SRR097977.fastq:GATTGCTTTAATGAAAAAGTCATATAAGTTGCCATG
--
SRR097977.fastq-@SRR097977.67 209DTAAXX_Lenski2_1_7:8:3:544:566 length=36
SRR097977.fastq:TTGTCCACGCTTTTCTATGTAAAGTTTATTTGCTTT
--
SRR097977.fastq-@SRR097977.68 209DTAAXX_Lenski2_1_7:8:3:724:110 length=36
SRR097977.fastq:TGAAGCCTGCTTTTTTATACTAAGTTTGCATTATAA
--
SRR097977.fastq-@SRR097977.80 209DTAAXX_Lenski2_1_7:8:3:258:281 length=36
SRR097977.fastq:GTGGCGCTGCTGCATAAGTTGGGTTATCAGGTCGTT
--
SRR097977.fastq-@SRR097977.92 209DTAAXX_Lenski2_1_7:8:3:353:318 length=36
SRR097977.fastq:GGCAAAATGGTCCTCCAGCCAGGCCAGAAGCAAGTT
--
SRR097977.fastq-@SRR097977.139 209DTAAXX_Lenski2_1_7:8:3:703:655 length=36
SRR097977.fastq:TTTATTTGTAAAGTTTTGTTGAAATAAGGGTTGTAA
--
SRR097977.fastq-@SRR097977.238 209DTAAXX_Lenski2_1_7:8:3:592:919 length=36
SRR097977.fastq:TTCTTACCATCCTGAAGTTTTTTCATCTTCCCTGAT
--
SRR098026.fastq-@SRR098026.158 HWUSI-EAS1599_1:2:1:1:1505 length=35
SRR098026.fastq:GNNNNNNNNCAAAGTTGATCNNNNNNNNNTGTGCG</code></pre></li>
</ol>
</blockquote>
</blockquote>
</p>
</details>
<p> </p>
</div>
<div id="redirecting-output" class="section level2 hasAnchor" number="3.9">
<h2><span class="header-section-number">3.9</span> Redirecting output<a href="week-2-working-with-files.html#redirecting-output" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><code>grep</code> allowed us to identify sequences in our FASTQ files that match a particular pattern.
All of these sequences were printed to our terminal screen, but in order to work with these
sequences and perform other operations on them, we will need to capture that output in some
way.</p>
<p>We can do this with something called “redirection.” The idea is that
we are taking what would ordinarily be printed to the terminal screen and redirecting it to another location.
In our case, we want to print this information to a file so that we can look at it later and
use other commands to analyze this data.</p>
<p>The command for redirecting output to a file is <code>&gt;</code>.</p>
<p>Let’s try out this command and copy all the records (including all four lines of each record)
in our FASTQ files that contain
‘NNNNNNNNNN’ to another file called <code>bad_reads.fastq</code>.</p>
<div class="sourceCode" id="cb111"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb111-1"><a href="week-2-working-with-files.html#cb111-1" aria-hidden="true" tabindex="-1"></a>$ grep -B1 -A2 NNNNNNNNNN SRR098026.fastq &gt; bad_reads.fastq</span></code></pre></div>
<blockquote>
<h3 id="file-extensions" class="hasAnchor">File extensions<a href="week-2-working-with-files.html#file-extensions" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In the code above, we have saved the bad reads as a FASTQ file, which may seem like the obvious file extension as
it will be holding FASTQ formatted data that we’re extracting from our FASTQ files. But as we proceed with the exercises, we will be working with the exact same file, which we will now
save as a <code>.txt</code> file. This will still be a FASTQ file even though it does not have a<br />
name with a <code>.fastq</code> extension. We are doing this as using a <code>.fastq</code> extension will lead us to problems
when we move to using wildcards later in this episode. We’ll point out where this becomes
important. For now, it’s good that we’re thinking about file extensions!</p>
</blockquote>
<div class="sourceCode" id="cb112"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb112-1"><a href="week-2-working-with-files.html#cb112-1" aria-hidden="true" tabindex="-1"></a>$ grep -B1 -A2 NNNNNNNNNN SRR098026.fastq &gt; bad_reads.txt</span></code></pre></div>
<p>The prompt should sit there a little bit, and then it should look like nothing
happened. But type <code>ls</code>. You should see a new file called <code>bad_reads.txt</code>.</p>
<p>We can check the number of lines in our new file using a command called <code>wc</code>.
<code>wc</code> stands for <strong>word count</strong>. This command counts the number of words, lines, and characters
in a file. The FASTQ file may change over time, so given the potential for updates,
make sure your file matches your instructor’s output.</p>
<p>‘wc’ gives the following output:</p>
<div class="sourceCode" id="cb113"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb113-1"><a href="week-2-working-with-files.html#cb113-1" aria-hidden="true" tabindex="-1"></a>$ wc bad_reads.txt</span></code></pre></div>
<div class="sourceCode" id="cb114"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb114-1"><a href="week-2-working-with-files.html#cb114-1" aria-hidden="true" tabindex="-1"></a>  537  1073 23217 bad_reads.txt</span></code></pre></div>
<p>This will tell us the number of lines, words and characters in the file. If we
want only the number of lines, we can use the <code>-l</code> flag for <code>lines</code>.</p>
<div class="sourceCode" id="cb115"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb115-1"><a href="week-2-working-with-files.html#cb115-1" aria-hidden="true" tabindex="-1"></a>$ wc -l bad_reads.txt</span></code></pre></div>
<div class="sourceCode" id="cb116"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb116-1"><a href="week-2-working-with-files.html#cb116-1" aria-hidden="true" tabindex="-1"></a>537 bad_reads.txt</span></code></pre></div>
<blockquote>
<h3 id="exercise-11" class="hasAnchor">Exercise<a href="week-2-working-with-files.html#exercise-11" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>How many sequences are there in <code>SRR098026.fastq</code>? Remember that every sequence is formed by four lines.</p>
</blockquote>
<details>
<summary>
<span style="color: orange;">Solution</span>
</summary>
<p>
<blockquote>
<blockquote>
<h3 id="solution-7" class="hasAnchor">Solution<a href="week-2-working-with-files.html#solution-7" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb117"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb117-1"><a href="week-2-working-with-files.html#cb117-1" aria-hidden="true" tabindex="-1"></a>$ wc -l SRR098026.fastq</span></code></pre></div>
<div class="sourceCode" id="cb118"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb118-1"><a href="week-2-working-with-files.html#cb118-1" aria-hidden="true" tabindex="-1"></a>996</span></code></pre></div>
</blockquote>
<p>Now you can divide this number by four to get the number of sequences in your fastq file</p>
</blockquote>
</p>
</details>
<p> </p>
<blockquote>
<h3 id="exercise-12" class="hasAnchor">Exercise<a href="week-2-working-with-files.html#exercise-12" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>How many sequences in <code>SRR098026.fastq</code> contain at least 3 consecutive Ns?</p>
</blockquote>
<details>
<summary>
<span style="color: orange;">Solution</span>
</summary>
<p>
<blockquote>
<blockquote>
<h3 id="solution-8" class="hasAnchor">Solution<a href="week-2-working-with-files.html#solution-8" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb119"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb119-1"><a href="week-2-working-with-files.html#cb119-1" aria-hidden="true" tabindex="-1"></a>$ grep NNN SRR098026.fastq &gt; bad_reads.txt</span>
<span id="cb119-2"><a href="week-2-working-with-files.html#cb119-2" aria-hidden="true" tabindex="-1"></a>$ wc -l bad_reads.txt</span></code></pre></div>
<div class="sourceCode" id="cb120"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb120-1"><a href="week-2-working-with-files.html#cb120-1" aria-hidden="true" tabindex="-1"></a>249</span></code></pre></div>
</blockquote>
</blockquote>
</p>
</details>
<p> </p>
<p>We might want to search multiple FASTQ files for sequences that match our search pattern.
However, we need to be careful, because each time we use the <code>&gt;</code> command to redirect output
to a file, the new output will replace the output that was already present in the file.
This is called “overwriting” and, just like you don’t want to overwrite your video recording
of your kid’s first birthday party, you also want to avoid overwriting your data files.</p>
<div class="sourceCode" id="cb121"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb121-1"><a href="week-2-working-with-files.html#cb121-1" aria-hidden="true" tabindex="-1"></a>$ grep -B1 -A2 NNNNNNNNNN SRR098026.fastq &gt; bad_reads.txt</span>
<span id="cb121-2"><a href="week-2-working-with-files.html#cb121-2" aria-hidden="true" tabindex="-1"></a>$ wc -l bad_reads.txt</span></code></pre></div>
<div class="sourceCode" id="cb122"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb122-1"><a href="week-2-working-with-files.html#cb122-1" aria-hidden="true" tabindex="-1"></a>537 bad_reads.txt</span></code></pre></div>
<div class="sourceCode" id="cb123"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb123-1"><a href="week-2-working-with-files.html#cb123-1" aria-hidden="true" tabindex="-1"></a>$ grep -B1 -A2 NNNNNNNNNN SRR097977.fastq &gt; bad_reads.txt</span>
<span id="cb123-2"><a href="week-2-working-with-files.html#cb123-2" aria-hidden="true" tabindex="-1"></a>$ wc -l bad_reads.txt</span></code></pre></div>
<div class="sourceCode" id="cb124"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb124-1"><a href="week-2-working-with-files.html#cb124-1" aria-hidden="true" tabindex="-1"></a>0 bad_reads.txt</span></code></pre></div>
<p>Here, the output of our second call to <code>wc</code> shows that we no longer have any lines in our <code>bad_reads.txt</code> file. This is
because the second file we searched (<code>SRR097977.fastq</code>) does not contain any lines that match our
search sequence. So our file was overwritten and is now empty.</p>
<p>We can avoid overwriting our files by using the command <code>&gt;&gt;</code>. <code>&gt;&gt;</code> is known as the “append redirect” and will
append new output to the end of a file, rather than overwriting it.</p>
<div class="sourceCode" id="cb125"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb125-1"><a href="week-2-working-with-files.html#cb125-1" aria-hidden="true" tabindex="-1"></a>$ grep -B1 -A2 NNNNNNNNNN SRR098026.fastq &gt; bad_reads.txt</span>
<span id="cb125-2"><a href="week-2-working-with-files.html#cb125-2" aria-hidden="true" tabindex="-1"></a>$ wc -l bad_reads.txt</span></code></pre></div>
<div class="sourceCode" id="cb126"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb126-1"><a href="week-2-working-with-files.html#cb126-1" aria-hidden="true" tabindex="-1"></a>537 bad_reads.txt</span></code></pre></div>
<div class="sourceCode" id="cb127"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb127-1"><a href="week-2-working-with-files.html#cb127-1" aria-hidden="true" tabindex="-1"></a>$ grep -B1 -A2 NNNNNNNNNN SRR097977.fastq &gt;&gt; bad_reads.txt</span>
<span id="cb127-2"><a href="week-2-working-with-files.html#cb127-2" aria-hidden="true" tabindex="-1"></a>$ wc -l bad_reads.txt</span></code></pre></div>
<div class="sourceCode" id="cb128"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb128-1"><a href="week-2-working-with-files.html#cb128-1" aria-hidden="true" tabindex="-1"></a>537 bad_reads.txt</span></code></pre></div>
<p>The output of our second call to <code>wc</code> shows that we have not overwritten our original data.</p>
<p>We can also do this with a single line of code by using a wildcard:</p>
<div class="sourceCode" id="cb129"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb129-1"><a href="week-2-working-with-files.html#cb129-1" aria-hidden="true" tabindex="-1"></a>$ grep -B1 -A2 NNNNNNNNNN *.fastq &gt; bad_reads.txt</span>
<span id="cb129-2"><a href="week-2-working-with-files.html#cb129-2" aria-hidden="true" tabindex="-1"></a>$ wc -l bad_reads.txt</span></code></pre></div>
<div class="sourceCode" id="cb130"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb130-1"><a href="week-2-working-with-files.html#cb130-1" aria-hidden="true" tabindex="-1"></a>537 bad_reads.txt</span></code></pre></div>
<blockquote>
<h3 id="file-extensions---part-2" class="hasAnchor">File extensions - part 2<a href="week-2-working-with-files.html#file-extensions---part-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This is where we would have trouble if we were naming our output file with a <code>.fastq</code> extension.
If we already had a file called <code>bad_reads.fastq</code> (from our previous <code>grep</code> practice)
and then ran the command above using a <code>.fastq</code> extension instead of a <code>.txt</code> extension, <code>grep</code>
would give us a warning.</p>
<div class="sourceCode" id="cb131"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb131-1"><a href="week-2-working-with-files.html#cb131-1" aria-hidden="true" tabindex="-1"></a>grep -B1 -A2 NNNNNNNNNN *.fastq &gt; bad_reads.fastq</span></code></pre></div>
<div class="sourceCode" id="cb132"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb132-1"><a href="week-2-working-with-files.html#cb132-1" aria-hidden="true" tabindex="-1"></a>grep: input file ‘bad_reads.fastq’ is also the output</span></code></pre></div>
<p><code>grep</code> is letting you know that the output file <code>bad_reads.fastq</code> is also included in your
<code>grep</code> call because it matches the <code>*.fastq</code> pattern. Be careful with this as it can lead to
some unintended results.</p>
</blockquote>
<p>Since we might have multiple different criteria we want to search for,
creating a new output file each time has the potential to clutter up our workspace. We also
thus far haven’t been interested in the actual contents of those files, only in the number of
reads that we’ve found. We created the files to store the reads and then counted the lines in
the file to see how many reads matched our criteria. There’s a way to do this, however, that
doesn’t require us to create these intermediate files - the pipe command (<code>|</code>).</p>
<p>What <code>|</code> does is take the output of that command and immediately uses it as an input to another command.
Maybe our goal is to just look at an output of a grep command, so instead of typing a <code>grep</code> command and then subsequently typing <code>less</code>, we can save time and effort by piping them together!
Here we redirect our output from our <code>grep</code> call through the <code>less</code> command.</p>
<div class="sourceCode" id="cb133"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb133-1"><a href="week-2-working-with-files.html#cb133-1" aria-hidden="true" tabindex="-1"></a>$ grep -B1 -A2 NNNNNNNNNN SRR098026.fastq | less</span></code></pre></div>
<p>We can now see the output from our <code>grep</code> call within the <code>less</code> interface. We can use the up and down arrows
to scroll through the output and use <code>q</code> to exit <code>less</code>.</p>
<p>If we don’t want to create a file before counting lines of output from our <code>grep</code> search, we could directly pipe
the output of the grep search to the command <code>wc -l</code>. This can be helpful for investigating your output if you are not sure
you would like to save it to a file.</p>
<div class="sourceCode" id="cb134"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb134-1"><a href="week-2-working-with-files.html#cb134-1" aria-hidden="true" tabindex="-1"></a>$ grep -B1 -A2 NNNNNNNNNN SRR098026.fastq | wc -l </span></code></pre></div>
<p>Because we asked <code>grep</code> for all four lines of each FASTQ record, we need to divide the output by
four to get the number of sequences that match our search pattern. Since 802 / 4 = 200.5 and we
are expecting an integer number of records, there is something added or missing in <code>bad_reads.txt</code>.
If we explore <code>bad_reads.txt</code> using <code>less</code>, we might be able to notice what is causing the uneven
number of lines. Luckily, this issue happens by the end of the file so we can also spot it with <code>tail</code>.</p>
<div class="sourceCode" id="cb135"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb135-1"><a href="week-2-working-with-files.html#cb135-1" aria-hidden="true" tabindex="-1"></a>$ grep -B1 -A2 NNNNNNNNNN SRR098026.fastq &gt; bad_reads.txt</span>
<span id="cb135-2"><a href="week-2-working-with-files.html#cb135-2" aria-hidden="true" tabindex="-1"></a>$ tail bad_reads.txt</span></code></pre></div>
<div class="sourceCode" id="cb136"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb136-1"><a href="week-2-working-with-files.html#cb136-1" aria-hidden="true" tabindex="-1"></a>@SRR098026.133 HWUSI-EAS1599_1:2:1:0:1978 length=35</span>
<span id="cb136-2"><a href="week-2-working-with-files.html#cb136-2" aria-hidden="true" tabindex="-1"></a>ANNNNNNNNNTTCAGCGACTNNNNNNNNNNGTNGN</span>
<span id="cb136-3"><a href="week-2-working-with-files.html#cb136-3" aria-hidden="true" tabindex="-1"></a>+SRR098026.133 HWUSI-EAS1599_1:2:1:0:1978 length=35</span>
<span id="cb136-4"><a href="week-2-working-with-files.html#cb136-4" aria-hidden="true" tabindex="-1"></a>#!!!!!!!!!##########!!!!!!!!!!##!#!</span>
<span id="cb136-5"><a href="week-2-working-with-files.html#cb136-5" aria-hidden="true" tabindex="-1"></a>--</span>
<span id="cb136-6"><a href="week-2-working-with-files.html#cb136-6" aria-hidden="true" tabindex="-1"></a>--</span>
<span id="cb136-7"><a href="week-2-working-with-files.html#cb136-7" aria-hidden="true" tabindex="-1"></a>@SRR098026.177 HWUSI-EAS1599_1:2:1:1:2025 length=35</span>
<span id="cb136-8"><a href="week-2-working-with-files.html#cb136-8" aria-hidden="true" tabindex="-1"></a>CNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</span>
<span id="cb136-9"><a href="week-2-working-with-files.html#cb136-9" aria-hidden="true" tabindex="-1"></a>+SRR098026.177 HWUSI-EAS1599_1:2:1:1:2025 length=35</span>
<span id="cb136-10"><a href="week-2-working-with-files.html#cb136-10" aria-hidden="true" tabindex="-1"></a>#!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!</span></code></pre></div>
<p>The fifth and six lines in the output display “–” which is the default action for <code>grep</code> to separate groups of
lines matching the pattern, and indicate groups of lines which did not match the pattern so are not displayed.
To fix this issue, we can redirect the output of grep to a second instance of <code>grep</code> as follows.</p>
<div class="sourceCode" id="cb137"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb137-1"><a href="week-2-working-with-files.html#cb137-1" aria-hidden="true" tabindex="-1"></a>$ grep -B1 -A2 NNNNNNNNNN SRR098026.fastq | grep -v &#39;^--&#39; &gt; bad_reads.fastq</span>
<span id="cb137-2"><a href="week-2-working-with-files.html#cb137-2" aria-hidden="true" tabindex="-1"></a>tail bad_reads.fastq</span></code></pre></div>
<div class="sourceCode" id="cb138"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb138-1"><a href="week-2-working-with-files.html#cb138-1" aria-hidden="true" tabindex="-1"></a>+SRR098026.132 HWUSI-EAS1599_1:2:1:0:320 length=35</span>
<span id="cb138-2"><a href="week-2-working-with-files.html#cb138-2" aria-hidden="true" tabindex="-1"></a>#!!!!!!!!!##########!!!!!!!!!!##!#!</span>
<span id="cb138-3"><a href="week-2-working-with-files.html#cb138-3" aria-hidden="true" tabindex="-1"></a>@SRR098026.133 HWUSI-EAS1599_1:2:1:0:1978 length=35</span>
<span id="cb138-4"><a href="week-2-working-with-files.html#cb138-4" aria-hidden="true" tabindex="-1"></a>ANNNNNNNNNTTCAGCGACTNNNNNNNNNNGTNGN</span>
<span id="cb138-5"><a href="week-2-working-with-files.html#cb138-5" aria-hidden="true" tabindex="-1"></a>+SRR098026.133 HWUSI-EAS1599_1:2:1:0:1978 length=35</span>
<span id="cb138-6"><a href="week-2-working-with-files.html#cb138-6" aria-hidden="true" tabindex="-1"></a>#!!!!!!!!!##########!!!!!!!!!!##!#!</span>
<span id="cb138-7"><a href="week-2-working-with-files.html#cb138-7" aria-hidden="true" tabindex="-1"></a>@SRR098026.177 HWUSI-EAS1599_1:2:1:1:2025 length=35</span>
<span id="cb138-8"><a href="week-2-working-with-files.html#cb138-8" aria-hidden="true" tabindex="-1"></a>CNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</span>
<span id="cb138-9"><a href="week-2-working-with-files.html#cb138-9" aria-hidden="true" tabindex="-1"></a>+SRR098026.177 HWUSI-EAS1599_1:2:1:1:2025 length=35</span>
<span id="cb138-10"><a href="week-2-working-with-files.html#cb138-10" aria-hidden="true" tabindex="-1"></a>#!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!</span></code></pre></div>
<p>The <code>-v</code> option in the second <code>grep</code> search stands for <code>--invert-match</code> meaning <code>grep</code> will now only display the
lines which do not match the searched pattern, in this case <code>'^--'</code>. The caret (<code>^</code>) is an <strong>anchoring</strong>
character matching the beginning of the line, and the pattern has to be enclose by single quotes so <code>grep</code> does
not interpret the pattern as an extended option (starting with –).</p>
<p>Redirecting output is often not intuitive, and can take some time to get used to. Once you’re
comfortable with redirection, however, you’ll be able to combine any number of commands to
do all sorts of exciting things with your data!</p>
<p>None of the command line programs we’ve been learning
do anything all that impressive on their own, but when you start chaining
them together, you can do some really powerful things very
efficiently.</p>
<p> </p>
<ul>
<li>Keypoints:
<ul>
<li><code>grep</code> is a powerful search tool with many options for customization.</li>
<li><code>&gt;</code>, <code>&gt;&gt;</code>, and <code>|</code> are different ways of redirecting output.</li>
<li><code>command &gt; file</code> redirects a command’s output to a file.</li>
<li><code>command &gt;&gt; file</code> redirects a command’s output to a file without overwriting the existing contents of the file.</li>
<li><code>command_1 | command_2</code> redirects the output of the first command as input to the second command.</li>
</ul></li>
</ul>
</p>
</details>
<p> </p>
</div>
<div id="writing-scripts-and-working-with-data" class="section level2 hasAnchor" number="3.10">
<h2><span class="header-section-number">3.10</span> Writing Scripts and Working with Data<a href="week-2-working-with-files.html#writing-scripts-and-working-with-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Questions:
<ul>
<li>How can we automate a commonly used set of commands?</li>
</ul></li>
<li>Objectives:
<ul>
<li>Use the <code>nano</code> text editor to modify text files.</li>
<li>Write a basic shell script.</li>
<li>Use the <code>bash</code> command to execute a shell script.</li>
<li>Use <code>chmod</code> to make a script an executable program.</li>
</ul></li>
</ul>
<script language="javascript" type="text/javascript">
function set_page_view_defaults() {
    document.getElementById('div_win').style.display = 'block';
    document.getElementById('div_unix').style.display = 'none';
};

function change_content_by_platform(form_control){
    if (!form_control || document.getElementById(form_control).value == 'win') {
        set_page_view_defaults();
    } else if (document.getElementById(form_control).value == 'unix') {
        document.getElementById('div_win').style.display = 'none';
        document.getElementById('div_unix').style.display = 'block';
    } else {
        alert("Error: Missing platform value for 'change_content_by_platform()' script!");
    }
}

window.onload = set_page_view_defaults;
</script>
</div>
<div id="writing-files" class="section level2 hasAnchor" number="3.11">
<h2><span class="header-section-number">3.11</span> Writing files<a href="week-2-working-with-files.html#writing-files" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We’ve been able to do a lot of work with files that already exist, but what if we want to write our own files? We’re not going to type in a FASTA file, but we’ll see as we go through other tutorials, there are a lot of reasons we’ll want to write a file, or edit an existing file.</p>
<p>To add text to files, we’re going to use a text editor called Nano. We’re going to create a file to take notes about what we’ve been doing with the data files.</p>
<p>This is good practice when working in bioinformatics. We can create a file called <code>README.txt</code> that describes the data files in the directory or documents how the files in that directory were generated. As the name suggests, it’s a file that we or others should read to understand the information in that directory.</p>
<p>Let’s change our working directory to <code>~/MarineGenomicsData/Week2/untrimmed_fastq</code> using <code>cd</code>,
then run <code>nano</code> to create a file called <code>README.txt</code>:</p>
<div class="sourceCode" id="cb139"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb139-1"><a href="week-2-working-with-files.html#cb139-1" aria-hidden="true" tabindex="-1"></a>$ cd ~/MarineGenomicsData/Week2/untrimmed_fastq</span>
<span id="cb139-2"><a href="week-2-working-with-files.html#cb139-2" aria-hidden="true" tabindex="-1"></a>$ nano README.txt</span></code></pre></div>
<p>You should see something like this:</p>
<p><img src="figs/img_week2/nano.png" /></p>
<p>The text at the bottom of the screen shows the keyboard shortcuts for performing various tasks in <code>nano</code>. We will talk more about how to interpret this information soon.</p>
<details>
<summary>
Which Editor?
</summary>
<p>
<blockquote>
<h3 id="which-editor" class="hasAnchor">Which Editor?<a href="week-2-working-with-files.html#which-editor" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>When we say, “<code>nano</code> is a text editor,” we really do mean “text”: it can
only work with plain character data, not tables, images, or any other
human-friendly media. We use it in examples because it is one of the
least complex text editors. However, because of this trait, it may
not be powerful enough or flexible enough for the work you need to do
after this workshop. On Unix systems (such as Linux and Mac OS X),
many programmers use <a href="http://www.gnu.org/software/emacs/">Emacs</a> or
<a href="http://www.vim.org/">Vim</a> (both of which require more time to learn),
or a graphical editor such as
<a href="http://projects.gnome.org/gedit/">Gedit</a>. On Windows, you may wish to
use <a href="http://notepad-plus-plus.org/">Notepad++</a>. Windows also has a built-in
editor called <code>notepad</code> that can be run from the command line in the same
way as <code>nano</code> for the purposes of this lesson.</p>
<p>No matter what editor you use, you will need to know where it searches
for and saves files. If you start it from the shell, it will (probably)
use your current working directory as its default location.</p>
</blockquote>
</p>
</details>
<p> </p>
<p>Let’s type in a few lines of text. Describe what the files in this
directory are, what you’ve been doing with them, what date you created/edited them.
Once we’re happy with our text, we can press <kbd>Ctrl</kbd>-<kbd>O</kbd> (press the <kbd>Ctrl</kbd> or <kbd>Control</kbd> key and, while
holding it down, press the <kbd>O</kbd> key) to write our data to disk. You’ll be asked what file we want to save this to:
press <kbd>Return</kbd> to accept the suggested default of <code>README.txt</code>.</p>
<p>Once our file is saved, we can use <kbd>Ctrl</kbd>-<kbd>X</kbd> to quit the editor and
return to the shell.</p>
<blockquote>
<h3 id="control-ctrl-or-key" class="hasAnchor">Control, Ctrl, or ^ Key<a href="week-2-working-with-files.html#control-ctrl-or-key" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The Control key is also called the “Ctrl” key. There are various ways
in which using the Control key may be described. For example, you may
see an instruction to exit as pressing the <kbd>Ctrl</kbd> key and
the <kbd>X</kbd> key, described as any of:</p>
<ul>
<li><code>Control-X</code></li>
<li><code>Control+X</code></li>
<li><code>Ctrl-X</code></li>
<li><code>Ctrl+X</code></li>
<li><code>^X</code></li>
<li><code>C-x</code></li>
</ul>
<p>In <code>nano</code>, along the bottom of the screen you’ll see <code>^G Get Help ^O WriteOut</code>.
This means that you can use <kbd>Ctrl</kbd>-<kbd>G</kbd> to get help and <kbd>Ctrl</kbd>-<kbd>O</kbd> to save your
file.</p>
</blockquote>
<p>Now you’ve written a file! You can take a look at it with <code>less</code> or <code>cat</code>, or open it up again and edit it with <code>nano</code>.</p>
<details>
<summary>
Exercise
</summary>
<p>
<blockquote>
<h3 id="exercise-13" class="hasAnchor">Exercise<a href="week-2-working-with-files.html#exercise-13" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Open <code>README.txt</code> and at the bottom add a song title from your ideal ‘coding playlist’ and save the file.</p>
</blockquote>
<details>
<summary>
<span style="color: orange;">Solution</span>
</summary>
<p>
<blockquote>
<blockquote>
<h3 id="solution-9" class="hasAnchor">Solution<a href="week-2-working-with-files.html#solution-9" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Use <code>nano README.txt</code> to open the file.<br />
Add your song and then use <kbd>Ctrl</kbd>-<kbd>X</kbd> followed by <code>y</code> and <kbd>Enter</kbd> to save.</p>
</blockquote>
</blockquote>
</p>
</details>
<p> </p>
</p>
</details>
<p> </p>
</div>
<div id="writing-scripts" class="section level2 hasAnchor" number="3.12">
<h2><span class="header-section-number">3.12</span> Writing scripts<a href="week-2-working-with-files.html#writing-scripts" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A really powerful thing about the command line is that you can write scripts. Scripts let you save commands to run them and also lets you put multiple commands together. Though writing scripts may require an additional time investment initially, this can save you time as you run them repeatedly. Scripts can also address the challenge of reproducibility: if you need to repeat an analysis, you retain a record of your command history within the script.</p>
<p>One thing we will commonly want to do with sequencing results is pull out bad reads and write them to a file to see if we can figure out what’s going on with them. We’re going to look for reads with long sequences of N’s like we did before, but now we’re going to write a script, so we can run it each time we get new sequences, rather than type the code in by hand each time.</p>
<p>We’re going to create a new file to put this command in. We’ll call it <code>bad-reads-script.sh</code>. The <code>sh</code> isn’t required, but using that extension tells us that it’s a shell script.</p>
<div class="sourceCode" id="cb140"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb140-1"><a href="week-2-working-with-files.html#cb140-1" aria-hidden="true" tabindex="-1"></a>$ nano bad-reads-script.sh</span></code></pre></div>
<p>Bad reads have a lot of N’s, so we’re going to look for <code>NNNNNNNNNN</code> with <code>grep</code>. We want the whole FASTQ record, so we’re also going to get the one line above the sequence and the two lines below. We also want to look in all the files that end with <code>.fastq</code>, so we’re going to use the <code>*</code> wildcard.</p>
<div class="sourceCode" id="cb141"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb141-1"><a href="week-2-working-with-files.html#cb141-1" aria-hidden="true" tabindex="-1"></a>grep -B1 -A2 -h NNNNNNNNNN *.fastq | grep -v &#39;^--&#39; &gt; scripted_bad_reads.txt</span></code></pre></div>
<blockquote>
<h3 id="custom-grep-control" class="hasAnchor">Custom <code>grep</code> control<a href="week-2-working-with-files.html#custom-grep-control" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We introduced the <code>-v</code> option in <a href="http://www.datacarpentry.org/shell-genomics/04-redirection/">the previous episode</a>, now we
are using <code>-h</code> to “Suppress the prefixing of file names on output” according to the documentation shown by <code>man grep</code>.</p>
</blockquote>
<p>Type your <code>grep</code> command into the file and save it as before. Be careful that you did not add the <code>$</code> at the beginning of the line.</p>
<p>Now comes the neat part. We can run this script. Type:</p>
<div class="sourceCode" id="cb142"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb142-1"><a href="week-2-working-with-files.html#cb142-1" aria-hidden="true" tabindex="-1"></a>$ bash bad-reads-script.sh</span></code></pre></div>
<p>It will look like nothing happened, but now if you look at <code>scripted_bad_reads.txt</code>, you can see that there are now reads in the file.</p>
</p>
</details>
<p> </p>
<details>
<summary>
Making the script into a program Exercise
</summary>
<p>
<p>We had to type <code>bash</code> because we needed to tell the computer what program to use to run this script. Instead, we can turn this script into its own program. We need to tell it that it’s a program by making it executable. We can do this by changing the file permissions. We talked about permissions in <a href="http://www.datacarpentry.org/shell-genomics/03-working-with-files/">an earlier episode</a>.</p>
<p>First, let’s look at the current permissions.</p>
<div class="sourceCode" id="cb143"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb143-1"><a href="week-2-working-with-files.html#cb143-1" aria-hidden="true" tabindex="-1"></a>$ ls -l bad-reads-script.sh</span></code></pre></div>
<div class="sourceCode" id="cb144"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb144-1"><a href="week-2-working-with-files.html#cb144-1" aria-hidden="true" tabindex="-1"></a>-rw-rw-r-- 1 exouser exouser 75 Apr  8 12:46 bad-reads-script.sh</span></code></pre></div>
<p>We see that it says <code>-rw-r--r--</code>. This shows that the file can be read by any user and written to by the file owner (you). We want to change these permissions so that the file can be executed as a program using <code>./</code> .</p>
<p>We use the command <code>chmod</code> like we did earlier when we removed write permissions. Here we are adding (<code>+</code>) executable permissions (<code>+x</code>).</p>
<div class="sourceCode" id="cb145"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb145-1"><a href="week-2-working-with-files.html#cb145-1" aria-hidden="true" tabindex="-1"></a>$ chmod +x bad-reads-script.sh</span></code></pre></div>
<p>Now let’s look at the permissions again.</p>
<div class="sourceCode" id="cb146"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb146-1"><a href="week-2-working-with-files.html#cb146-1" aria-hidden="true" tabindex="-1"></a>$ ls -l bad-reads-script.sh</span></code></pre></div>
<div class="sourceCode" id="cb147"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb147-1"><a href="week-2-working-with-files.html#cb147-1" aria-hidden="true" tabindex="-1"></a>-rwxrwxr-x 1 exouser exouser 75 Apr  8 12:48 bad-reads-script.sh</span></code></pre></div>
<p>Now we see that it says <code>-rwxr-xr-x</code>. The <code>x</code>’s that are there now tell us we can run it as a program. So, let’s try it! We’ll need to put <code>./</code> at the beginning so the computer knows to look here in this directory for the program.</p>
<div class="sourceCode" id="cb148"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb148-1"><a href="week-2-working-with-files.html#cb148-1" aria-hidden="true" tabindex="-1"></a>$ ./bad-reads-script.sh</span></code></pre></div>
<p>The script should run the same way as before, but now we’ve created our very own computer program!</p>
</p>
</details>
<p> </p>
<div id="uploading-and-downloading-data-to-your-virtual-machine-with-scp---unix" class="section level3 hasAnchor" number="3.12.1">
<h3><span class="header-section-number">3.12.1</span> Uploading and Downloading Data to your Virtual Machine with scp - UNIX<a href="week-2-working-with-files.html#uploading-and-downloading-data-to-your-virtual-machine-with-scp---unix" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This section is for your general knowledge.</p>
<p><code>scp</code> stands for ‘secure copy protocol,’ and is a widely used UNIX tool for moving files
between computers. The simplest way to use <code>scp</code> is to run it in your local terminal,
and use it to copy a single file:</p>
<div class="sourceCode" id="cb149"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb149-1"><a href="week-2-working-with-files.html#cb149-1" aria-hidden="true" tabindex="-1"></a>scp <span class="kw">&lt;file</span><span class="ot"> I want to move</span><span class="kw">&gt;</span> <span class="kw">&lt;where</span><span class="ot"> I want to move it</span><span class="kw">&gt;</span></span></code></pre></div>
<p>In terminal, you can use the <code>scp</code> command to upload a file (e.g. local_file.txt) to the cluster home directory:</p>
<div class="sourceCode" id="cb150"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb150-1"><a href="week-2-working-with-files.html#cb150-1" aria-hidden="true" tabindex="-1"></a>$  scp local_file.txt UserName@cluster.address:/scratch/</span></code></pre></div>
<p>If you wanted to download data from your virtual machine, we would put the location of the folder within the virtual machine in the location of <file I want to move></p>
<div class="sourceCode" id="cb151"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb151-1"><a href="week-2-working-with-files.html#cb151-1" aria-hidden="true" tabindex="-1"></a>$  scp UserName@cluster.address:/scratch/VirtualMachine_file.txt /home/exouser/MarineGenomicsData/Week2/untrimmed_fastq/</span></code></pre></div>
</div>
</div>
<div id="creature-of-the-week-1" class="section level2 hasAnchor" number="3.13">
<h2><span class="header-section-number">3.13</span> Creature of the Week!<a href="week-2-working-with-files.html#creature-of-the-week-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="figure">
<img src="figs/creatures/sheepshead_wrasse.jpg" alt="" />
<p class="caption">California Sheepshead Wrasse smiling for a photo!</p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="week-1--welcome.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["MarGen2022.pdf", "MarGen2022.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
