<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>10 Week 9: Genome wide association study (GWAS) | MarGen2022</title>
  <meta name="description" content="" />
  <meta name="generator" content="bookdown 0.26 and GitBook 2.6.7" />

  <meta property="og:title" content="10 Week 9: Genome wide association study (GWAS) | MarGen2022" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="10 Week 9: Genome wide association study (GWAS) | MarGen2022" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="week-8--fst-and-outlier-analysis.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="libs/plotly-binding-4.10.0/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-2.5.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-2.5.1/plotly-latest.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Marine Genomics 2022</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction to Marine Genomics</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#cloud-computing-versus-individual-laptops"><i class="fa fa-check"></i><b>1.1</b> Cloud computing versus individual laptops</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#meet-our-instructors"><i class="fa fa-check"></i><b>1.2</b> Meet our instructors!</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#schedule"><i class="fa fa-check"></i><b>1.3</b> Schedule</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="week-1--welcome.html"><a href="week-1--welcome.html"><i class="fa fa-check"></i><b>2</b> Week 1- Welcome!</a>
<ul>
<li class="chapter" data-level="2.1" data-path="week-1--welcome.html"><a href="week-1--welcome.html#introduction-to-shell-computing-via-the-data-carpentry-tutorial"><i class="fa fa-check"></i><b>2.1</b> Introduction to shell computing via the data carpentry tutorial</a></li>
<li class="chapter" data-level="2.2" data-path="week-1--welcome.html"><a href="week-1--welcome.html#how-to-access-the-shell"><i class="fa fa-check"></i><b>2.2</b> How to access the shell</a></li>
<li class="chapter" data-level="2.3" data-path="week-1--welcome.html"><a href="week-1--welcome.html#best-practices-for-jetstream"><i class="fa fa-check"></i><b>2.3</b> Best practices for Jetstream</a></li>
<li class="chapter" data-level="2.4" data-path="week-1--welcome.html"><a href="week-1--welcome.html#week-1-objectives"><i class="fa fa-check"></i><b>2.4</b> Week 1 Objectives</a></li>
<li class="chapter" data-level="2.5" data-path="week-1--welcome.html"><a href="week-1--welcome.html#navigating-your-file-system"><i class="fa fa-check"></i><b>2.5</b> Navigating your file system</a></li>
<li class="chapter" data-level="2.6" data-path="week-1--welcome.html"><a href="week-1--welcome.html#shortcut-tab-completion"><i class="fa fa-check"></i><b>2.6</b> Shortcut: Tab Completion</a></li>
<li class="chapter" data-level="2.7" data-path="week-1--welcome.html"><a href="week-1--welcome.html#summary-key-points"><i class="fa fa-check"></i><b>2.7</b> Summary &amp; Key Points</a></li>
<li class="chapter" data-level="2.8" data-path="week-1--welcome.html"><a href="week-1--welcome.html#navigating-files-and-directories"><i class="fa fa-check"></i><b>2.8</b> Navigating Files and Directories</a></li>
<li class="chapter" data-level="2.9" data-path="week-1--welcome.html"><a href="week-1--welcome.html#moving-around-the-file-system"><i class="fa fa-check"></i><b>2.9</b> Moving around the file system</a></li>
<li class="chapter" data-level="2.10" data-path="week-1--welcome.html"><a href="week-1--welcome.html#navigating-practice-and-finding-hidden-directories"><i class="fa fa-check"></i><b>2.10</b> Navigating practice and finding hidden directories</a></li>
<li class="chapter" data-level="2.11" data-path="week-1--welcome.html"><a href="week-1--welcome.html#examining-the-contents-of-other-directories"><i class="fa fa-check"></i><b>2.11</b> Examining the contents of other directories</a></li>
<li class="chapter" data-level="2.12" data-path="week-1--welcome.html"><a href="week-1--welcome.html#full-vs-relative-paths"><i class="fa fa-check"></i><b>2.12</b> Full vs Relative Paths</a></li>
<li class="chapter" data-level="2.13" data-path="week-1--welcome.html"><a href="week-1--welcome.html#navigational-shortcuts"><i class="fa fa-check"></i><b>2.13</b> Navigational shortcuts</a></li>
<li class="chapter" data-level="2.14" data-path="week-1--welcome.html"><a href="week-1--welcome.html#key-points"><i class="fa fa-check"></i><b>2.14</b> Key Points</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html"><i class="fa fa-check"></i><b>3</b> Week 2-Working With Files</a>
<ul>
<li class="chapter" data-level="3.1" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#our-data-set-fastq-files"><i class="fa fa-check"></i><b>3.1</b> Our data set: FASTQ files</a></li>
<li class="chapter" data-level="3.2" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#wildcards"><i class="fa fa-check"></i><b>3.2</b> Wildcards</a></li>
<li class="chapter" data-level="3.3" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#command-history"><i class="fa fa-check"></i><b>3.3</b> Command History</a></li>
<li class="chapter" data-level="3.4" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#examining-files"><i class="fa fa-check"></i><b>3.4</b> Examining Files</a></li>
<li class="chapter" data-level="3.5" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#details-on-the-fastq-format"><i class="fa fa-check"></i><b>3.5</b> Details on the FASTQ format</a></li>
<li class="chapter" data-level="3.6" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#creating-moving-copying-and-removing"><i class="fa fa-check"></i><b>3.6</b> Creating, moving, copying, and removing</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#copying-files"><i class="fa fa-check"></i><b>3.6.1</b> Copying Files</a></li>
<li class="chapter" data-level="3.6.2" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#creating-directories"><i class="fa fa-check"></i><b>3.6.2</b> Creating Directories</a></li>
<li class="chapter" data-level="3.6.3" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#moving-renaming"><i class="fa fa-check"></i><b>3.6.3</b> Moving / Renaming</a></li>
<li class="chapter" data-level="3.6.4" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#file-permissions"><i class="fa fa-check"></i><b>3.6.4</b> File Permissions</a></li>
<li class="chapter" data-level="3.6.5" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#removing"><i class="fa fa-check"></i><b>3.6.5</b> Removing</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#searching-within-editing-and-redirecting-files"><i class="fa fa-check"></i><b>3.7</b> Searching within, editing, and redirecting files</a></li>
<li class="chapter" data-level="3.8" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#searching-files"><i class="fa fa-check"></i><b>3.8</b> Searching files</a></li>
<li class="chapter" data-level="3.9" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#redirecting-output"><i class="fa fa-check"></i><b>3.9</b> Redirecting output</a></li>
<li class="chapter" data-level="3.10" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#writing-scripts-and-working-with-data"><i class="fa fa-check"></i><b>3.10</b> Writing Scripts and Working with Data</a></li>
<li class="chapter" data-level="3.11" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#writing-files"><i class="fa fa-check"></i><b>3.11</b> Writing files</a></li>
<li class="chapter" data-level="3.12" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#writing-scripts"><i class="fa fa-check"></i><b>3.12</b> Writing scripts</a></li>
<li class="chapter" data-level="3.13" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#computing-magic-with-for-loops"><i class="fa fa-check"></i><b>3.13</b> Computing magic with for loops</a></li>
<li class="chapter" data-level="3.14" data-path="week-2-working-with-files.html"><a href="week-2-working-with-files.html#using-basename-in-for-loops"><i class="fa fa-check"></i><b>3.14</b> Using Basename in for loops</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html"><i class="fa fa-check"></i><b>4</b> Week 3- What is a Genetic Variant?</a>
<ul>
<li class="chapter" data-level="4.1" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#to-get-started-lets-dowload-the-data-and-install-a-few-programs"><i class="fa fa-check"></i><b>4.1</b> To get started lets dowload the data and install a few programs</a></li>
<li class="chapter" data-level="4.2" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#raw-read-quality-control"><i class="fa fa-check"></i><b>4.2</b> Raw read quality control</a></li>
<li class="chapter" data-level="4.3" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#trimming-to-remove-adapters"><i class="fa fa-check"></i><b>4.3</b> Trimming to remove adapters</a></li>
<li class="chapter" data-level="4.4" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#building-an-index-of-our-genome"><i class="fa fa-check"></i><b>4.4</b> Building an index of our genome</a></li>
<li class="chapter" data-level="4.5" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#map-reads-to-the-genome"><i class="fa fa-check"></i><b>4.5</b> Map reads to the genome</a></li>
<li class="chapter" data-level="4.6" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#sam-to-bam-file-conversion"><i class="fa fa-check"></i><b>4.6</b> sam to bam file conversion</a></li>
<li class="chapter" data-level="4.7" data-path="week-3--what-is-a-genetic-variant.html"><a href="week-3--what-is-a-genetic-variant.html#genotype-likelihoods"><i class="fa fa-check"></i><b>4.7</b> Genotype likelihoods</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="week-4---the-r-environment.html"><a href="week-4---the-r-environment.html"><i class="fa fa-check"></i><b>5</b> Week 4 - The R environment</a>
<ul>
<li class="chapter" data-level="5.1" data-path="week-4---the-r-environment.html"><a href="week-4---the-r-environment.html#create-a-project-in-rstudio"><i class="fa fa-check"></i><b>5.1</b> Create a project in Rstudio</a></li>
<li class="chapter" data-level="5.2" data-path="week-4---the-r-environment.html"><a href="week-4---the-r-environment.html#orientation-to-r-studio"><i class="fa fa-check"></i><b>5.2</b> Orientation to R studio</a></li>
<li class="chapter" data-level="5.3" data-path="week-4---the-r-environment.html"><a href="week-4---the-r-environment.html#manipulating-a-vector-object"><i class="fa fa-check"></i><b>5.3</b> Manipulating a vector object</a></li>
<li class="chapter" data-level="5.4" data-path="week-4---the-r-environment.html"><a href="week-4---the-r-environment.html#characterizing-a-dataframe"><i class="fa fa-check"></i><b>5.4</b> Characterizing a dataframe</a></li>
<li class="chapter" data-level="5.5" data-path="week-4---the-r-environment.html"><a href="week-4---the-r-environment.html#how-to-access-parts-of-the-data"><i class="fa fa-check"></i><b>5.5</b> How to access parts of the data</a></li>
<li class="chapter" data-level="5.6" data-path="week-4---the-r-environment.html"><a href="week-4---the-r-environment.html#data-manipulation"><i class="fa fa-check"></i><b>5.6</b> Data Manipulation</a></li>
<li class="chapter" data-level="5.7" data-path="week-4---the-r-environment.html"><a href="week-4---the-r-environment.html#practice-exploring-a-dataframe"><i class="fa fa-check"></i><b>5.7</b> Practice exploring a dataframe</a></li>
<li class="chapter" data-level="5.8" data-path="week-4---the-r-environment.html"><a href="week-4---the-r-environment.html#subsetting-datasets-logicals"><i class="fa fa-check"></i><b>5.8</b> Subsetting datasets &amp; logicals</a></li>
<li class="chapter" data-level="5.9" data-path="week-4---the-r-environment.html"><a href="week-4---the-r-environment.html#practice-subsetting-datasetslogicals"><i class="fa fa-check"></i><b>5.9</b> Practice Subsetting datasets/logicals</a></li>
<li class="chapter" data-level="5.10" data-path="week-4---the-r-environment.html"><a href="week-4---the-r-environment.html#a-few-tips-for-coding-in-rstudio"><i class="fa fa-check"></i><b>5.10</b> A few tips for coding in Rstudio</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html"><i class="fa fa-check"></i><b>6</b> Week 5- R Continued</a>
<ul>
<li class="chapter" data-level="6.1" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#plotting"><i class="fa fa-check"></i><b>6.1</b> 2.1 Plotting</a></li>
<li class="chapter" data-level="6.2" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#scatterplots"><i class="fa fa-check"></i><b>6.2</b> Scatterplots</a></li>
<li class="chapter" data-level="6.3" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#customizing-your-plot"><i class="fa fa-check"></i><b>6.3</b> Customizing your plot</a></li>
<li class="chapter" data-level="6.4" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#histograms-and-box-plots"><i class="fa fa-check"></i><b>6.4</b> Histograms and box plots</a></li>
<li class="chapter" data-level="6.5" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#practice-problems-2.1"><i class="fa fa-check"></i><b>6.5</b> Practice Problems 2.1</a></li>
<li class="chapter" data-level="6.6" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#plotting-with-ggplot2"><i class="fa fa-check"></i><b>6.6</b> 2.2 plotting with ggplot2</a></li>
<li class="chapter" data-level="6.7" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#group-exercise-on-data-visualization"><i class="fa fa-check"></i><b>6.7</b> Group Exercise on Data Visualization</a></li>
<li class="chapter" data-level="6.8" data-path="week-5--r-continued.html"><a href="week-5--r-continued.html#practice-problems-2.2"><i class="fa fa-check"></i><b>6.8</b> Practice Problems 2.2</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="population-structure-via-construct.html"><a href="population-structure-via-construct.html"><i class="fa fa-check"></i><b>7</b> Population Structure via conStruct</a>
<ul>
<li class="chapter" data-level="7.1" data-path="population-structure-via-construct.html"><a href="population-structure-via-construct.html#download-the-data"><i class="fa fa-check"></i><b>7.1</b> Download the data</a></li>
<li class="chapter" data-level="7.2" data-path="population-structure-via-construct.html"><a href="population-structure-via-construct.html#update-our-compiler-in-bashunix-and-install-everything-for-r-in-the-terminal-in-bashunix-not-in-rstudio"><i class="fa fa-check"></i><b>7.2</b> update our compiler in bash/UNIX and install everything for R in the terminal (in bash/UNIX, NOT in Rstudio)</a></li>
<li class="chapter" data-level="7.3" data-path="population-structure-via-construct.html"><a href="population-structure-via-construct.html#install-construct-in-r"><i class="fa fa-check"></i><b>7.3</b> install conStruct in R</a></li>
<li class="chapter" data-level="7.4" data-path="population-structure-via-construct.html"><a href="population-structure-via-construct.html#read-in-the-data"><i class="fa fa-check"></i><b>7.4</b> read in the data</a></li>
<li class="chapter" data-level="7.5" data-path="population-structure-via-construct.html"><a href="population-structure-via-construct.html#get-our-meta-data-read-into-r-and-ready-to-use"><i class="fa fa-check"></i><b>7.5</b> Get our meta data read into R and ready to use</a></li>
<li class="chapter" data-level="7.6" data-path="population-structure-via-construct.html"><a href="population-structure-via-construct.html#run-a-structure-analysis"><i class="fa fa-check"></i><b>7.6</b> Run a structure analysis</a></li>
<li class="chapter" data-level="7.7" data-path="population-structure-via-construct.html"><a href="population-structure-via-construct.html#incoorporate-geographic-distances-in-a-spatial-model"><i class="fa fa-check"></i><b>7.7</b> Incoorporate geographic distances in a spatial model</a></li>
<li class="chapter" data-level="7.8" data-path="population-structure-via-construct.html"><a href="population-structure-via-construct.html#comparisons-across-different-values-of-k"><i class="fa fa-check"></i><b>7.8</b> Comparisons across different values of K</a></li>
<li class="chapter" data-level="7.9" data-path="population-structure-via-construct.html"><a href="population-structure-via-construct.html#exercises"><i class="fa fa-check"></i><b>7.9</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="week-7---principle-component-analyses.html"><a href="week-7---principle-component-analyses.html"><i class="fa fa-check"></i><b>8</b> Week 7 - Principle Component Analyses</a>
<ul>
<li class="chapter" data-level="8.1" data-path="week-7---principle-component-analyses.html"><a href="week-7---principle-component-analyses.html#download-the-data-1"><i class="fa fa-check"></i><b>8.1</b> Download the data</a></li>
<li class="chapter" data-level="8.2" data-path="week-7---principle-component-analyses.html"><a href="week-7---principle-component-analyses.html#installing-programs"><i class="fa fa-check"></i><b>8.2</b> Installing programs</a></li>
<li class="chapter" data-level="8.3" data-path="week-7---principle-component-analyses.html"><a href="week-7---principle-component-analyses.html#run-pcangsd-on-our-data"><i class="fa fa-check"></i><b>8.3</b> Run pcangsd on our data</a></li>
<li class="chapter" data-level="8.4" data-path="week-7---principle-component-analyses.html"><a href="week-7---principle-component-analyses.html#setting-up-a-new-project-in-r"><i class="fa fa-check"></i><b>8.4</b> Setting up a new project in R</a></li>
<li class="chapter" data-level="8.5" data-path="week-7---principle-component-analyses.html"><a href="week-7---principle-component-analyses.html#reading-data-into-r"><i class="fa fa-check"></i><b>8.5</b> Reading data into R</a></li>
<li class="chapter" data-level="8.6" data-path="week-7---principle-component-analyses.html"><a href="week-7---principle-component-analyses.html#group-exercises"><i class="fa fa-check"></i><b>8.6</b> Group Exercises</a></li>
<li class="chapter" data-level="8.7" data-path="week-7---principle-component-analyses.html"><a href="week-7---principle-component-analyses.html#additional-exercise"><i class="fa fa-check"></i><b>8.7</b> Additional Exercise</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="week-8--fst-and-outlier-analysis.html"><a href="week-8--fst-and-outlier-analysis.html"><i class="fa fa-check"></i><b>9</b> Week 8- F<sub>st</sub> and outlier analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="week-8--fst-and-outlier-analysis.html"><a href="week-8--fst-and-outlier-analysis.html#download-the-data-2"><i class="fa fa-check"></i><b>9.1</b> Download the data</a></li>
<li class="chapter" data-level="9.2" data-path="week-8--fst-and-outlier-analysis.html"><a href="week-8--fst-and-outlier-analysis.html#getting-r-set-up"><i class="fa fa-check"></i><b>9.2</b> Getting R set up</a></li>
<li class="chapter" data-level="9.3" data-path="week-8--fst-and-outlier-analysis.html"><a href="week-8--fst-and-outlier-analysis.html#finding-outliers-using-pcadapt"><i class="fa fa-check"></i><b>9.3</b> Finding outliers using pcadapt</a></li>
<li class="chapter" data-level="9.4" data-path="week-8--fst-and-outlier-analysis.html"><a href="week-8--fst-and-outlier-analysis.html#pcadapt-exercises"><i class="fa fa-check"></i><b>9.4</b> pcadapt Exercises</a></li>
<li class="chapter" data-level="9.5" data-path="week-8--fst-and-outlier-analysis.html"><a href="week-8--fst-and-outlier-analysis.html#using-fst-to-find-outliers"><i class="fa fa-check"></i><b>9.5</b> Using F<sub>st</sub> to find outliers</a></li>
<li class="chapter" data-level="9.6" data-path="week-8--fst-and-outlier-analysis.html"><a href="week-8--fst-and-outlier-analysis.html#outflank-practice"><i class="fa fa-check"></i><b>9.6</b> OutFLANK Practice</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="week-9-genome-wide-association-study-gwas.html"><a href="week-9-genome-wide-association-study-gwas.html"><i class="fa fa-check"></i><b>10</b> Week 9: Genome wide association study (GWAS)</a>
<ul>
<li class="chapter" data-level="10.1" data-path="week-9-genome-wide-association-study-gwas.html"><a href="week-9-genome-wide-association-study-gwas.html#load-in-the-data"><i class="fa fa-check"></i><b>10.1</b> load in the data</a></li>
<li class="chapter" data-level="10.2" data-path="week-9-genome-wide-association-study-gwas.html"><a href="week-9-genome-wide-association-study-gwas.html#install-packages-in-r"><i class="fa fa-check"></i><b>10.2</b> install packages in R</a></li>
<li class="chapter" data-level="10.3" data-path="week-9-genome-wide-association-study-gwas.html"><a href="week-9-genome-wide-association-study-gwas.html#the-data"><i class="fa fa-check"></i><b>10.3</b> The data</a></li>
<li class="chapter" data-level="10.4" data-path="week-9-genome-wide-association-study-gwas.html"><a href="week-9-genome-wide-association-study-gwas.html#take-beagle-file-and-generate-lrt-file"><i class="fa fa-check"></i><b>10.4</b> take Beagle file and generate lrt file</a></li>
<li class="chapter" data-level="10.5" data-path="week-9-genome-wide-association-study-gwas.html"><a href="week-9-genome-wide-association-study-gwas.html#take-lrt-file-and-make-a-manhattan-plot"><i class="fa fa-check"></i><b>10.5</b> take lrt file and make a manhattan plot</a></li>
<li class="chapter" data-level="10.6" data-path="week-9-genome-wide-association-study-gwas.html"><a href="week-9-genome-wide-association-study-gwas.html#exercises-1"><i class="fa fa-check"></i><b>10.6</b> Exercises</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="week-9-genome-wide-association-study-gwas" class="section level1 hasAnchor" number="10">
<h1><span class="header-section-number">10</span> Week 9: Genome wide association study (GWAS)<a href="week-9-genome-wide-association-study-gwas.html#week-9-genome-wide-association-study-gwas" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This week we’re going to show you how to perform a genome wide association study or GWAS.</p>
<div id="load-in-the-data" class="section level2 hasAnchor" number="10.1">
<h2><span class="header-section-number">10.1</span> load in the data<a href="week-9-genome-wide-association-study-gwas.html#load-in-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The data for this week consist of several files in beagle format, a genome index file (.fai), and a few small text files.
Let’s start by downloading the data we will be working with this week:</p>
<div class="sourceCode" id="cb475"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb475-1"><a href="week-9-genome-wide-association-study-gwas.html#cb475-1" aria-hidden="true" tabindex="-1"></a>wget https://raw.githubusercontent.com/BayLab/MarineGenomicsData/main/Week9_2022.tar.gz</span>
<span id="cb475-2"><a href="week-9-genome-wide-association-study-gwas.html#cb475-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb475-3"><a href="week-9-genome-wide-association-study-gwas.html#cb475-3" aria-hidden="true" tabindex="-1"></a>tar -xzvf Week9_2022.tar.gz</span></code></pre></div>
<p>Move the data from MarineGenomicsData into MarineGenomics and change its name to Week9:</p>
<pre><code>mv MarineGenomicsData/Week8 MarineGenomics/Week9</code></pre>
<p>and one more file that was too big to upload with the others</p>
<div class="sourceCode" id="cb477"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb477-1"><a href="week-9-genome-wide-association-study-gwas.html#cb477-1" aria-hidden="true" tabindex="-1"></a>cd MarineGenomics/Week9</span>
<span id="cb477-2"><a href="week-9-genome-wide-association-study-gwas.html#cb477-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb477-3"><a href="week-9-genome-wide-association-study-gwas.html#cb477-3" aria-hidden="true" tabindex="-1"></a>wget https://raw.githubusercontent.com/BayLab/MarineGenomicsData/main/salmon_chr6_19ind.BEAGLE.PL.gz</span></code></pre></div>
<p>Now that we’ve finished downloading the data, let’s open Rsutdio and install the package necessary for our today’s lesson</p>
</div>
<div id="install-packages-in-r" class="section level2 hasAnchor" number="10.2">
<h2><span class="header-section-number">10.2</span> install packages in R<a href="week-9-genome-wide-association-study-gwas.html#install-packages-in-r" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We’ll be using the package <code>qqman</code> to make a manhattan plot. Make sure not to copy the # sign into the R console/script.</p>
<div class="sourceCode" id="cb478"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb478-1"><a href="week-9-genome-wide-association-study-gwas.html#cb478-1" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages(&quot;qqman&quot;)</span></span></code></pre></div>
</div>
<div id="the-data" class="section level2 hasAnchor" number="10.3">
<h2><span class="header-section-number">10.3</span> The data<a href="week-9-genome-wide-association-study-gwas.html#the-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The data we’re working on for the GWAS comes from the paper by Kijas et al. 2018 where they studied the genetic basis of sex determination in the atlantic Salmon. The paper can be found <a href="https://www.nature.com/articles/s41598-018-23984-1#Sec2">here</a> or on Canvas. In short they examined the genetic basis of sex in 19 salmon for which they have whole genome sequence data. We’ll only be looking at two chromosomes (2 and 6) of this data.</p>
</div>
<div id="take-beagle-file-and-generate-lrt-file" class="section level2 hasAnchor" number="10.4">
<h2><span class="header-section-number">10.4</span> take Beagle file and generate lrt file<a href="week-9-genome-wide-association-study-gwas.html#take-beagle-file-and-generate-lrt-file" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To do the test of genome wide association we need to take our Beagle file and test whether there is an association with our phenotype (in this case whether a fish has a male or female phenotype). The phenotypes are coded as 0 = Female and 1 = Male in the <code>phenobin</code> file.</p>
<p>Let’s move back to the terminal and execute this command:</p>
<div class="sourceCode" id="cb479"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb479-1"><a href="week-9-genome-wide-association-study-gwas.html#cb479-1" aria-hidden="true" tabindex="-1"></a>$HOME/angsd/angsd -doMaf 4 -beagle salmon_chr2_19ind.BEAGLE.PL.gz -yBin phenobin -doAsso 2 -fai Salmon.fai</span></code></pre></div>
<p>This example looks for associations between the genotypes (the genotype likelihood data - beagle file) and the phenotypes (male/female - binary ‘phenobin’ file). We are using the ‘Salmon.fai’ file as an index (remember we are using an index for the genome so that it is easier to search against it).
This will generate several output files labeled <code>angsdput</code>. We’ll use the file with the <code>lrt0</code> extension to plot our manattan plot.</p>
<p>Let’s go back to Rstudio:</p>
</div>
<div id="take-lrt-file-and-make-a-manhattan-plot" class="section level2 hasAnchor" number="10.5">
<h2><span class="header-section-number">10.5</span> take lrt file and make a manhattan plot<a href="week-9-genome-wide-association-study-gwas.html#take-lrt-file-and-make-a-manhattan-plot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>LRT is the likelihood ratio statistic. This statistic is chi square distributed with one degree of freedom. Sites that fails one of the filters are given the value -999.</p>
<div class="sourceCode" id="cb480"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb480-1"><a href="week-9-genome-wide-association-study-gwas.html#cb480-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Set the working directory for this week&#39;s directory</span></span>
<span id="cb480-2"><a href="week-9-genome-wide-association-study-gwas.html#cb480-2" aria-hidden="true" tabindex="-1"></a><span class="co">#setwd(&quot;~/MarineGenomics/Week9&quot;)</span></span>
<span id="cb480-3"><a href="week-9-genome-wide-association-study-gwas.html#cb480-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb480-4"><a href="week-9-genome-wide-association-study-gwas.html#cb480-4" aria-hidden="true" tabindex="-1"></a><span class="co">#read in the data</span></span>
<span id="cb480-5"><a href="week-9-genome-wide-association-study-gwas.html#cb480-5" aria-hidden="true" tabindex="-1"></a>lrt <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="fu">gzfile</span>(<span class="st">&quot;angsdput.lrt0.gz&quot;</span>), <span class="at">header=</span>T, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb480-6"><a href="week-9-genome-wide-association-study-gwas.html#cb480-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb480-7"><a href="week-9-genome-wide-association-study-gwas.html#cb480-7" aria-hidden="true" tabindex="-1"></a><span class="co">#look at it</span></span>
<span id="cb480-8"><a href="week-9-genome-wide-association-study-gwas.html#cb480-8" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(lrt)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    224460 obs. of  8 variables:
##  $ Chromosome   : int  2 2 2 2 2 2 2 2 2 2 ...
##  $ Position     : int  13 59 74 153 548 550 613 1182 1361 1533 ...
##  $ Major        : Factor w/ 4 levels &quot;A&quot;,&quot;C&quot;,&quot;G&quot;,&quot;T&quot;: 3 3 3 2 2 2 2 2 2 2 ...
##  $ Minor        : Factor w/ 4 levels &quot;A&quot;,&quot;C&quot;,&quot;G&quot;,&quot;T&quot;: 4 1 1 3 4 4 1 4 1 1 ...
##  $ Frequency    : num  0.237 0.5 0.316 0.184 0.316 ...
##  $ N            : int  19 19 19 19 19 19 19 19 19 19 ...
##  $ LRT          : num  -999 -999 -999 -999 -999 -999 -999 -999 -999 -999 ...
##  $ high_WT.HE.HO: Factor w/ 803 levels &quot;0/0/19&quot;,&quot;0/1/18&quot;,..: 125 67 679 198 679 679 372 327 358 372 ...</code></pre>
<div class="sourceCode" id="cb482"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb482-1"><a href="week-9-genome-wide-association-study-gwas.html#cb482-1" aria-hidden="true" tabindex="-1"></a><span class="co">#we have a few LRT values that are -999, that means that the test failed. we should remove them. How many do we have?</span></span>
<span id="cb482-2"><a href="week-9-genome-wide-association-study-gwas.html#cb482-2" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">which</span>(lrt<span class="sc">$</span>LRT <span class="sc">==</span> <span class="sc">-</span><span class="dv">999</span>))</span></code></pre></div>
<pre><code>## [1] 210597</code></pre>
<p>The answer is 210597.</p>
<p>How many LRT values (in this case, loci) do we have in total?</p>
<div class="sourceCode" id="cb484"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb484-1"><a href="week-9-genome-wide-association-study-gwas.html#cb484-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(lrt<span class="sc">$</span>LRT)</span></code></pre></div>
<pre><code>## [1] 224460</code></pre>
<p>The answer is 224460, meaning that most are filtered out.</p>
<p>How many do we have left when we remove loci with failed tests?</p>
<div class="sourceCode" id="cb486"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb486-1"><a href="week-9-genome-wide-association-study-gwas.html#cb486-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(lrt<span class="sc">$</span>LRT)<span class="sc">-</span><span class="fu">length</span>(<span class="fu">which</span>(lrt<span class="sc">$</span>LRT <span class="sc">==</span> <span class="sc">-</span><span class="dv">999</span>))</span></code></pre></div>
<pre><code>## [1] 13863</code></pre>
<p>The answer is 13863.</p>
<p>Let’s remove the values that are not -999 and or negative and assign them to a new object called ‘lrt_filt.’</p>
<div class="sourceCode" id="cb488"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb488-1"><a href="week-9-genome-wide-association-study-gwas.html#cb488-1" aria-hidden="true" tabindex="-1"></a>lrt_filt <span class="ot">&lt;-</span> lrt[<span class="sc">-</span><span class="fu">c</span>(<span class="fu">which</span>(lrt<span class="sc">$</span>LRT <span class="sc">==</span> <span class="sc">-</span><span class="dv">999</span>),<span class="fu">which</span>(lrt<span class="sc">$</span>LRT <span class="sc">&lt;=</span> <span class="dv">0</span>)),]</span></code></pre></div>
<p>We can look at a histogram of the filtered LRT results:</p>
<div class="sourceCode" id="cb489"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb489-1"><a href="week-9-genome-wide-association-study-gwas.html#cb489-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(lrt_filt<span class="sc">$</span>LRT, <span class="at">breaks=</span><span class="dv">50</span>)</span></code></pre></div>
<p><img src="09-Week9.gwas_files/figure-html/8-1-1.png" width="672" /></p>
<p>Everything looks good to proceed to making our manhattan plot.</p>
<p>The function ‘manhattan’ requires each SNP to have it’s own “name.” Let’s make a vector for row numbers that start with the letter ‘r.’</p>
<div class="sourceCode" id="cb490"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb490-1"><a href="week-9-genome-wide-association-study-gwas.html#cb490-1" aria-hidden="true" tabindex="-1"></a>lrt_filt<span class="sc">$</span>SNP <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;r&quot;</span>,<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(lrt_filt<span class="sc">$</span>Chromosome), <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span></code></pre></div>
<p>We also need to convert our LRT values to pvalues. We’ll use the command <code>dchisq</code> to get pvalues from the LRT values.</p>
<div class="sourceCode" id="cb491"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb491-1"><a href="week-9-genome-wide-association-study-gwas.html#cb491-1" aria-hidden="true" tabindex="-1"></a><span class="co">#get pvalues</span></span>
<span id="cb491-2"><a href="week-9-genome-wide-association-study-gwas.html#cb491-2" aria-hidden="true" tabindex="-1"></a>lrt_filt<span class="sc">$</span>pvalue<span class="ot">&lt;-</span><span class="fu">dchisq</span>(lrt_filt<span class="sc">$</span>LRT, <span class="at">df=</span><span class="dv">1</span>)</span>
<span id="cb491-3"><a href="week-9-genome-wide-association-study-gwas.html#cb491-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb491-4"><a href="week-9-genome-wide-association-study-gwas.html#cb491-4" aria-hidden="true" tabindex="-1"></a><span class="co">#we also need to make sure we don&#39;t have any tricky values like those below</span></span>
<span id="cb491-5"><a href="week-9-genome-wide-association-study-gwas.html#cb491-5" aria-hidden="true" tabindex="-1"></a>lrt_filt <span class="ot">&lt;-</span> lrt_filt[<span class="sc">!</span>(lrt_filt<span class="sc">$</span>pvalue<span class="sc">==</span><span class="st">&quot;NaN&quot;</span> <span class="sc">|</span> lrt_filt<span class="sc">$</span>LRT<span class="sc">==</span><span class="st">&quot;Inf&quot;</span><span class="sc">|</span> lrt_filt<span class="sc">$</span>pvalue<span class="sc">==</span><span class="st">&quot;Inf&quot;</span>),]</span></code></pre></div>
<p>Create the manhattan plot:</p>
<div class="sourceCode" id="cb492"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb492-1"><a href="week-9-genome-wide-association-study-gwas.html#cb492-1" aria-hidden="true" tabindex="-1"></a>qqman<span class="sc">::</span><span class="fu">manhattan</span>(lrt_filt, <span class="at">chr=</span><span class="st">&quot;Chromosome&quot;</span>, <span class="at">bp=</span><span class="st">&quot;Position&quot;</span>, <span class="at">p=</span><span class="st">&quot;pvalue&quot;</span>, <span class="at">suggestiveline =</span> F, <span class="at">genomewideline =</span> F)</span></code></pre></div>
<p><img src="09-Week9.gwas_files/figure-html/8-2-1.png" width="672" /></p>
<p>Let’s look at a qq-plot of our pvalues to check the model fit</p>
<div class="sourceCode" id="cb493"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb493-1"><a href="week-9-genome-wide-association-study-gwas.html#cb493-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qq</span>(lrt_filt<span class="sc">$</span>pvalue)</span></code></pre></div>
<p><img src="09-Week9.gwas_files/figure-html/8-3-1.png" width="672" /></p>
<p>This looks a bit weird, we would expect it to be mostly a straight line with some deviations at the upper right. If we were moving forward with this analyses we’d want to do more filtering of our data.</p>
<p>We can highlight the values that exceed a threshold. There are several ways to determine a threshold, but one is to make a vector of random phenotypes and re-run our association test. We can then set the highest LRT value from the random phenotype test as our upper limit for our plot with the actual phenotypes.</p>
<div class="sourceCode" id="cb494"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb494-1"><a href="week-9-genome-wide-association-study-gwas.html#cb494-1" aria-hidden="true" tabindex="-1"></a><span class="co">#make a vector with 19 1&#39;s and 0&#39;s</span></span>
<span id="cb494-2"><a href="week-9-genome-wide-association-study-gwas.html#cb494-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb494-3"><a href="week-9-genome-wide-association-study-gwas.html#cb494-3" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">0</span>), <span class="dv">19</span>, <span class="at">replace=</span>T)</span>
<span id="cb494-4"><a href="week-9-genome-wide-association-study-gwas.html#cb494-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb494-5"><a href="week-9-genome-wide-association-study-gwas.html#cb494-5" aria-hidden="true" tabindex="-1"></a><span class="co">#write this to our week 9 directory</span></span>
<span id="cb494-6"><a href="week-9-genome-wide-association-study-gwas.html#cb494-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb494-7"><a href="week-9-genome-wide-association-study-gwas.html#cb494-7" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(x, <span class="st">&quot;rando_pheno&quot;</span>, <span class="at">row.names =</span> F, <span class="at">col.names =</span> F)</span></code></pre></div>
<p>And now use that phenotype file to run our association test again, making sure to specify a different output file.
Let’s go back to the terminal:</p>
<div class="sourceCode" id="cb495"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb495-1"><a href="week-9-genome-wide-association-study-gwas.html#cb495-1" aria-hidden="true" tabindex="-1"></a>$HOME/angsd/angsd -doMaf 4 -beagle salmon_chr2_19ind.BEAGLE.PL.gz -yBin rando_pheno -doAsso 2 -fai Salmon.fai -out randotest</span></code></pre></div>
<p>And rerun the code in R to see what our maximum LRT values are in this random phenotype test.</p>
<div class="sourceCode" id="cb496"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb496-1"><a href="week-9-genome-wide-association-study-gwas.html#cb496-1" aria-hidden="true" tabindex="-1"></a>lrt_rando <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="fu">gzfile</span>(<span class="st">&quot;randotest.lrt0.gz&quot;</span>), <span class="at">header=</span>T, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb496-2"><a href="week-9-genome-wide-association-study-gwas.html#cb496-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb496-3"><a href="week-9-genome-wide-association-study-gwas.html#cb496-3" aria-hidden="true" tabindex="-1"></a><span class="co">#we need to remove those -999 values again</span></span>
<span id="cb496-4"><a href="week-9-genome-wide-association-study-gwas.html#cb496-4" aria-hidden="true" tabindex="-1"></a>rando_filt <span class="ot">&lt;-</span> lrt_rando[<span class="sc">!</span>(lrt_rando<span class="sc">$</span>LRT<span class="sc">==-</span><span class="dv">999</span> <span class="sc">|</span> lrt_rando<span class="sc">$</span>LRT<span class="sc">&lt;=</span> <span class="dv">0</span>),]</span>
<span id="cb496-5"><a href="week-9-genome-wide-association-study-gwas.html#cb496-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb496-6"><a href="week-9-genome-wide-association-study-gwas.html#cb496-6" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(rando_filt<span class="sc">$</span>LRT, <span class="at">na.rm=</span>T)</span></code></pre></div>
<pre><code>##     Min.  1st Qu.   Median     Mean  3rd Qu.     Max.     NA&#39;s 
## 0.000001 0.214816 0.669551      Inf 2.029913      Inf        1</code></pre>
<div class="sourceCode" id="cb498"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb498-1"><a href="week-9-genome-wide-association-study-gwas.html#cb498-1" aria-hidden="true" tabindex="-1"></a><span class="co">#we have some Inf values we also need to remove, let&#39;s add those to our filtering line above</span></span>
<span id="cb498-2"><a href="week-9-genome-wide-association-study-gwas.html#cb498-2" aria-hidden="true" tabindex="-1"></a>rando_filt <span class="ot">&lt;-</span> lrt_rando[<span class="sc">!</span>(lrt_rando<span class="sc">$</span>LRT<span class="sc">==-</span><span class="dv">999</span> <span class="sc">|</span> lrt_rando<span class="sc">$</span>LRT<span class="sc">&lt;=</span> <span class="dv">0</span> <span class="sc">|</span> lrt_rando<span class="sc">$</span>LRT <span class="sc">==</span> <span class="cn">Inf</span>),]</span>
<span id="cb498-3"><a href="week-9-genome-wide-association-study-gwas.html#cb498-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb498-4"><a href="week-9-genome-wide-association-study-gwas.html#cb498-4" aria-hidden="true" tabindex="-1"></a><span class="co">#let&#39;s look at the maximum value of LRT</span></span>
<span id="cb498-5"><a href="week-9-genome-wide-association-study-gwas.html#cb498-5" aria-hidden="true" tabindex="-1"></a><span class="fu">max</span>(rando_filt<span class="sc">$</span>LRT, <span class="at">na.rm=</span>T)</span></code></pre></div>
<pre><code>## [1] 12.33287</code></pre>
<p>The maximum value is ~12 This value was generated at random, so any value higher than this is expected to be generated not at random (which is what we are looking for). So we can highlight all of the SNPs that have an LRT greater than 12 in our association test.</p>
<div class="sourceCode" id="cb500"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb500-1"><a href="week-9-genome-wide-association-study-gwas.html#cb500-1" aria-hidden="true" tabindex="-1"></a><span class="co">#make a list of the candidates</span></span>
<span id="cb500-2"><a href="week-9-genome-wide-association-study-gwas.html#cb500-2" aria-hidden="true" tabindex="-1"></a>candidates <span class="ot">&lt;-</span> lrt_filt[<span class="fu">which</span>(lrt_filt<span class="sc">$</span>LRT <span class="sc">&gt;</span> <span class="dv">12</span>),]<span class="sc">$</span>SNP </span></code></pre></div>
<p>Let’s highlight this list of candidate loci in our manhattan plot:</p>
<div class="sourceCode" id="cb501"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb501-1"><a href="week-9-genome-wide-association-study-gwas.html#cb501-1" aria-hidden="true" tabindex="-1"></a>qqman<span class="sc">::</span><span class="fu">manhattan</span>(lrt_filt, <span class="at">chr=</span><span class="st">&quot;Chromosome&quot;</span>, <span class="at">bp=</span><span class="st">&quot;Position&quot;</span>,  <span class="at">p=</span><span class="st">&quot;pvalue&quot;</span>, <span class="at">highlight =</span> candidates, <span class="at">suggestiveline =</span> F, <span class="at">genomewideline =</span> F)</span></code></pre></div>
<p><img src="09-Week9.gwas_files/figure-html/8-4-1.png" width="672" /></p>
<p>Comparing our results to the Kijas et al. 2018 paper, we have a similar pattern of many SNPs across the chromosome showing a relationship with phenotype (sex). Interestingly this paper found that there are three chromosomes that are associated with sex in Atlantic Salmon, but not all chromosomes give a strong signal in all individuals. For example, only three male individuals were found to have a clear association with chromosome 2, and the other males in the study were found to have an association with chromosomes 3 and 6.</p>
<p>These results highlight the fluid nature of sex determination in animals, even those with a genetic basis to sex determination.</p>
<p>For these exercises you’ll take a closer look at chromosome 6, where you’ll try to find the individuals that are males from a PCA plot.</p>
</div>
<div id="exercises-1" class="section level2 hasAnchor" number="10.6">
<h2><span class="header-section-number">10.6</span> Exercises<a href="week-9-genome-wide-association-study-gwas.html#exercises-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<blockquote>
<p>1.For this exercise, we want to see which individuals actually show genomic levels of variation at chromosome 6. Using the code that we ran for week 6. Make a PCA plot of the salmon_chr3_19ind.BEAGLE.PL.gz file. Use the function <code>indentify()</code> to find the points that are clustering apart from the other points. Verify that these are males by referring to the table in the Kijas et al. paper <a href="https://www.nature.com/articles/s41598-018-23984-1/tables/1">here</a>. Note their individuals are in the same order as our samples though the names don’t match.</p>
</blockquote>
<details>
<summary>
<span style="color: orange;">Solution</span>
</summary>
<p>
<p>Run pcangsd on our chr6 data</p>
<div class="sourceCode" id="cb502"><pre class="sourceCode html"><code class="sourceCode html"><span id="cb502-1"><a href="week-9-genome-wide-association-study-gwas.html#cb502-1" aria-hidden="true" tabindex="-1"></a>pcangsd -b salmon_chr6_19ind.BEAGLE.PL.gz -o pca6_out -t 28</span></code></pre></div>
<p>In R make a PCA plot</p>
<div class="sourceCode" id="cb503"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb503-1"><a href="week-9-genome-wide-association-study-gwas.html#cb503-1" aria-hidden="true" tabindex="-1"></a><span class="co">#read in the data</span></span>
<span id="cb503-2"><a href="week-9-genome-wide-association-study-gwas.html#cb503-2" aria-hidden="true" tabindex="-1"></a>cov <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">read.table</span>(<span class="st">&quot;pca6_out.cov&quot;</span>))</span>
<span id="cb503-3"><a href="week-9-genome-wide-association-study-gwas.html#cb503-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb503-4"><a href="week-9-genome-wide-association-study-gwas.html#cb503-4" aria-hidden="true" tabindex="-1"></a><span class="co">#compute the eigen values</span></span>
<span id="cb503-5"><a href="week-9-genome-wide-association-study-gwas.html#cb503-5" aria-hidden="true" tabindex="-1"></a>e <span class="ot">&lt;-</span> <span class="fu">eigen</span>(cov)</span>
<span id="cb503-6"><a href="week-9-genome-wide-association-study-gwas.html#cb503-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb503-7"><a href="week-9-genome-wide-association-study-gwas.html#cb503-7" aria-hidden="true" tabindex="-1"></a><span class="co">#how much variation are we explaining here?</span></span>
<span id="cb503-8"><a href="week-9-genome-wide-association-study-gwas.html#cb503-8" aria-hidden="true" tabindex="-1"></a>e<span class="sc">$</span>values<span class="sc">/</span><span class="fu">sum</span>(e<span class="sc">$</span>values)</span></code></pre></div>
<pre><code>##  [1]  0.1379496011  0.1246126748  0.0969103029  0.0920252715  0.0774570788
##  [6]  0.0685079248  0.0605808079  0.0554445095  0.0493884028  0.0444119246
## [11]  0.0423030396  0.0368221717  0.0329494215  0.0316195242  0.0269138675
## [16]  0.0184940705  0.0149661517 -0.0006450986 -0.0107116466</code></pre>
<p>It looks like ~13% is explained by the first axis.</p>
<p>Let’s create the PCA plot:</p>
<div class="sourceCode" id="cb505"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb505-1"><a href="week-9-genome-wide-association-study-gwas.html#cb505-1" aria-hidden="true" tabindex="-1"></a><span class="co">#First, we&#39;ll install a new package called &#39;plotly&#39; that lets us create interactive plots:</span></span>
<span id="cb505-2"><a href="week-9-genome-wide-association-study-gwas.html#cb505-2" aria-hidden="true" tabindex="-1"></a><span class="co">#install.packages(&quot;plotly&quot;)</span></span>
<span id="cb505-3"><a href="week-9-genome-wide-association-study-gwas.html#cb505-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb505-4"><a href="week-9-genome-wide-association-study-gwas.html#cb505-4" aria-hidden="true" tabindex="-1"></a><span class="co">#create a dataframe from the first two PC in object e</span></span>
<span id="cb505-5"><a href="week-9-genome-wide-association-study-gwas.html#cb505-5" aria-hidden="true" tabindex="-1"></a>e_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(e<span class="sc">$</span>vectors[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>])</span></code></pre></div>
<p>make a plot of the first two axes</p>
<div class="sourceCode" id="cb506"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb506-1"><a href="week-9-genome-wide-association-study-gwas.html#cb506-1" aria-hidden="true" tabindex="-1"></a>plotly<span class="sc">::</span><span class="fu">plot_ly</span>(e_df, <span class="at">type =</span> <span class="st">&quot;scatter&quot;</span>, <span class="at">mode =</span> <span class="st">&quot;markers&quot;</span>, <span class="at">x=</span>e_df<span class="sc">$</span>V1, <span class="at">y=</span>e_df<span class="sc">$</span>V2, <span class="at">text=</span><span class="fu">row.names</span>(e_df))</span></code></pre></div>
<div id="htmlwidget-1291c197f20c861a6d30" style="width:672px;height:480px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-1291c197f20c861a6d30">{"x":{"visdat":{"5f608b97ce6":["function () ","plotlyVisDat"]},"cur_data":"5f608b97ce6","attrs":{"5f608b97ce6":{"mode":"markers","x":[-0.519139785802849,-0.421668322093998,0.0728823589597481,-0.00945195017208254,0.164050295585754,0.0374794925622059,0.161893975465155,0.0167349431615272,0.217324975228053,0.205556689134182,0.235649945753755,-0.486973827070867,0.169393148665165,0.0652232253635954,0.0814548483289771,-0.110122505165937,0.214924885835122,0.0484035713690156,-0.100388847510803],"y":[-0.0126347016271288,0.0245300098185363,0.0722020301364673,-0.131409063184534,-0.276071775898028,0.155576443973626,-0.418972580420151,0.0898106283574495,0.368142497812644,0.360218778748402,0.293275329148616,-0.0298863719708143,-0.0478190017745415,0.0999834088658637,0.0181757283239082,0.100042658910541,-0.562109565077771,0.0225581007555425,0.0380157664014052],"text":["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19"],"alpha_stroke":1,"sizes":[10,100],"spans":[1,20],"type":"scatter"}},"layout":{"margin":{"b":40,"l":60,"t":25,"r":10},"xaxis":{"domain":[0,1],"automargin":true,"title":[]},"yaxis":{"domain":[0,1],"automargin":true,"title":[]},"hovermode":"closest","showlegend":false},"source":"A","config":{"modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"data":[{"mode":"markers","x":[-0.519139785802849,-0.421668322093998,0.0728823589597481,-0.00945195017208254,0.164050295585754,0.0374794925622059,0.161893975465155,0.0167349431615272,0.217324975228053,0.205556689134182,0.235649945753755,-0.486973827070867,0.169393148665165,0.0652232253635954,0.0814548483289771,-0.110122505165937,0.214924885835122,0.0484035713690156,-0.100388847510803],"y":[-0.0126347016271288,0.0245300098185363,0.0722020301364673,-0.131409063184534,-0.276071775898028,0.155576443973626,-0.418972580420151,0.0898106283574495,0.368142497812644,0.360218778748402,0.293275329148616,-0.0298863719708143,-0.0478190017745415,0.0999834088658637,0.0181757283239082,0.100042658910541,-0.562109565077771,0.0225581007555425,0.0380157664014052],"text":["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19"],"type":"scatter","marker":{"color":"rgba(31,119,180,1)","line":{"color":"rgba(31,119,180,1)"}},"error_y":{"color":"rgba(31,119,180,1)"},"error_x":{"color":"rgba(31,119,180,1)"},"line":{"color":"rgba(31,119,180,1)"},"xaxis":"x","yaxis":"y","frame":null}],"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<p>Use the cursor to identify the points that are clustered apart from the others. If you select the points on the left most of the screen, you will find they are rows 1, 2, and 12</p>
<p>The other two points that are leftmost from the remaining points are 16 and 19, but according to Table 1 in the Kilas paper, they are females. Therefore, it is possible that more filtering is needed to identify the particular informative SNPs that correlate well with sex in this fish species.</p>
</p>
</details>
<p> </p>
<blockquote>
<ol start="2" style="list-style-type: decimal">
<li>Make a manhattan plot for chromosome 6. Use a phenotype file called <code>pheno_chr6</code>.</li>
</ol>
</blockquote>
<details>
<summary>
<span style="color: orange;">Solution</span>
</summary>
<p>
<div class="sourceCode" id="cb507"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb507-1"><a href="week-9-genome-wide-association-study-gwas.html#cb507-1" aria-hidden="true" tabindex="-1"></a><span class="co">#In the terminal, run the command in angsd again using the chr6 BEAGLE file and the &#39;pheno_chr6&#39; file.</span></span>
<span id="cb507-2"><a href="week-9-genome-wide-association-study-gwas.html#cb507-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb507-3"><a href="week-9-genome-wide-association-study-gwas.html#cb507-3" aria-hidden="true" tabindex="-1"></a><span class="co"># $HOME/angsd/angsd -doMaf 4 -beagle salmon_chr6_19ind.BEAGLE.PL.gz -yBin pheno_chr6 -doAsso 2 -fai Salmon.fai -out chr6_out                 </span></span>
<span id="cb507-4"><a href="week-9-genome-wide-association-study-gwas.html#cb507-4" aria-hidden="true" tabindex="-1"></a>                </span>
<span id="cb507-5"><a href="week-9-genome-wide-association-study-gwas.html#cb507-5" aria-hidden="true" tabindex="-1"></a><span class="co">#and then we can remake our manhattan plot</span></span>
<span id="cb507-6"><a href="week-9-genome-wide-association-study-gwas.html#cb507-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb507-7"><a href="week-9-genome-wide-association-study-gwas.html#cb507-7" aria-hidden="true" tabindex="-1"></a>lrt2 <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="fu">gzfile</span>(<span class="st">&quot;chr6_out.lrt0.gz&quot;</span>), <span class="at">header=</span>T, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb507-8"><a href="week-9-genome-wide-association-study-gwas.html#cb507-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb507-9"><a href="week-9-genome-wide-association-study-gwas.html#cb507-9" aria-hidden="true" tabindex="-1"></a><span class="co">#look at it</span></span>
<span id="cb507-10"><a href="week-9-genome-wide-association-study-gwas.html#cb507-10" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(lrt2)</span>
<span id="cb507-11"><a href="week-9-genome-wide-association-study-gwas.html#cb507-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb507-12"><a href="week-9-genome-wide-association-study-gwas.html#cb507-12" aria-hidden="true" tabindex="-1"></a><span class="co">#remove the values that are not -999 and that are negative</span></span>
<span id="cb507-13"><a href="week-9-genome-wide-association-study-gwas.html#cb507-13" aria-hidden="true" tabindex="-1"></a>lrt_filt2 <span class="ot">&lt;-</span> lrt2[<span class="sc">-</span><span class="fu">c</span>(<span class="fu">which</span>(lrt2<span class="sc">$</span>LRT <span class="sc">==</span> <span class="sc">-</span><span class="dv">999</span>),<span class="fu">which</span>(lrt2<span class="sc">$</span>LRT <span class="sc">&lt;=</span> <span class="dv">0</span>)),]                 </span>
<span id="cb507-14"><a href="week-9-genome-wide-association-study-gwas.html#cb507-14" aria-hidden="true" tabindex="-1"></a>lrt_filt2<span class="sc">$</span>SNP <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;r&quot;</span>,<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(lrt_filt2<span class="sc">$</span>Chromosome), <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb507-15"><a href="week-9-genome-wide-association-study-gwas.html#cb507-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb507-16"><a href="week-9-genome-wide-association-study-gwas.html#cb507-16" aria-hidden="true" tabindex="-1"></a><span class="co">#we also need to make sure we don&#39;t have any tricky values like those below</span></span>
<span id="cb507-17"><a href="week-9-genome-wide-association-study-gwas.html#cb507-17" aria-hidden="true" tabindex="-1"></a>lrt_filt2 <span class="ot">&lt;-</span> lrt_filt2[<span class="sc">!</span>(lrt_filt2<span class="sc">$</span>pvalue<span class="sc">==</span><span class="st">&quot;NaN&quot;</span> <span class="sc">|</span> lrt_filt2<span class="sc">$</span>LRT<span class="sc">==</span><span class="st">&quot;Inf&quot;</span><span class="sc">|</span> lrt_filt2<span class="sc">$</span>pvalue<span class="sc">==</span><span class="st">&quot;Inf&quot;</span>),]</span>
<span id="cb507-18"><a href="week-9-genome-wide-association-study-gwas.html#cb507-18" aria-hidden="true" tabindex="-1"></a>                      </span>
<span id="cb507-19"><a href="week-9-genome-wide-association-study-gwas.html#cb507-19" aria-hidden="true" tabindex="-1"></a><span class="co">#get pvalues</span></span>
<span id="cb507-20"><a href="week-9-genome-wide-association-study-gwas.html#cb507-20" aria-hidden="true" tabindex="-1"></a>lrt_filt2<span class="sc">$</span>pvalue<span class="ot">&lt;-</span><span class="fu">dchisq</span>(lrt_filt2<span class="sc">$</span>LRT, <span class="at">df=</span><span class="dv">1</span>)</span>
<span id="cb507-21"><a href="week-9-genome-wide-association-study-gwas.html#cb507-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb507-22"><a href="week-9-genome-wide-association-study-gwas.html#cb507-22" aria-hidden="true" tabindex="-1"></a><span class="co">#make our plot</span></span>
<span id="cb507-23"><a href="week-9-genome-wide-association-study-gwas.html#cb507-23" aria-hidden="true" tabindex="-1"></a>qqman<span class="sc">::</span><span class="fu">manhattan</span>(lrt_filt2, <span class="at">chr=</span><span class="st">&quot;Chromosome&quot;</span>, <span class="at">bp=</span><span class="st">&quot;Position&quot;</span>, <span class="at">p=</span><span class="st">&quot;pvalue&quot;</span>)</span></code></pre></div>
</p>
</details>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="week-8--fst-and-outlier-analysis.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["MarGen2022.pdf", "MarGen2022.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
